{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1769cdd8-4558-4373-9cc4-1b7aaeac0b4c",
   "metadata": {},
   "source": [
    "# Caution: This notebook uses the OpenAI API. Token costs can add up quickly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5afdec6e-35f4-4052-82f5-9248d3e5a8e4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import openai\n",
    "from pathlib import Path\n",
    "import json\n",
    "import llm2geneset\n",
    "import time\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "63a1f072-52a0-40a8-971a-56f883cddaa6",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "\n",
    "def clean_elements(array):\n",
    "    \"\"\"Use regular expression to remove (GO:xxx) substring,  R-HSA-xxx substrings, and WPxxx substrings\"\"\"\n",
    "    cleaned_array = []\n",
    "    for element in array:\n",
    "        cleaned_element = re.sub(r'\\s*\\(GO:\\d+\\)\\s*|\\s*R-HSA-\\d+\\s*|\\s*WP\\d+\\s*', '', element)\n",
    "        cleaned_array.append(cleaned_element)\n",
    "    return cleaned_array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "1f204121-25b6-4f3f-94f1-58299a95a42b",
   "metadata": {},
   "outputs": [],
   "source": [
    "lib_names = [\"KEGG_2021_Human\", \n",
    "             \"Reactome_2022\", \n",
    "             \"WikiPathway_2023_Human\", \n",
    "             \"GO_Biological_Process_2023\",\n",
    "             \"GO_Molecular_Function_2023\",\n",
    "             \"GO_Cellular_Component_2023\"]\n",
    "lib_names = [\"KEGG_2021_Human\", \n",
    "             \"Reactome_2022\", \n",
    "             \"WikiPathway_2023_Human\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a690cfce-1248-4d52-82a4-7be12b82d339",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "aclient = openai.AsyncClient()\n",
    "models = [\"gpt-3.5-turbo-0125\", \"gpt-4o-2024-05-13\"]\n",
    "for model in models:\n",
    "    for lib_name in lib_names:\n",
    "        (descr, genes) = llm2geneset.read_gmt(\"libs_human/gmt/\" + lib_name + \".txt\")\n",
    "        # Generate cleaned version of gene set description w/o identifiers.\n",
    "        descr_cleaned = clean_elements(descr)\n",
    "        \n",
    "        # Generate genes sets with a system message with role prompt.\n",
    "        start_time1 = time.time()\n",
    "        llm_genes_role = await llm2geneset.get_genes(aclient,\n",
    "                                                     descr_cleaned,\n",
    "                                                     model=model,\n",
    "                                                     prompt_type='basic',\n",
    "                                                     use_sysmsg=True)\n",
    "        end_time1 = time.time()\n",
    "        gen_time_role = end_time1 - start_time1\n",
    "\n",
    "        # Generate gene sets without role prompt.\n",
    "        start_time2 = time.time()\n",
    "        llm_genes_norole = await llm2geneset.get_genes(aclient,\n",
    "                                                       descr_cleaned,\n",
    "                                                       model=model)\n",
    "        end_time2 = time.time()\n",
    "        gen_time_norole = end_time2 - start_time2\n",
    "\n",
    "        # Generate gene sets with reasoning.\n",
    "        start_time3 = time.time()\n",
    "        llm_genes_reason = await llm2geneset.get_genes(aclient,\n",
    "                                                       descr_cleaned,\n",
    "                                                       model=model, \n",
    "                                                       prompt_type='reason')\n",
    "        end_time3 = time.time()\n",
    "        gen_time_reasoning = end_time3 - start_time3\n",
    "\n",
    "        # Generate gene sets with confidence.\n",
    "        start_time4 = time.time()\n",
    "        llm_genes_conf = await llm2geneset.get_genes(aclient,\n",
    "                                                     descr_cleaned,\n",
    "                                                     model=model, \n",
    "                                                     prompt_type='conf')\n",
    "        end_time4 = time.time()\n",
    "        gen_time_conf = end_time4 - start_time4\n",
    "\n",
    "\n",
    "        # Assemble and save generation results.\n",
    "        gen_res = {}\n",
    "        gen_res[\"lib_name\"] = lib_name\n",
    "        gen_res[\"model\"] = model\n",
    "        gen_res[\"descr\"] = descr\n",
    "        gen_res[\"descr_cleaned\"] = descr_cleaned \n",
    "        gen_res[\"gen_time_role\"] = gen_time_role\n",
    "        gen_res[\"gen_time_norole\"] = gen_time_norole\n",
    "        gen_res[\"gen_time_reasoning\"] = gen_time_reasoning\n",
    "        gen_res[\"gen_time_conf\"] = gen_time_conf\n",
    "        gen_res[\"curated_genesets\"] = genes\n",
    "        gen_res[\"llm_genes_role\"] = llm_genes_role\n",
    "        gen_res[\"llm_genes_norole\"] = llm_genes_norole\n",
    "        gen_res[\"llm_genes_reason\"] = llm_genes_reason\n",
    "        gen_res[\"llm_genes_conf\"] = llm_genes_conf\n",
    "    \n",
    "        with open('libs_human/' + model + '/' + lib_name + '.json', 'w') as json_file:\n",
    "            json.dump(gen_res, json_file, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "9084cc22-0c91-45c9-a5ab-0e9f7c6f9c8d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 15%|██████████████████▎                                                                                                        | 271/1818 [00:05<00:16, 94.38it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"Signaling By ERBB2 TMD/JMD Mutants\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Signaling By ERBB2 TMD/JMD Mutants\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\n",
      "        \"gene\": \"ERBB2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"EGFR\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"GRB2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SHC1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3CA\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PTK2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SRC\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"MAPK1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"MAPK3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"GAB1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AKT1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AKT2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AKT3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"STAT3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PLC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 26%|███████████████████████████████▎                                                                                          | 467/1818 [00:07<00:10, 128.47it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"Signaling By PDGFR In Disease\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Signaling By PDGFR In Disease\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\"gene\": \"PDGFRA\"},\n",
      "    {\"gene\": \"PDGFRB\"},\n",
      "    {\"gene\": \"PIK3CA\"},\n",
      "    {\"gene\": \"PIK3R1\"},\n",
      "    {\"gene\": \"PTEN\"},\n",
      "    {\"gene\": \"AKT1\"},\n",
      "    {\"gene\": \"AKT2\"},\n",
      "    {\"gene\": \"AKT3\"},\n",
      "    {\"gene\": \"MTOR\"},\n",
      "    {\"gene\": \"MAPK1\"},\n",
      "    {\"gene\": \"MAPK3\"},\n",
      "    {\"gene\": \"RAF1\"},\n",
      "    {\"gene\": \"BRAF\"},\n",
      "    {\"gene\": \"KRAS\"},\n",
      "    {\"gene\": \"HRAS\"},\n",
      "    {\"gene\": \"NRAS\"},\n",
      "    {\"gene\": \"SHC1\"},\n",
      "    {\"gene\": \"GRB2\"},\n",
      "    {\"gene\": \"SOS1\"},\n",
      "    {\"gene\": \"STAT1\"},\n",
      "    {\"gene\": \"STAT3\"},\n",
      "    {\"gene\": \"STAT5A\"},\n",
      "    {\"gene\": \"STAT5B\"},\n",
      "    {\"gene\": \"PLC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 51%|█████████████████████████████████████████████████████████████▋                                                            | 919/1818 [00:10<00:05, 162.20it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"Potential Therapeutics For SARS\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Potential Therapeutics For SARS\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\n",
      "        \"gene\": \"ACE2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"TMPRSS2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"DPP4\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AGT\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"IFITM3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ADAM17\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"TMEM16F\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"CTS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1818/1818 [00:55<00:00, 32.48it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1818/1818 [00:37<00:00, 49.06it/s]\n",
      " 64%|████████████████████████████████████████████████████████████████████████████▊                                            | 1155/1818 [00:11<00:05, 121.49it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"SARS-CoV Infections\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"SARS-CoV Infections\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\n",
      "        \"gene\": \"ACE2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"TMPRSS2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"DPP4\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"CTS\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 71%|██████████████████████████████████████████████████████████████████████████████████████▍                                  | 1298/1818 [00:12<00:03, 170.90it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "string indices must be integers, not 'str'\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"HIV Life Cycle\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"HIV Life Cycle\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    { \"gene\": \"CD4\" },\n",
      "    { \"gene\": \"CCR5\" },\n",
      "    { \"gene\": \"CXCR4\" },\n",
      "    { \"gene\": \"FURIN\" },\n",
      "    { \"gene\": \"APOBEC3G\" },\n",
      "    { \"gene\": \"APOBEC3H\" },\n",
      "    { \"gene\": \"EIF2AK2\" },\n",
      "    { \"gene\": \"TRIM5\" },\n",
      "    { \"gene\": \"SAMHD1\" },\n",
      "    { \"gene\": \"MOV10\" },\n",
      "    { \"gene\": \"SERINC3\" },\n",
      "    { \"gene\": \"SERINC5\" },\n",
      "    { \"gene\": \"PPIA\" },\n",
      "    { \"gene\": \"PSIP1\" },\n",
      "    { \"gene\": \"LEDGF\" },\n",
      "    { \"gene\": \"NUP153\" },\n",
      "    { \"gene\": \"NUP62\" },\n",
      "    { \"gene\": \"KPNA2\" },\n",
      "    { \"gene\": \"TNPO3\" },\n",
      "    { \"gene\": \"ZNRD1\" },\n",
      "    { \"gene\": \"BST2\" },\n",
      "    { \"gene\": \"CXCL12\" },\n",
      "    { \"gene\": \"TNF\" },\n",
      "    { \"gene\": \"IFNB1\" },\n",
      "    { \"gene\": \"MX1\" },\n",
      "    { \"gene\": \"OAS1\" },\n",
      "    { \"gene\": \"OAS2\" },\n",
      "    { \"gene\": \"OAS3\" },\n",
      "    { \"gene\": \"RSAD2\" },\n",
      "    { \"gene\": \"TARDBP\" },\n",
      "    { \"gene\": \"IL2\" },\n",
      "    { \"gene\": \"TP53\" },\n",
      "    { \"gene\": \"STAT1\" },\n",
      "    { \"gene\": \"STAT2\" },\n",
      "    { \"gene\": \"STAT3\" },\n",
      "    { \"gene\": \"STAT5A\" },\n",
      "    { \"gene\": \"STAT5B\" },\n",
      "    { \"gene\": \"IFITM1\" },\n",
      "    { \"gene\": \"IFITM2\" },\n",
      "    { \"gene\": \"IFITM3\" },\n",
      "    { \"gene\": \"TNFSF10\" },\n",
      "    { \"gene\": \"TNP03\" },\n",
      "    { \"gene\": \"RANBP2\" },\n",
      "    { \"gene\": \"MLH1\" },\n",
      "    { \"gene\": \"PRKDC\" },\n",
      "    \"BRCA1\",\n",
      "    \"XRCC5\",\n",
      "    \"XRCC6\",\n",
      "    \"GSK3B\",\n",
      "    \"SIRT1\",\n",
      "    \"NFKB1\",\n",
      "    \"RELA\",\n",
      "    \"HSP90AA1\",\n",
      "    \"HSPD1\",\n",
      "    \"HSPA1A\",\n",
      "    \"HSPA5\",\n",
      "    \"HSPA8\",\n",
      "    { \"gene\": \"PRMT6\" },\n",
      "    { \"gene\": \"DHFR\" },\n",
      "    { \"gene\": \"IL7\" },\n",
      "    { \"gene\": \"IL15\" },\n",
      "    { \"gene\": \"CYP3A4\" },\n",
      "    { \"gene\": \"CYPA\" },\n",
      "    { \"gene\": \"PTPN6\" }\n",
      "]\n",
      "```\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 76%|████████████████████████████████████████████████████████████████████████████████████████████▏                            | 1385/1818 [00:13<00:03, 142.95it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "list indices must be integers or slices, not str\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"FGFR3b Ligand Binding And Activation\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"FGFR3b Ligand Binding And Activation\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\"gene\": \"FGFR3\"},\n",
      "    {\"gene\": \"FGF1\"},\n",
      "    {\"gene\": \"FGF2\"},\n",
      "    {\"gene\": \"FGF3\"},\n",
      "    {\"gene\": \"FGF4\"},\n",
      "    {\"gene\": \"FGF5\"},\n",
      "    {\"gene\": \"FGF6\"},\n",
      "    {\"gene\": \"FGF7\"},  // Also known as KGFR\n",
      "    {\"gene\": \"FGF8\"},\n",
      "    {\"gene\": \"FGF9\"},\n",
      "    {\"gene\": \"FGF10\"},\n",
      "    {\"gene\": \"FGF16\"},\n",
      "    {\"gene\": \"FGF18\"},\n",
      "    {\"gene\": \"FGF20\"},\n",
      "    {\"gene\": \"FGF21\"},\n",
      "    {\"gene\": \"FGF22\"},\n",
      "    {\"gene\": \"FGF23\"},\n",
      "    {\"gene\": \"HS6ST1\"},\n",
      "    {\"gene\": \"HS6ST2\"},\n",
      "    {\"gene\": \"HS6ST3\"},\n",
      "    {\"gene\": \"HS2ST1\"},\n",
      "    {\"gene\": \"HSPG2\"},\n",
      "    {\"gene\": \"PDPK1\"},\n",
      "    {\"gene\": \"GRB2\"},\n",
      "    {\"gene\": \"SHC1\"},\n",
      "    {\"gene\": \"GAB1\"},\n",
      "    {\"gene\": \"SRC\"},\n",
      "    {\"gene\": \"PIK3CA\"},\n",
      "    {\"gene\": \"PIK3CB\"},\n",
      "    {\"gene\": \"PIK3CD\"},\n",
      "    {\"gene\": \"PIK3CG\"},\n",
      "    {\"gene\": \"PIK3R1\"},\n",
      "    {\"gene\": \"PIK3R2\"},\n",
      "    {\"gene\": \"PIK3R3\"},\n",
      "    {\"gene\": \"SOS1\"},\n",
      "    {\"gene\": \"RASGRF1\"},\n",
      "    {\"gene\": \"HRAS\"},\n",
      "    {\"gene\": \"KRAS\"},\n",
      "    {\"gene\": \"NRAS\"},\n",
      "    {\"gene\": \"RAF1\"},\n",
      "    {\"gene\": \"BRAF\"},\n",
      "    {\"gene\": \"MAP2K1\"},\n",
      "    {\"gene\": \"MAP2K2\"},\n",
      "    {\"gene\": \"MAPK1\"},\n",
      "    {\"gene\": \"MAPK3\"},\n",
      "    {\"gene\": \"PLCG1\"},\n",
      "    {\"gene\": \"PLCG2\"},\n",
      "    {\"gene\": \"STAT1\"},\n",
      "    {\"gene\": \"STAT3\"}\n",
      "]\n",
      "```\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1818/1818 [00:59<00:00, 30.61it/s]\n",
      " 73%|███████████████████████████████████████████████████████████████████████████████████████▊                                 | 1320/1818 [00:12<00:03, 156.61it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      "List all the known genes directly and indirectly involved in the following biological process or cellular component \"\"\"Activation Of TRKA Receptors\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Activation Of TRKA Receptors\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\n",
      "        \"gene\": \"NTRK1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"NGF\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SHC1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"GRB2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SOS1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3R1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3CA\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PLC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1818/1818 [00:51<00:00, 35.05it/s]\n"
     ]
    }
   ],
   "source": [
    "# Generate gene sets for ensembling.\n",
    "aclient = openai.AsyncClient()\n",
    "models = [\"gpt-3.5-turbo-0125\", \"gpt-4o-2024-05-13\"]\n",
    "for model in models:\n",
    "    for lib_name in lib_names:\n",
    "        with open(\"libs_human/\" + model + \"/\" + lib_name + \".json\") as f:\n",
    "            gen_res = json.load(f)\n",
    "\n",
    "        for i in range(4):\n",
    "            start_time = time.time()\n",
    "            gen_res[\"llm_ensemble_\" + str(i)] = await llm2geneset.get_genes(aclient,\n",
    "                                                                            gen_res[\"descr_cleaned\"],\n",
    "                                                                            model=model)\n",
    "            end_time = time.time()\n",
    "            gen_res[\"gen_time_ensemble_\" + str(i)] = end_time - start_time\n",
    "\n",
    "        with open('libs_human/' + model + '/' + lib_name + '.json', 'w') as json_file:\n",
    "            json.dump(gen_res, json_file, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "e71abd13-70f0-4f9c-b718-efd94c4ce4a3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gpt-4o-2024-05-13\n",
      "Reactome_2022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████| 1818/1818 [15:10<00:00,  2.00it/s]\n",
      " 80%|███████████████████████████████        | 1450/1818 [12:06<03:07,  1.97it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Attempt 0 failed with error: Server disconnected\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████| 1818/1818 [15:11<00:00,  2.00it/s]\n",
      " 50%|███████████████████▎                   | 903/1818 [00:12<00:06, 131.39it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "string indices must be integers, not 'str'\n",
      "BACKGROUND Neuromyelitis optica is an inflammatory demyelinating disease with generally poor prognosis that selectively targets optic nerves and spinal cord. It is commonly misdiagnosed as multiple sclerosis. Neither disease has a distinguishing biomarker, but optimum treatments differ. The relation of neuromyelitis optica to optic-spinal multiple sclerosis in Asia is uncertain. We assessed the capacity of a putative marker for neuromyelitis optica (NMO-IgG) to distinguish neuromyelitis optica and related disorders from multiple sclerosis. METHODS Indirect immunofluorescence with a composite substrate of mouse tissues identified a distinctive NMO-IgG staining pattern, which we characterised further by dual immunostaining. We tested masked serum samples from 102 North American patients with neuromyelitis optica or with syndromes that suggest high risk of the disorder, and 12 Japanese patients with optic-spinal multiple sclerosis. Control patients had multiple sclerosis, other myelopathies, optic neuropathies, and miscellaneous disorders. We also established clinical diagnoses for 14 patients incidentally shown to have NMO-IgG among 85000 tested for suspected paraneoplastic autoimmunity. FINDINGS NMO-IgG outlines CNS microvessels, pia, subpia, and Virchow-Robin space. It partly colocalises with laminin. Sensitivity and specificity were 73% (95% CI 60-86) and 91% (79-100) for neuromyelitis optica and 58% (30-86) and 100% (66-100) for optic-spinal multiple sclerosis. NMO-IgG was detected in half of patients with high-risk syndromes. Of 14 seropositive cases identified incidentally, 12 had neuromyelitis optica or a high-risk syndrome for the disease. INTERPRETATION NMO-IgG is a specific marker autoantibody of neuromyelitis optica and binds at or near the blood-brain barrier. It distinguishes neuromyelitis optica from multiple sclerosis. Asian optic-spinal multiple sclerosis seems to be the same as neuromyelitis optica.\n",
      "You are an expert that wrote the above abstract on the biological process or cellular component \"\"\"Miscellaneous Substrates\"\"\". List all the known genes directly and indirectly involved in \"\"\"Miscellaneous Substrates\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Miscellaneous Substrates\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": [\n",
      "        {\n",
      "            \"gene\": \"NMO-IgG\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMC1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMC2\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMC3\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMC4\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMC5\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMA1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMA2\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMA3\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMA4\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMA5\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMB1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMB2\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMB3\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LAMB4\"\n",
      "        }\n",
      "    ]\n",
      "}\n",
      "```\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 79%|██████████████████████████████▏       | 1445/1818 [00:16<00:03, 104.11it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      " FGFR signaling is deregulated in many human cancers, and FGFR is considered a valid target in FGFR-deregulated tumors. Here, we examine the preclinical profile of futibatinib (TAS-120; 1-[(3S)-[4-amino-3-[(3,5-dimethoxyphenyl)ethynyl]-1H-pyrazolo[3, 4-d] pyrimidin-1-yl]-1-pyrrolidinyl]-2-propen-1-one), a structurally novel, irreversible FGFR1-4 inhibitor. Among a panel of 296 human kinases, futibatinib selectively inhibited FGFR1-4 with IC50 values of 1.4 to 3.7 nmol/L. Futibatinib covalently bound the FGFR kinase domain, inhibiting FGFR phosphorylation and, in turn, downstream signaling in FGFR-deregulated tumor cell lines. Futibatinib exhibited potent, selective growth inhibition of several tumor cell lines (gastric, lung, multiple myeloma, bladder, endometrial, and breast) harboring various FGFR genomic aberrations. Oral administration of futibatinib led to significant dose-dependent tumor reduction in various FGFR-driven human tumor xenograft models, and tumor reduction was associated with sustained FGFR inhibition, which was proportional to the administered dose. The frequency of appearance of drug-resistant clones was lower with futibatinib than a reversible ATP-competitive FGFR inhibitor, and futibatinib inhibited several drug-resistant FGFR2 mutants, including the FGFR2 V565I/L gatekeeper mutants, with greater potency than any reversible FGFR inhibitors tested (IC50, 1.3-50.6 nmol/L). These results indicate that futibatinib is a novel orally available, potent, selective, and irreversible inhibitor of FGFR1-4 with a broad spectrum of antitumor activity in cell lines and xenograft models. These findings provide a strong rationale for testing futibatinib in patients with tumors oncogenically driven by FGFR genomic aberrations, with phase I to III trials ongoing. SIGNIFICANCE: Preclinical characterization of futibatinib, an irreversible FGFR1-4 inhibitor, demonstrates selective and potent antitumor activity against FGFR-deregulated cancer cell lines and xenograft models, supporting clinical evaluation in patients with FGFR-driven tumors. GRAPHICAL ABSTRACT: http://cancerres.aacrjournals.org/content/canres/80/22/4986/F1.large.jpg.\n",
      "You are an expert that wrote the above abstract on the biological process or cellular component \"\"\"FGFR2 Mutant Receptor Activation\"\"\". List all the known genes directly and indirectly involved in \"\"\"FGFR2 Mutant Receptor Activation\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"FGFR2 Mutant Receptor Activation\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\n",
      "        \"gene\": \"FGFR2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"FGF1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"FGF2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"FGF7\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"FGF10\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3CA\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3CB\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3CD\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PIK3CG\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AKT1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AKT2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"AKT3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PTEN\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"KRAS\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"HRAS\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"NRAS\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"BRAF\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"RAF1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"MAP2K1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"MAP2K2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"MAPK1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"MAPK3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"GRB2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SOS1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SOS2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"GAB1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PLC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 83%|████████████████████████████████▎      | 1508/1818 [00:17<00:03, 90.99it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "No code blocks found\n",
      " Circadian rhythms are biological systems that synchronize cellular circadian oscillators with the organism's daily feeding-fasting or rest-activity cycles in mammals. Circadian rhythms regulate nutrient absorption and utilization at the cellular level and are closely related to obesity and metabolic disorders. Bile acids are important modulators that facilitate nutrient absorption and regulate energy metabolism. Here, we provide an overview of the current connections and future perspectives between the circadian clock and bile acid metabolism as well as related metabolic diseases. Feeding and fasting cycles influence bile acid pool size and composition, and bile acid signaling can respond to acute lipid and glucose utilization and mediate energy balance. Disruption of circadian rhythms such as shift work, irregular diet, and gene mutations can contribute to altered bile acid metabolism and heighten obesity risk. High-fat diets, alcohol, and gene mutations related to bile acid signaling result in desynchronized circadian rhythms. Gut microbiome also plays a role in connecting circadian rhythms with bile acid metabolism. The underlying mechanism of how circadian rhythms interact with bile acid metabolism has not been fully explored. Sustaining bile acid homeostasis based on circadian rhythms may be a potential therapy to alleviate metabolic disturbance.\n",
      "You are an expert that wrote the above abstract on the biological process or cellular component \"\"\"Bile Acid And Bile Salt Metabolism\"\"\". List all the known genes directly and indirectly involved in \"\"\"Bile Acid And Bile Salt Metabolism\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Bile Acid And Bile Salt Metabolism\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "[\n",
      "    {\n",
      "        \"gene\": \"CYP7A1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"CYP7B1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"CYP27A1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"CYP8B1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"NR1H4\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"FXR\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"NR1I2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"PXR\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"NR1H3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"LXR\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"FDFT1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"HMGCR\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"HMGCS1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SREBF1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SREBF2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCB11\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"SLC10A2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCG5\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCG8\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"OSTA\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"OSTB\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCB1\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCC2\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCC3\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"ABCC4\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"NTCP\"\n",
      "    },\n",
      "    {\n",
      "        \"gene\": \"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████| 1818/1818 [00:59<00:00, 30.43it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gpt-4o-2024-05-13\n",
      "WikiPathway_2023_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 801/801 [06:41<00:00,  2.00it/s]\n",
      "100%|█████████████████████████████████████████| 801/801 [06:41<00:00,  2.00it/s]\n",
      " 21%|████████▌                                | 168/801 [00:06<00:09, 67.53it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "string indices must be integers, not 'str'\n",
      " Photodynamic therapy (PDT) is a non-to-minimally invasive treatment modality that utilizes photoactivatable drugs called photosensitizers to disrupt tumors with locally photoproduced reactive oxygen species (ROS). Photosensitizer activation by light results in hyperoxidative stress and subsequent tumor cell death, vascular shutdown and hypoxia, and an antitumor immune response. However, sublethally afflicted tumor cells initiate several survival mechanisms that account for decreased PDT efficacy. The hypoxia inducible factor 1 (HIF-1) pathway is one of the most effective cell survival pathways that contributes to cell recovery from PDT-induced damage. Several hundred target genes of the HIF-1 heterodimeric complex collectively mediate processes that are involved in tumor cell survival directly and indirectly (e.g., vascularization, glucose metabolism, proliferation, and metastasis). The broad spectrum of biological ramifications culminating from the activation of HIF-1 target genes reflects the importance of HIF-1 in the context of therapeutic recalcitrance. This chapter elaborates on the involvement of HIF-1 in cancer biology, the hypoxic response mechanisms, and the role of HIF-1 in PDT. An overview of inhibitors that either directly or indirectly impede HIF-1-mediated survival signaling is provided. The inhibitors may be used as pharmacological adjuvants in combination with PDT to augment therapeutic efficacy.\n",
      "You are an expert that wrote the above abstract on the biological process or cellular component \"\"\"Photodynamic Therapy Induced HIF 1 Survival Signaling\"\"\". List all the known genes directly and indirectly involved in \"\"\"Photodynamic Therapy Induced HIF 1 Survival Signaling\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Photodynamic Therapy Induced HIF 1 Survival Signaling\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": [\n",
      "        {\n",
      "            \"gene\": \"HIF1A\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"VEGFA\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"GLUT1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"ALDOA\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"LDHA\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"PGK1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"BNIP3\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"EGLN1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"EGLN2\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"EGLN3\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"PDK1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"EPAS1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"SLC2A1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"ENO1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"HK2\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"PFKFB3\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"PFKFB4\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"CA9\"\n",
      "        }\n",
      "    ]\n",
      "}\n",
      "```\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 25%|██████████▏                              | 199/801 [00:06<00:07, 84.36it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "string indices must be integers, not 'str'\n",
      " Zinc oxide nanoparticles (ZnONPs) hold great promise for biomedical applications. Previous studies have revealed that ZnONPs exposure can induce toxicity in endothelial cells, but the underlying mechanisms have not been fully elucidated. In this study, we report that ZnONPs can induce ferroptosis of both HUVECs and EA.hy926 cells, as evidenced by the elevation of intracellular iron levels, lipid peroxidation and cell death in a dose- and time-dependent manner. In addition, both the lipid reactive oxygen species (ROS) scavenger ferrostatin-1 and the iron chelator deferiprone attenuated ZnONPs-induced cell death. Intriguingly, we found that ZnONPs-induced ferroptosis is macroautophagy/autophagy-dependent, because the inhibition of autophagy with a pharmacological inhibitor or by ATG5 gene knockout profoundly mitigated ZnONPs-induced ferroptosis. We further demonstrated that NCOA4 (nuclear receptor coactivator 4)-mediated ferritinophagy (autophagic degradation of the major intracellular iron storage protein ferritin) was required for the ferroptosis induced by ZnONPs, by showing that NCOA4 knockdown can reduce the intracellular iron level and lipid peroxidation, and subsequently alleviate ZnONPs-induced cell death. Furthermore, we showed that ROS originating from mitochondria (mtROS) probably activated the AMPK-ULK1 axis to trigger ferritinophagy. Most importantly, pulmonary ZnONPs exposure caused vascular inflammation and ferritinophagy in mice, and ferrostatin-1 supplementation significantly reversed the vascular injury induced by pulmonary ZnONPs exposure. Overall, our study indicates that ferroptosis is a novel mechanism for ZnONPs-induced endothelial cytotoxicity, and that NCOA4-mediated ferritinophagy is required for ZnONPs-induced ferroptotic cell death.Abbreviations: 3-MA: 3-methyladenine; ACTB: Actin beta; AMPK: AMP-activated protein kinase; ATG: Autophagy-related; BafA1: Bafilomycin A1; CQ: Choloroquine; DFP: Deferiprone; FACS: Fluorescence-activated cell sorting; Fer-1: Ferrostatin-1; FTH1: Ferritin heavy chain 1; GPX4: Glutathione peroxidase 4; GSH: Glutathione; IREB2/IRP2: Iron responsive element binding protein 2; LIP: Labile iron pool; MAP1LC3B/LC3B: Microtubule associated protein 1 light chain 3 beta; MTOR: Mechanistic target of rapamycin kinase; NCOA4: Nuclear receptor coactivator 4; NFE2L2/NRF2: Nuclear factor, erythroid 2 like 2; PGSK: Phen Green™ SK; ROS: Reactive oxygen species; siRNA: Small interfering RNA; SQSTM1/p62: Sequestosome 1; TEM: Transmission electron microscopy; ULK1: Unc-51 like autophagy activating kinase 1; ZnONPs: Zinc oxide nanoparticles.\n",
      "You are an expert that wrote the above abstract on the biological process or cellular component \"\"\"Nanoparticle Triggered Autophagic Cell Death\"\"\". List all the known genes directly and indirectly involved in \"\"\"Nanoparticle Triggered Autophagic Cell Death\"\"\". Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"gene\": {\n",
      "                \"type\": \"string\",\n",
      "            }\n",
      "        },\n",
      "        \"required\": [\"gene\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "The field `gene` is a gene involved in the following biological process or cellular component: \"\"\"Nanoparticle Triggered Autophagic Cell Death\"\"\". Use the HUGO Gene Nomenclature Committee (HGNC) gene abbreviations. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": [\n",
      "        {\n",
      "            \"gene\": \"NCOA4\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"AMPK\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"ULK1\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"ATG5\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"MTOR\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"MAP1LC3B\"\n",
      "        },\n",
      "        {\n",
      "            \"gene\": \"SQSTM1\"\n",
      "        }\n",
      "    ]\n",
      "}\n",
      "```\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████████████████████████████████████| 801/801 [00:33<00:00, 23.75it/s]\n"
     ]
    }
   ],
   "source": [
    "# Generate gene sets using RAG.\n",
    "aclient = openai.AsyncClient()\n",
    "#models = [\"gpt-3.5-turbo-0125\", \"gpt-4o-2024-05-13\"]\n",
    "models = [\"gpt-4o-2024-05-13\"]\n",
    "\n",
    "for model in models:    \n",
    "    for lib_name in lib_names:\n",
    "        print(model)\n",
    "        print(lib_name)\n",
    "        with open(\"libs_human/\" + model + \"/\" + lib_name + \".json\") as f:\n",
    "            gen_res = json.load(f)\n",
    "        descr_cleaned = gen_res[\"descr_cleaned\"]\n",
    "\n",
    "\n",
    "        start_time = time.time()\n",
    "        pmids = await llm2geneset.esearch_async(descr_cleaned, \"pubmed\", 1)\n",
    "        pmids_cleaned = []\n",
    "        for i in pmids:\n",
    "            if len(i)>0:\n",
    "                pmids_cleaned.append(i[0])\n",
    "            else:\n",
    "                pmids_cleaned.append(\"\")\n",
    "\n",
    "\n",
    "        abstracts = await llm2geneset.efetch_pubmed_async([[i] for i in pmids_cleaned])\n",
    "        context = []\n",
    "        for x in abstracts:\n",
    "            if len(x) > 0:\n",
    "                context.append(x[0]['abstract'])\n",
    "            else:\n",
    "                context.append(\"\")\n",
    "        \n",
    "        \n",
    "        llm_genes_rag = await llm2geneset.get_genes_context(aclient,\n",
    "                                                                          context,\n",
    "                                                                          descr_cleaned, \n",
    "                                                                          model = model)\n",
    "        end_time = time.time()\n",
    "        gen_time_rag = end_time - start_time\n",
    "\n",
    "        gen_res[\"gen_time_rag\"] = gen_time_rag\n",
    "        gen_res[\"llm_genes_rag\"] = llm_genes_rag\n",
    "        \n",
    "        with open('libs_human/' + model + '/' + lib_name + '.json', 'w') as json_file:\n",
    "            json.dump(gen_res, json_file, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "95afa9e2-f975-4105-8f9e-73b5b5450732",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:27<00:00, 27.12s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name: One-carbon metabolism and serine biosynthesis\n",
      "LLM self-assessed confidence: 0.97\n",
      "\n",
      "1. SRR, PSAT1, and PSPH are involved in the serine biosynthesis pathway. SRR (serine racemase) catalyzes the isomerization of L-serine to D-serine, which acts as a coagonist with glutamate on NMDA receptors. PSAT1 (phosphoserine aminotransferase 1) and PSPH (phosphoserine phosphatase) are enzymes that convert 3-phosphohydroxypyruvate to L-serine through an intermediate, 3-phosphoserine. This pathway links glycolysis to the synthesis of this non-essential amino acid, impacting cell proliferation and growth.\n",
      "\n",
      "2. PHGDH (phosphoglycerate dehydrogenase) is another pivotal enzyme in the serine biosynthesis pathway, converting 3-phosphoglycerate into 3-phosphohydroxypyruvate. This reaction is the first committed step in the synthesis of serine from glycolytic intermediates and is crucial for the provision of serine, glycine, and one-carbon units essential for various biosynthetic processes.\n",
      "\n",
      "3. SHMT1 and SHMT2 (serine hydroxymethyltransferase 1 and 2) are enzymes involved in the interconversion of serine and glycine with the concurrent production of one-carbon units, specifically tetrahydrofolate (THF)-bound one-carbon units. This reaction integrates the serine biosynthesis pathway with the folate-dependent one-carbon metabolism, which is essential for the synthesis of nucleotides, purines, thymidylate, and the regeneration of methionine from homocysteine.\n",
      "\n",
      "4. TYMS (thymidylate synthase) and DHFR (dihydrofolate reductase) are crucial enzymes in the folate cycle, playing essential roles in DNA synthesis and repair. TYMS catalyzes the conversion of deoxyuridylate (dUMP) to deoxythymidylate (dTMP), providing the sole de novo source of thymidylate, a nucleotide required for DNA replication and repair. DHFR reduces dihydrofolate to tetrahydrofolate, replenishing the pool of THF necessary for numerous one-carbon transfer reactions.\n",
      "\n",
      "Integration and Functional Context:\n",
      "The interacting proteins in this set majorly contribute to one-carbon metabolism and serine biosynthesis. Both pathways are interconnected and essential for cellular proliferation, nucleotide synthesis, and methylation reactions. The serine biosynthesis pathway provides critical intermediates that fuel the folate cycle, impacting DNA synthesis and repair. This coupling ensures a balanced supply of serine and one-carbon units, necessary for both proliferative signaling and the synthesis of biomolecules. The synergistic actions of these proteins maintain cellular homeostasis, growth, and division, linking metabolic and biosynthetic pathways in a coherent manner.\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[{'name': 'One-carbon metabolism and serine biosynthesis',\n",
       "  'conf': 0.97,\n",
       "  'annot': ['1. SRR, PSAT1, and PSPH are involved in the serine biosynthesis pathway. SRR (serine racemase) catalyzes the isomerization of L-serine to D-serine, which acts as a coagonist with glutamate on NMDA receptors. PSAT1 (phosphoserine aminotransferase 1) and PSPH (phosphoserine phosphatase) are enzymes that convert 3-phosphohydroxypyruvate to L-serine through an intermediate, 3-phosphoserine. This pathway links glycolysis to the synthesis of this non-essential amino acid, impacting cell proliferation and growth.',\n",
       "   '2. PHGDH (phosphoglycerate dehydrogenase) is another pivotal enzyme in the serine biosynthesis pathway, converting 3-phosphoglycerate into 3-phosphohydroxypyruvate. This reaction is the first committed step in the synthesis of serine from glycolytic intermediates and is crucial for the provision of serine, glycine, and one-carbon units essential for various biosynthetic processes.',\n",
       "   '3. SHMT1 and SHMT2 (serine hydroxymethyltransferase 1 and 2) are enzymes involved in the interconversion of serine and glycine with the concurrent production of one-carbon units, specifically tetrahydrofolate (THF)-bound one-carbon units. This reaction integrates the serine biosynthesis pathway with the folate-dependent one-carbon metabolism, which is essential for the synthesis of nucleotides, purines, thymidylate, and the regeneration of methionine from homocysteine.',\n",
       "   '4. TYMS (thymidylate synthase) and DHFR (dihydrofolate reductase) are crucial enzymes in the folate cycle, playing essential roles in DNA synthesis and repair. TYMS catalyzes the conversion of deoxyuridylate (dUMP) to deoxythymidylate (dTMP), providing the sole de novo source of thymidylate, a nucleotide required for DNA replication and repair. DHFR reduces dihydrofolate to tetrahydrofolate, replenishing the pool of THF necessary for numerous one-carbon transfer reactions.'],\n",
       "  'in_toks': 992,\n",
       "  'out_toks': 605,\n",
       "  'ntries': 1}]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Generate gene sets using RAG.\n",
    "aclient = openai.AsyncClient()\n",
    "set0 = [['CETP', 'TRNP1', 'P4HB', 'APOF', 'LPA' , 'APOB']]\n",
    "set0 = [['ADIPOQ'\n",
    "        'CPT1A', 'LEP', 'LEPR', 'ADIPOR1', 'PRKAG1',  'ADIPOR2', 'PRKAB1',  'PRKAA1',  'ACACA']]\n",
    "set0 = [['SRR'   ,  'PSAT1'  , 'SHMT2' ,  'SHMT1' ,  'PHGDH' ,  'TYMS' ,   'DHFR',   ' PSPH']]\n",
    "await llm2geneset.gsai(aclient,set0)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
