{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import llm2geneset\n",
    "import openai\n",
    "import pandas as pd\n",
    "from scipy.stats import hypergeom\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "import re\n",
    "import numpy as np\n",
    "client = openai.Client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load(gmt_file):\n",
    "    gmt = llm2geneset.read_gmt(gmt_file)\n",
    "    def clean_elements(array):\n",
    "        \"\"\"Use regular expression to remove (GO:xxx) substring,  \n",
    "        R-HSA-xxx substrings, and WPxxx substrings\"\"\"\n",
    "        cleaned_array = []\n",
    "        for element in array:\n",
    "            cleaned_element = re.sub(r'\\s*\\(GO:\\d+\\)\\s*|\\s*R-HSA-\\d+\\s*|\\s*WP\\d+\\s*', '', element)\n",
    "            cleaned_array.append(cleaned_element)\n",
    "        return cleaned_array\n",
    "    gmt[\"descr_cleaned\"] = clean_elements(gmt[\"descr\"]) \n",
    "    gmtemb = llm2geneset.get_embeddings(client, gmt[\"descr_cleaned\"])\n",
    "    return (gmt, gmtemb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "wiki, wikiemb = load(\"libs_human/gmt/Reactome_2022.txt\")\n",
    "kegg, keggemb = load(\"libs_human/gmt/KEGG_2021_Human.txt\")\n",
    "react, reactemb = load(\"libs_human/gmt/Reactome_2022.txt\")\n",
    "go, goemb = load(\"libs_human/gmt/GO_Biological_Process_2023.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Convert lists to matrices\n",
    "matrix1 = np.vstack(wikiemb)  # Shape (m, d), where m is the number of vectors\n",
    "matrix2 = np.vstack(goemb)  # Shape (n, d), where n is the number of vectors\n",
    "\n",
    "# Compute pairwise dot products using matrix multiplication\n",
    "A = matrix1 @ matrix2.T  # Shape (m, n)\n",
    "max_cols = np.argmax(A, axis=1)\n",
    "max_values = A[np.arange(A.shape[0]), max_cols]\n",
    "mask = max_values > 0.7\n",
    "# 4. Filter down to those rows\n",
    "f_cols = max_cols[mask].tolist()   # column indices of max\n",
    "f_vals = max_values[mask]     # corresponding max values\n",
    "f_rows = np.where(mask)[0].tolist()   # (optional) which row indices\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ATF6 (ATF6-alpha) Activates Chaperone Genes R-HSA-381183\tATF6-mediated Unfolded Protein Response (GO:0036500)\n",
      "\n",
      "ATF6 (ATF6-alpha) Activates Chaperones R-HSA-381033\tATF6-mediated Unfolded Protein Response (GO:0036500)\n",
      "\n",
      "AUF1 (hnRNP D0) Binds And Destabilizes mRNA R-HSA-450408\tmRNA Destabilization (GO:0061157)\n",
      "\n",
      "Acetylcholine Regulates Insulin Secretion R-HSA-399997\tRegulation Of Insulin Secretion (GO:0050796)\n",
      "\n",
      "Activation Of NF-kappaB In B Cells R-HSA-1169091\tActivation Of NF-kappaB-inducing Kinase Activity (GO:0007250)\n",
      "\n",
      "Activation Of RAC1 R-HSA-428540\tRegulation Of Rac Protein Signal Transduction (GO:0035020)\n",
      "\n",
      "Activation Of TRKA Receptors R-HSA-187015\tNeurotrophin TRK Receptor Signaling Pathway (GO:0048011)\n",
      "\n",
      "Acyl Chain Remodeling Of CL R-HSA-1482798\tCardiolipin Acyl-Chain Remodeling (GO:0035965)\n",
      "\n",
      "Acyl Chain Remodeling Of DAG And TAG R-HSA-1482883\tPhosphatidylinositol Acyl-Chain Remodeling (GO:0036149)\n",
      "\n",
      "Acyl Chain Remodelling Of PC R-HSA-1482788\tPhosphatidylcholine Acyl-Chain Remodeling (GO:0036151)\n",
      "\n",
      "Acyl Chain Remodelling Of PE R-HSA-1482839\tPhosphatidylethanolamine Acyl-Chain Remodeling (GO:0036152)\n",
      "\n",
      "Acyl Chain Remodelling Of PG R-HSA-1482925\tPhosphatidylglycerol Acyl-Chain Remodeling (GO:0036148)\n",
      "\n",
      "Acyl Chain Remodelling Of PI R-HSA-1482922\tPhosphatidylinositol Acyl-Chain Remodeling (GO:0036149)\n",
      "\n",
      "Acyl Chain Remodelling Of PS R-HSA-1482801\tPhosphatidylserine Acyl-Chain Remodeling (GO:0036150)\n",
      "\n",
      "Adenylate Cyclase Activating Pathway R-HSA-170660\tAdenylate Cyclase-Activating Adrenergic Receptor Signaling Pathway (GO:0071880)\n",
      "\n",
      "Adenylate Cyclase Inhibitory Pathway R-HSA-170670\tAdenylate Cyclase-Inhibiting G Protein-Coupled Receptor Signaling Pathway (GO:0007193)\n",
      "\n",
      "Adherens Junctions Interactions R-HSA-418990\tAdherens Junction Assembly (GO:0034333)\n",
      "\n",
      "Aggrephagy R-HSA-9646399\tAggrephagy (GO:0035973)\n",
      "\n",
      "Alpha-oxidation Of Phytanate R-HSA-389599\tFatty Acid Alpha-Oxidation (GO:0001561)\n",
      "\n",
      "Amino Acid Transport Across Plasma Membrane R-HSA-352230\tAmino Acid Import Across Plasma Membrane (GO:0089718)\n",
      "\n",
      "Amyloid Fiber Formation R-HSA-977225\tAmyloid Fibril Formation (GO:1990000)\n",
      "\n",
      "Androgen Biosynthesis R-HSA-193048\tAndrogen Biosynthetic Process (GO:0006702)\n",
      "\n",
      "Antigen Presentation: Folding, Assembly, Peptide Loading Of Class I MHC R-HSA-983170\tPeptide Antigen Assembly With MHC Protein Complex (GO:0002501)\n",
      "\n",
      "Antigen Activates B Cell Receptor Leading To Second Messenger Generation R-HSA-983695\tB Cell Activation (GO:0042113)\n",
      "\n",
      "Antigen processing-Cross Presentation R-HSA-1236975\tAntigen Processing And Presentation Of Exogenous Peptide Antigen (GO:0002478)\n",
      "\n",
      "Antimicrobial Peptides R-HSA-6803157\tAntimicrobial Humoral Immune Response Mediated By Antimicrobial Peptide (GO:0061844)\n",
      "\n",
      "Apoptosis Induced DNA Fragmentation R-HSA-140342\tApoptotic DNA Fragmentation (GO:0006309)\n",
      "\n",
      "Apoptotic Execution Phase R-HSA-75153\tRegulation Of Execution Phase Of Apoptosis (GO:1900117)\n",
      "\n",
      "Arachidonic Acid Metabolism R-HSA-2142753\tArachidonic Acid Metabolic Process (GO:0019369)\n",
      "\n",
      "Asparagine N-linked Glycosylation R-HSA-446203\tProtein N-linked Glycosylation Via Asparagine (GO:0018279)\n",
      "\n",
      "Aspartate And Asparagine Metabolism R-HSA-8963693\tAspartate Metabolic Process (GO:0006531)\n",
      "\n",
      "Attachment Of GPI Anchor To uPAR R-HSA-162791\tAttachment Of GPI Anchor To Protein (GO:0016255)\n",
      "\n",
      "Autophagy R-HSA-9612973\tMacroautophagy (GO:0016236)\n",
      "\n",
      "Axon Guidance R-HSA-422475\tAxon Guidance (GO:0007411)\n",
      "\n",
      "B-WICH Complex Positively Regulates rRNA Expression R-HSA-5250924\tPositive Regulation Of rRNA Processing (GO:2000234)\n",
      "\n",
      "Base Excision Repair R-HSA-73884\tBase-Excision Repair (GO:0006284)\n",
      "\n",
      "Base-Excision Repair, AP Site Formation R-HSA-73929\tBase-Excision Repair, AP Site Formation (GO:0006285)\n",
      "\n",
      "Beta-catenin Independent WNT Signaling R-HSA-3858494\tNon-Canonical Wnt Signaling Pathway (GO:0035567)\n",
      "\n",
      "Beta-oxidation Of Very Long Chain Fatty Acids R-HSA-390247\tFatty Acid Beta-Oxidation (GO:0006635)\n",
      "\n",
      "Bicarbonate Transporters R-HSA-425381\tBicarbonate Transport (GO:0015701)\n",
      "\n",
      "Bile Acid And Bile Salt Metabolism R-HSA-194068\tBile Acid And Bile Salt Transport (GO:0015721)\n",
      "\n",
      "Branched-chain Amino Acid Catabolism R-HSA-70895\tBranched-Chain Amino Acid Catabolic Process (GO:0009083)\n",
      "\n",
      "CDC42 GTPase Cycle R-HSA-9013148\tCdc42 Protein Signal Transduction (GO:0032488)\n",
      "\n",
      "CLEC7A (Dectin-1) Signaling R-HSA-5607764\tStimulatory C-type Lectin Receptor Signaling Pathway (GO:0002223)\n",
      "\n",
      "COPI-dependent Golgi-to-ER Retrograde Traffic R-HSA-6811434\tRetrograde Vesicle-Mediated Transport, Golgi To Endoplasmic Reticulum (GO:0006890)\n",
      "\n",
      "COPI-independent Golgi-to-ER Retrograde Traffic R-HSA-6811436\tRetrograde Vesicle-Mediated Transport, Golgi To Endoplasmic Reticulum (GO:0006890)\n",
      "\n",
      "COPI-mediated Anterograde Transport R-HSA-6807878\tIntraciliary Anterograde Transport (GO:0035720)\n",
      "\n",
      "COPII-mediated Vesicle Transport R-HSA-204005\tCOPII-coated Vesicle Cargo Loading (GO:0090110)\n",
      "\n",
      "CREB1 Phosphorylation Thru Activation Of Adenylate Cyclase R-HSA-442720\tActivation Of Adenylate Cyclase Activity (GO:0007190)\n",
      "\n",
      "Calcineurin Activates NFAT R-HSA-2025928\tcalcineurin-NFAT Signaling Cascade (GO:0033173)\n",
      "\n",
      "Calcitonin-like Ligand Receptors R-HSA-419812\tCalcitonin Family Receptor Signaling Pathway (GO:0097646)\n",
      "\n",
      "Cap-dependent Translation Initiation R-HSA-72737\tCap-Dependent Translational Initiation (GO:0002191)\n",
      "\n",
      "Cardiac Conduction R-HSA-5576891\tCardiac Conduction (GO:0061337)\n",
      "\n",
      "Cargo Recognition For Clathrin-Mediated Endocytosis R-HSA-8856825\tClathrin-Coated Vesicle Cargo Loading (GO:0035652)\n",
      "\n",
      "Carnitine Metabolism R-HSA-200425\tCarnitine Metabolic Process (GO:0009437)\n",
      "\n",
      "Caspase Activation Via Dependence Receptors In Absence Of Ligand R-HSA-418889\tExtrinsic Apoptotic Signaling Pathway In Absence Of Ligand (GO:0097192)\n",
      "\n",
      "Caspase Activation Via Extrinsic Apoptotic Signaling Pathway R-HSA-5357769\tExtrinsic Apoptotic Signaling Pathway (GO:0097191)\n",
      "\n",
      "Cell Cycle Checkpoints R-HSA-69620\tMitotic Cell Cycle Checkpoint Signaling (GO:0007093)\n",
      "\n",
      "Cell Junction Organization R-HSA-446728\tCell Junction Organization (GO:0034330)\n",
      "\n",
      "Cell-Cell Communication R-HSA-1500931\tRegulation Of Cell Communication (GO:0010646)\n",
      "\n",
      "Cell-cell Junction Organization R-HSA-421270\tCell-Cell Junction Organization (GO:0045216)\n",
      "\n",
      "Cell-extracellular Matrix Interactions R-HSA-446353\tCell-Matrix Adhesion (GO:0007160)\n",
      "\n",
      "Cellular Senescence R-HSA-2559583\tRegulation Of Cellular Senescence (GO:2000772)\n",
      "\n",
      "Cellular Hexose Transport R-HSA-189200\tHexose Transmembrane Transport (GO:0008645)\n",
      "\n",
      "Cellular Response To Chemical Stress R-HSA-9711123\tCellular Response To Chemical Stress (GO:0062197)\n",
      "\n",
      "Cellular Response To Heat Stress R-HSA-3371556\tCellular Response To Heat (GO:0034605)\n",
      "\n",
      "Cellular Response To Hypoxia R-HSA-1234174\tCellular Response To Hypoxia (GO:0071456)\n",
      "\n",
      "Cellular Response To Starvation R-HSA-9711097\tCellular Response To Starvation (GO:0009267)\n",
      "\n",
      "Cellular Responses To Stimuli R-HSA-8953897\tCellular Response To Hormone Stimulus (GO:0032870)\n",
      "\n",
      "Cellular Responses To Stress R-HSA-2262752\tRegulation Of Cellular Response To Stress (GO:0080135)\n",
      "\n",
      "Chaperone Mediated Autophagy R-HSA-9613829\tRegulation Of Chaperone-Mediated Autophagy (GO:1904714)\n",
      "\n",
      "Chaperonin-mediated Protein Folding R-HSA-390466\tChaperone Cofactor-Dependent Protein Refolding (GO:0051085)\n",
      "\n",
      "Cholesterol Biosynthesis R-HSA-191273\tCholesterol Biosynthetic Process (GO:0006695)\n",
      "\n",
      "Choline Catabolism R-HSA-6798163\tCholine Metabolic Process (GO:0019695)\n",
      "\n",
      "Chondroitin Sulfate Biosynthesis R-HSA-2022870\tChondroitin Sulfate Biosynthetic Process (GO:0030206)\n",
      "\n",
      "Chondroitin Sulfate/Dermatan Sulfate Metabolism R-HSA-1793185\tDermatan Sulfate Metabolic Process (GO:0030205)\n",
      "\n",
      "Chromosome Maintenance R-HSA-73886\tTelomere Maintenance (GO:0000723)\n",
      "\n",
      "Chylomicron Assembly R-HSA-8963888\tHigh-Density Lipoprotein Particle Assembly (GO:0034380)\n",
      "\n",
      "Chylomicron Remodeling R-HSA-8963901\tTriglyceride-Rich Lipoprotein Particle Remodeling (GO:0034370)\n",
      "\n",
      "Cilium Assembly R-HSA-5617833\tCilium Assembly (GO:0060271)\n",
      "\n",
      "Class I MHC Mediated Antigen Processing And Presentation R-HSA-983169\tAntigen Processing And Presentation Of Peptide Antigen Via MHC Class I (GO:0002474)\n",
      "\n",
      "Class I Peroxisomal Membrane Protein Import R-HSA-9603798\tPeroxisomal Membrane Transport (GO:0015919)\n",
      "\n",
      "Classical Antibody-Mediated Complement Activation R-HSA-173623\tRegulation Of Complement Activation, Classical Pathway (GO:0030450)\n",
      "\n",
      "Clathrin-mediated Endocytosis R-HSA-8856828\tClathrin-Dependent Endocytosis (GO:0072583)\n",
      "\n",
      "Cobalamin (Cbl) Metabolism R-HSA-9759218\tCobalamin Metabolic Process (GO:0009235)\n",
      "\n",
      "Cobalamin (Cbl, Vitamin B12) Transport And Metabolism R-HSA-196741\tCobalamin Transport (GO:0015889)\n",
      "\n",
      "Coenzyme A Biosynthesis R-HSA-196783\tCoenzyme A Biosynthetic Process (GO:0015937)\n",
      "\n",
      "Collagen Biosynthesis And Modifying Enzymes R-HSA-1650814\tRegulation Of Collagen Biosynthetic Process (GO:0032965)\n",
      "\n",
      "Collagen Chain Trimerization R-HSA-8948216\tProtein Trimerization (GO:0070206)\n",
      "\n",
      "Condensation Of Prometaphase Chromosomes R-HSA-2514853\tChromosome Condensation (GO:0030261)\n",
      "\n",
      "Condensation Of Prophase Chromosomes R-HSA-2299718\tChromosome Condensation (GO:0030261)\n",
      "\n",
      "Constitutive Signaling By Overexpressed ERBB2 R-HSA-9634285\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "Cristae Formation R-HSA-8949613\tCristae Formation (GO:0042407)\n",
      "\n",
      "Cross-presentation Of Particulate Exogenous Antigens (Phagosomes) R-HSA-1236973\tAntigen Processing And Presentation Of Exogenous Peptide Antigen (GO:0002478)\n",
      "\n",
      "Cross-presentation Of Soluble Exogenous Antigens (Endosomes) R-HSA-1236978\tAntigen Processing And Presentation Of Exogenous Peptide Antigen (GO:0002478)\n",
      "\n",
      "Crosslinking Of Collagen Fibrils R-HSA-2243919\tCollagen Fibril Organization (GO:0030199)\n",
      "\n",
      "Cyclin A/B1/B2 Associated Events During G2/M Transition R-HSA-69273\tCell Cycle G2/M Phase Transition (GO:0044839)\n",
      "\n",
      "Cytokine Signaling In Immune System R-HSA-1280215\tCytokine-Mediated Signaling Pathway (GO:0019221)\n",
      "\n",
      "Cytosolic Iron-Sulfur Cluster Assembly R-HSA-2564830\tIron-Sulfur Cluster Assembly (GO:0016226)\n",
      "\n",
      "Cytosolic tRNA Aminoacylation R-HSA-379716\ttRNA Aminoacylation (GO:0043039)\n",
      "\n",
      "DNA Damage Bypass R-HSA-73893\tDNA Damage Response (GO:0006974)\n",
      "\n",
      "DNA Damage Reversal R-HSA-73942\tDNA Damage Response (GO:0006974)\n",
      "\n",
      "DNA Double Strand Break Response R-HSA-5693606\tDNA Double-Strand Break Processing (GO:0000729)\n",
      "\n",
      "DNA Double-Strand Break Repair R-HSA-5693532\tDouble-Strand Break Repair (GO:0006302)\n",
      "\n",
      "DNA Repair R-HSA-73894\tDNA Repair (GO:0006281)\n",
      "\n",
      "DNA Replication Pre-Initiation R-HSA-69002\tDNA Replication Initiation (GO:0006270)\n",
      "\n",
      "DNA Replication R-HSA-69306\tDNA Replication (GO:0006260)\n",
      "\n",
      "DNA Methylation R-HSA-5334118\tDNA Methylation (GO:0006306)\n",
      "\n",
      "DNA Replication Initiation R-HSA-68952\tDNA Replication Initiation (GO:0006270)\n",
      "\n",
      "DNA Strand Elongation R-HSA-69190\tDNA Strand Elongation (GO:0022616)\n",
      "\n",
      "Deadenylation Of mRNA R-HSA-429947\tDeadenylation-Dependent Decapping Of Nuclear-Transcribed mRNA (GO:0000290)\n",
      "\n",
      "Deadenylation-dependent mRNA Decay R-HSA-429914\tDeadenylation-Dependent Decapping Of Nuclear-Transcribed mRNA (GO:0000290)\n",
      "\n",
      "Death Receptor Signaling R-HSA-73887\tExtrinsic Apoptotic Signaling Pathway Via Death Domain Receptors (GO:0008625)\n",
      "\n",
      "Defective Pyroptosis R-HSA-9710421\tPyroptosis (GO:0070269)\n",
      "\n",
      "Degradation Of Extracellular Matrix R-HSA-1474228\tRegulation Of Extracellular Matrix Disassembly (GO:0010715)\n",
      "\n",
      "Deposition Of New CENPA-containing Nucleosomes At Centromere R-HSA-606279\tCENP-A Containing Chromatin Assembly (GO:0034080)\n",
      "\n",
      "Dermatan Sulfate Biosynthesis R-HSA-2022923\tDermatan Sulfate Proteoglycan Biosynthetic Process (GO:0050651)\n",
      "\n",
      "Detoxification Of Reactive Oxygen Species R-HSA-3299685\tResponse To Reactive Oxygen Species (GO:0000302)\n",
      "\n",
      "Deubiquitination R-HSA-5688426\tProtein Deubiquitination (GO:0016579)\n",
      "\n",
      "Diseases Of Base Excision Repair R-HSA-9605308\tBase-Excision Repair (GO:0006284)\n",
      "\n",
      "Diseases Of DNA Repair R-HSA-9675135\tRegulation Of DNA Repair (GO:0006282)\n",
      "\n",
      "Diseases Of Mitotic Cell Cycle R-HSA-9675126\tRegulation Of Mitotic Cell Cycle (GO:0007346)\n",
      "\n",
      "Diseases Of Programmed Cell Death R-HSA-9645723\tRegulation Of Programmed Cell Death (GO:0043067)\n",
      "\n",
      "Disinhibition Of SNARE Formation R-HSA-114516\tRegulation Of SNARE Complex Assembly (GO:0035542)\n",
      "\n",
      "Dissolution Of Fibrin Clot R-HSA-75205\tFibrinolysis (GO:0042730)\n",
      "\n",
      "Dopamine Receptors R-HSA-390651\tDopamine Transport (GO:0015872)\n",
      "\n",
      "Downregulation Of ERBB2 Signaling R-HSA-8863795\tNegative Regulation Of ERBB Signaling Pathway (GO:1901185)\n",
      "\n",
      "Downregulation Of ERBB2:ERBB3 Signaling R-HSA-1358803\tNegative Regulation Of ERBB Signaling Pathway (GO:1901185)\n",
      "\n",
      "Downregulation Of ERBB4 Signaling R-HSA-1253288\tNegative Regulation Of ERBB Signaling Pathway (GO:1901185)\n",
      "\n",
      "Downregulation Of SMAD2/3:SMAD4 Transcriptional Activity R-HSA-2173795\tNegative Regulation Of SMAD Protein Signal Transduction (GO:0060392)\n",
      "\n",
      "Downregulation Of TGF-beta Receptor Signaling R-HSA-2173788\tNegative Regulation Of Transforming Growth Factor Beta Receptor Signaling Pathway (GO:0030512)\n",
      "\n",
      "Downstream Signaling Events Of B Cell Receptor (BCR) R-HSA-1168372\tB Cell Receptor Signaling Pathway (GO:0050853)\n",
      "\n",
      "E2F Mediated Regulation Of DNA Replication R-HSA-113510\tPositive Regulation Of DNA Replication (GO:0045740)\n",
      "\n",
      "EPH-Ephrin Signaling R-HSA-2682334\tEphrin Receptor Signaling Pathway (GO:0048013)\n",
      "\n",
      "ER To Golgi Anterograde Transport R-HSA-199977\tRetrograde Transport, Endosome To Golgi (GO:0042147)\n",
      "\n",
      "ERBB2 Activates PTK6 Signaling R-HSA-8847993\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "ERBB2 Regulates Cell Motility R-HSA-6785631\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "Elastic Fibre Formation R-HSA-1566948\tElastic Fiber Assembly (GO:0048251)\n",
      "\n",
      "Endosomal/Vacuolar Pathway R-HSA-1236977\tLate Endosome To Vacuole Transport Via Multivesicular Body Sorting Pathway (GO:0032511)\n",
      "\n",
      "Entry Of Influenza Virion Into Host Cell Via Endocytosis R-HSA-168275\tReceptor-Mediated Endocytosis Of Virus By Host Cell (GO:0019065)\n",
      "\n",
      "Ephrin Signaling R-HSA-3928664\tEphrin Receptor Signaling Pathway (GO:0048013)\n",
      "\n",
      "Epigenetic Regulation Of Gene Expression R-HSA-212165\tEpigenetic Regulation Of Gene Expression (GO:0040029)\n",
      "\n",
      "Establishment Of Sister Chromatid Cohesion R-HSA-2468052\tEstablishment Of Mitotic Sister Chromatid Cohesion (GO:0034087)\n",
      "\n",
      "Estrogen Biosynthesis R-HSA-193144\tEstrogen Biosynthetic Process (GO:0006703)\n",
      "\n",
      "Ethanol Oxidation R-HSA-71384\tEthanol Oxidation (GO:0006069)\n",
      "\n",
      "Eukaryotic Translation Elongation R-HSA-156842\tTranslational Elongation (GO:0006414)\n",
      "\n",
      "Eukaryotic Translation Termination R-HSA-72764\tTranslational Termination (GO:0006415)\n",
      "\n",
      "Export Of Viral Ribonucleoproteins From Nucleus R-HSA-168274\tViral mRNA Export From Host Cell Nucleus (GO:0046784)\n",
      "\n",
      "Extension Of Telomeres R-HSA-180786\tTelomere Maintenance Via Telomere Lengthening (GO:0010833)\n",
      "\n",
      "Extra-nuclear Estrogen Signaling R-HSA-9009391\tIntracellular Estrogen Receptor Signaling Pathway (GO:0030520)\n",
      "\n",
      "Extracellular Matrix Organization R-HSA-1474244\tExtracellular Matrix Organization (GO:0030198)\n",
      "\n",
      "FCGR3A-mediated Phagocytosis R-HSA-9664422\tFc-gamma Receptor Signaling Pathway Involved In Phagocytosis (GO:0038096)\n",
      "\n",
      "FRS-mediated FGFR1 Signaling R-HSA-5654693\tFibroblast Growth Factor Receptor Signaling Pathway (GO:0008543)\n",
      "\n",
      "Factors Involved In Megakaryocyte Development And Platelet Production R-HSA-983231\tMegakaryocyte Development (GO:0035855)\n",
      "\n",
      "Fatty Acid Metabolism R-HSA-8978868\tFatty Acid Metabolic Process (GO:0006631)\n",
      "\n",
      "Fatty acyl-CoA Biosynthesis R-HSA-75105\tfatty-acyl-CoA Biosynthetic Process (GO:0046949)\n",
      "\n",
      "Fc Epsilon Receptor (FCERI) Signaling R-HSA-2454202\tFc-epsilon Receptor Signaling Pathway (GO:0038095)\n",
      "\n",
      "Fcgamma Receptor (FCGR) Dependent Phagocytosis R-HSA-2029480\tFc-gamma Receptor Signaling Pathway Involved In Phagocytosis (GO:0038096)\n",
      "\n",
      "Formation Of Fibrin Clot (Clotting Cascade) R-HSA-140877\tBlood Coagulation, Fibrin Clot Formation (GO:0072378)\n",
      "\n",
      "Formation Of RNA Pol II Elongation Complex R-HSA-112382\tTranscription Elongation By RNA Polymerase II (GO:0006368)\n",
      "\n",
      "Fructose Catabolism R-HSA-70350\tFructose Metabolic Process (GO:0006000)\n",
      "\n",
      "Fructose Metabolism R-HSA-5652084\tFructose Metabolic Process (GO:0006000)\n",
      "\n",
      "G0 And Early G1 R-HSA-1538133\tRegulation Of G0 To G1 Transition (GO:0070316)\n",
      "\n",
      "G1/S DNA Damage Checkpoints R-HSA-69615\tMitotic G1 DNA Damage Checkpoint Signaling (GO:0031571)\n",
      "\n",
      "G1/S Transition R-HSA-69206\tG1/S Transition Of Mitotic Cell Cycle (GO:0000082)\n",
      "\n",
      "G1/S-Specific Transcription R-HSA-69205\tRegulation Of Transcription Involved In G1/S Transition Of Mitotic Cell Cycle (GO:0000083)\n",
      "\n",
      "G2 Phase R-HSA-68911\tCell Cycle G2/M Phase Transition (GO:0044839)\n",
      "\n",
      "G2/M Checkpoints R-HSA-69481\tMitotic G2/M Transition Checkpoint (GO:0044818)\n",
      "\n",
      "G2/M DNA Damage Checkpoint R-HSA-69473\tMitotic G2 DNA Damage Checkpoint Signaling (GO:0007095)\n",
      "\n",
      "G2/M DNA Replication Checkpoint R-HSA-69478\tMitotic G2/M Transition Checkpoint (GO:0044818)\n",
      "\n",
      "G2/M Transition R-HSA-69275\tCell Cycle G2/M Phase Transition (GO:0044839)\n",
      "\n",
      "GDP-fucose Biosynthesis R-HSA-6787639\tGDP-mannose Metabolic Process (GO:0019673)\n",
      "\n",
      "GRB2 Events In ERBB2 Signaling R-HSA-1963640\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "GRB7 Events In ERBB2 Signaling R-HSA-1306955\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "Gap Junction Assembly R-HSA-190861\tGap Junction Assembly (GO:0016264)\n",
      "\n",
      "Gap Junction Degradation R-HSA-190873\tGap Junction Assembly (GO:0016264)\n",
      "\n",
      "Gap Junction Trafficking R-HSA-190828\tGap Junction Assembly (GO:0016264)\n",
      "\n",
      "Gap-filling DNA Repair Synthesis And Ligation In GG-NER R-HSA-5696397\tNucleotide-Excision Repair, DNA Gap Filling (GO:0006297)\n",
      "\n",
      "Gap-filling DNA Repair Synthesis And Ligation In TC-NER R-HSA-6782210\tNucleotide-Excision Repair, DNA Gap Filling (GO:0006297)\n",
      "\n",
      "Gastrulation R-HSA-9758941\tGastrulation (GO:0007369)\n",
      "\n",
      "Gene Silencing By RNA R-HSA-211000\tRNA-mediated Gene Silencing (GO:0031047)\n",
      "\n",
      "Gene Expression (Transcription) R-HSA-74160\tGene Expression (GO:0010467)\n",
      "\n",
      "Glucagon Signaling In Metabolic Regulation R-HSA-163359\tRegulation Of Glucagon Secretion (GO:0070092)\n",
      "\n",
      "Glucocorticoid Biosynthesis R-HSA-194002\tGlucocorticoid Biosynthetic Process (GO:0006704)\n",
      "\n",
      "Gluconeogenesis R-HSA-70263\tGluconeogenesis (GO:0006094)\n",
      "\n",
      "Glucose Metabolism R-HSA-70326\tGlucose Metabolic Process (GO:0006006)\n",
      "\n",
      "Glucuronidation R-HSA-156588\tCellular Glucuronidation (GO:0052695)\n",
      "\n",
      "Glutamate And Glutamine Metabolism R-HSA-8964539\tGlutamate Metabolic Process (GO:0006536)\n",
      "\n",
      "Glutathione Synthesis And Recycling R-HSA-174403\tGlutathione Biosynthetic Process (GO:0006750)\n",
      "\n",
      "Glycerophospholipid Biosynthesis R-HSA-1483206\tGlycerophospholipid Biosynthetic Process (GO:0046474)\n",
      "\n",
      "Glycerophospholipid Catabolism R-HSA-6814848\tGlycerophospholipid Catabolic Process (GO:0046475)\n",
      "\n",
      "Glycogen Metabolism R-HSA-8982491\tGlycogen Metabolic Process (GO:0005977)\n",
      "\n",
      "Glycogen Synthesis R-HSA-3322077\tGlycogen Biosynthetic Process (GO:0005978)\n",
      "\n",
      "Glycolysis R-HSA-70171\tGlycolytic Process (GO:0006096)\n",
      "\n",
      "Glycosaminoglycan Metabolism R-HSA-1630316\tGlycosaminoglycan Metabolic Process (GO:0030203)\n",
      "\n",
      "Glycosphingolipid Metabolism R-HSA-1660662\tGlycosphingolipid Metabolic Process (GO:0006687)\n",
      "\n",
      "Glyoxylate Metabolism And Glycine Degradation R-HSA-389661\tGlyoxylate Metabolic Process (GO:0046487)\n",
      "\n",
      "Golgi Associated Vesicle Biogenesis R-HSA-432722\tGolgi Vesicle Transport (GO:0048193)\n",
      "\n",
      "Golgi-to-ER Retrograde Transport R-HSA-8856688\tRetrograde Vesicle-Mediated Transport, Golgi To Endoplasmic Reticulum (GO:0006890)\n",
      "\n",
      "Growth Hormone Receptor Signaling R-HSA-982772\tGrowth Hormone Receptor Signaling Pathway (GO:0060396)\n",
      "\n",
      "HDACs Deacetylate Histones R-HSA-3214815\tHistone Deacetylation (GO:0016575)\n",
      "\n",
      "HDMs Demethylate Histones R-HSA-3214842\tHistone Lysine Demethylation (GO:0070076)\n",
      "\n",
      "HDR Thru Homologous Recombination (HRR) R-HSA-5685942\tHomologous Recombination (GO:0035825)\n",
      "\n",
      "HDR Thru Homologous Recombination (HRR) Or Single Strand Annealing (SSA) R-HSA-5693567\tDouble-Strand Break Repair Via Single-Strand Annealing (GO:0045002)\n",
      "\n",
      "HS-GAG Biosynthesis R-HSA-2022928\tGlycosaminoglycan Biosynthetic Process (GO:0006024)\n",
      "\n",
      "Heme Biosynthesis R-HSA-189451\tHeme Biosynthetic Process (GO:0006783)\n",
      "\n",
      "Heme Degradation R-HSA-189483\tHeme Catabolic Process (GO:0042167)\n",
      "\n",
      "Hemostasis R-HSA-109582\tHemostasis (GO:0007599)\n",
      "\n",
      "Heparan Sulfate/Heparin (HS-GAG) Metabolism R-HSA-1638091\tHeparan Sulfate Proteoglycan Metabolic Process (GO:0030201)\n",
      "\n",
      "Histidine Catabolism R-HSA-70921\tHistidine Catabolic Process (GO:0006548)\n",
      "\n",
      "Homology Directed Repair R-HSA-5693538\tHomologous Recombination (GO:0035825)\n",
      "\n",
      "Hyaluronan Metabolism R-HSA-2142845\tHyaluronan Metabolic Process (GO:0030212)\n",
      "\n",
      "Hyaluronan Uptake And Degradation R-HSA-2160916\tHyaluronan Catabolic Process (GO:0030214)\n",
      "\n",
      "IGF1R Signaling Cascade R-HSA-2428924\tInsulin-Like Growth Factor Receptor Signaling Pathway (GO:0048009)\n",
      "\n",
      "IRE1alpha Activates Chaperones R-HSA-381070\tIRE1-mediated Unfolded Protein Response (GO:0036498)\n",
      "\n",
      "IRF3 Mediated Activation Of Type 1 IFN R-HSA-1606341\tPositive Regulation Of Type I Interferon Production (GO:0032481)\n",
      "\n",
      "IRF3-mediated Induction Of Type I IFN R-HSA-3270619\tPositive Regulation Of Type I Interferon Production (GO:0032481)\n",
      "\n",
      "ISG15 Antiviral Mechanism R-HSA-1169408\tISG15-protein Conjugation (GO:0032020)\n",
      "\n",
      "Inhibition Of DNA Recombination At Telomere R-HSA-9670095\tNegative Regulation Of DNA Recombination (GO:0045910)\n",
      "\n",
      "Initiation Of Nuclear Envelope (NE) Reformation R-HSA-2995383\tNuclear Membrane Reassembly (GO:0031468)\n",
      "\n",
      "Inositol Phosphate Metabolism R-HSA-1483249\tInositol Phosphate Metabolic Process (GO:0043647)\n",
      "\n",
      "Insertion Of Tail-Anchored Proteins Into Endoplasmic Reticulum Membrane R-HSA-9609523\tTail-Anchored Membrane Protein Insertion Into ER Membrane (GO:0071816)\n",
      "\n",
      "Insulin Processing R-HSA-264876\tInsulin Processing (GO:0030070)\n",
      "\n",
      "Insulin Receptor Recycling R-HSA-77387\tReceptor Recycling (GO:0001881)\n",
      "\n",
      "Insulin Receptor Signaling Cascade R-HSA-74751\tInsulin Receptor Signaling Pathway (GO:0008286)\n",
      "\n",
      "Integrin Signaling R-HSA-354192\tIntegrin-Mediated Signaling Pathway (GO:0007229)\n",
      "\n",
      "Interferon Signaling R-HSA-913531\tInterferon-Mediated Signaling Pathway (GO:0140888)\n",
      "\n",
      "Interferon Alpha/Beta Signaling R-HSA-909733\tInterferon-Mediated Signaling Pathway (GO:0140888)\n",
      "\n",
      "Interferon Gamma Signaling R-HSA-877300\tInterferon-Mediated Signaling Pathway (GO:0140888)\n",
      "\n",
      "Interleukin-1 Family Signaling R-HSA-446652\tInterleukin-1-Mediated Signaling Pathway (GO:0070498)\n",
      "\n",
      "Interleukin-1 Signaling R-HSA-9020702\tInterleukin-1-Mediated Signaling Pathway (GO:0070498)\n",
      "\n",
      "Interleukin-10 Signaling R-HSA-6783783\tInterleukin-9-Mediated Signaling Pathway (GO:0038113)\n",
      "\n",
      "Interleukin-12 Family Signaling R-HSA-447115\tInterleukin-12-Mediated Signaling Pathway (GO:0035722)\n",
      "\n",
      "Interleukin-12 Signaling R-HSA-9020591\tInterleukin-12-Mediated Signaling Pathway (GO:0035722)\n",
      "\n",
      "Interleukin-15 Signaling R-HSA-8983432\tInterleukin-15-Mediated Signaling Pathway (GO:0035723)\n",
      "\n",
      "Interleukin-17 Signaling R-HSA-448424\tinterleukin-17A-mediated Signaling Pathway (GO:0038173)\n",
      "\n",
      "Interleukin-18 Signaling R-HSA-9012546\tInterleukin-12-Mediated Signaling Pathway (GO:0035722)\n",
      "\n",
      "Interleukin-2 Family Signaling R-HSA-451927\tInterleukin-2-Mediated Signaling Pathway (GO:0038110)\n",
      "\n",
      "Interleukin-2 Signaling R-HSA-9020558\tInterleukin-2-Mediated Signaling Pathway (GO:0038110)\n",
      "\n",
      "Interleukin-20 Family Signaling R-HSA-8854691\tInterleukin-27-Mediated Signaling Pathway (GO:0070106)\n",
      "\n",
      "Interleukin-21 Signaling R-HSA-9020958\tInterleukin-15-Mediated Signaling Pathway (GO:0035723)\n",
      "\n",
      "Interleukin-23 Signaling R-HSA-9020933\tInterleukin-27-Mediated Signaling Pathway (GO:0070106)\n",
      "\n",
      "Interleukin-27 Signaling R-HSA-9020956\tInterleukin-27-Mediated Signaling Pathway (GO:0070106)\n",
      "\n",
      "Interleukin-3, Interleukin-5 And GM-CSF Signaling R-HSA-512988\tInterleukin-3-Mediated Signaling Pathway (GO:0038156)\n",
      "\n",
      "Interleukin-35 Signaling R-HSA-8984722\tInterleukin-27-Mediated Signaling Pathway (GO:0070106)\n",
      "\n",
      "Interleukin-37 Signaling R-HSA-9008059\tInterleukin-27-Mediated Signaling Pathway (GO:0070106)\n",
      "\n",
      "Interleukin-4 And Interleukin-13 Signaling R-HSA-6785807\tInterleukin-4-Mediated Signaling Pathway (GO:0035771)\n",
      "\n",
      "Interleukin-6 Family Signaling R-HSA-6783589\tInterleukin-6-Mediated Signaling Pathway (GO:0070102)\n",
      "\n",
      "Interleukin-6 Signaling R-HSA-1059683\tInterleukin-6-Mediated Signaling Pathway (GO:0070102)\n",
      "\n",
      "Interleukin-7 Signaling R-HSA-1266695\tInterleukin-2-Mediated Signaling Pathway (GO:0038110)\n",
      "\n",
      "Interleukin-9 Signaling R-HSA-8985947\tInterleukin-9-Mediated Signaling Pathway (GO:0038113)\n",
      "\n",
      "Intestinal Absorption R-HSA-8963676\tIntestinal Absorption (GO:0050892)\n",
      "\n",
      "Intra-Golgi And Retrograde Golgi-to-ER Traffic R-HSA-6811442\tRetrograde Transport, Vesicle Recycling Within Golgi (GO:0000301)\n",
      "\n",
      "Intra-Golgi Traffic R-HSA-6811438\tintra-Golgi Vesicle-Mediated Transport (GO:0006891)\n",
      "\n",
      "Intracellular Signaling By Second Messengers R-HSA-9006925\tSecond-Messenger-Mediated Signaling (GO:0019932)\n",
      "\n",
      "Intrinsic Pathway For Apoptosis R-HSA-109606\tIntrinsic Apoptotic Signaling Pathway (GO:0097193)\n",
      "\n",
      "Intrinsic Pathway Of Fibrin Clot Formation R-HSA-140837\tBlood Coagulation, Intrinsic Pathway (GO:0007597)\n",
      "\n",
      "Ion Channel Transport R-HSA-983712\tCalcium Ion Transport (GO:0006816)\n",
      "\n",
      "Iron Uptake And Transport R-HSA-917937\tIron Ion Transport (GO:0006826)\n",
      "\n",
      "Keratan Sulfate Biosynthesis R-HSA-2022854\tKeratan Sulfate Biosynthetic Process (GO:0018146)\n",
      "\n",
      "Keratan Sulfate Degradation R-HSA-2022857\tKeratan Sulfate Metabolic Process (GO:0042339)\n",
      "\n",
      "Keratan Sulfate/Keratin Metabolism R-HSA-1638074\tKeratan Sulfate Metabolic Process (GO:0042339)\n",
      "\n",
      "Ketone Body Metabolism R-HSA-74182\tKetone Body Metabolic Process (GO:1902224)\n",
      "\n",
      "LDL Clearance R-HSA-8964038\tRegulation Of Low-Density Lipoprotein Particle Clearance (GO:0010988)\n",
      "\n",
      "LDL Remodeling R-HSA-8964041\tLow-Density Lipoprotein Particle Remodeling (GO:0034374)\n",
      "\n",
      "Late Endosomal Microautophagy R-HSA-9615710\tLysosomal Microautophagy (GO:0016237)\n",
      "\n",
      "Lectin Pathway Of Complement Activation R-HSA-166662\tComplement Activation, Lectin Pathway (GO:0001867)\n",
      "\n",
      "Leukotriene Receptors R-HSA-391906\tLeukotriene Transport (GO:0071716)\n",
      "\n",
      "Linoleic Acid (LA) Metabolism R-HSA-2046105\tLinoleic Acid Metabolic Process (GO:0043651)\n",
      "\n",
      "Lipid Particle Organization R-HSA-8964572\tLipid Droplet Organization (GO:0034389)\n",
      "\n",
      "Long-term Potentiation R-HSA-9620244\tLong-Term Synaptic Potentiation (GO:0060291)\n",
      "\n",
      "MAP Kinase Activation R-HSA-450294\tPositive Regulation Of MAP Kinase Activity (GO:0043406)\n",
      "\n",
      "MAP2K And MAPK Activation R-HSA-5674135\tPositive Regulation Of MAP Kinase Activity (GO:0043406)\n",
      "\n",
      "MAPK Family Signaling Cascades R-HSA-5683057\tMAPK Cascade (GO:0000165)\n",
      "\n",
      "MAPK1 (ERK2) Activation R-HSA-112411\tPositive Regulation Of ERK1 And ERK2 Cascade (GO:0070374)\n",
      "\n",
      "MAPK1/MAPK3 Signaling R-HSA-5684996\tMAPK Cascade (GO:0000165)\n",
      "\n",
      "MAPK3 (ERK1) Activation R-HSA-110056\tMAPK Cascade (GO:0000165)\n",
      "\n",
      "MET Promotes Cell Motility R-HSA-8875878\tPositive Regulation Of Cell Motility (GO:2000147)\n",
      "\n",
      "MET Receptor Recycling R-HSA-8875656\tReceptor Recycling (GO:0001881)\n",
      "\n",
      "MHC Class II Antigen Presentation R-HSA-2132295\tAntigen Processing And Presentation Of Peptide Antigen Via MHC Class II (GO:0002495)\n",
      "\n",
      "MTOR Signaling R-HSA-165159\tTORC1 Signaling (GO:0038202)\n",
      "\n",
      "Macroautophagy R-HSA-1632852\tMacroautophagy (GO:0016236)\n",
      "\n",
      "Major Pathway Of rRNA Processing In Nucleolus And Cytosol R-HSA-6791226\tRegulation Of rRNA Processing (GO:2000232)\n",
      "\n",
      "Meiosis R-HSA-1500620\tMeiosis I (GO:0007127)\n",
      "\n",
      "Meiotic Recombination R-HSA-912446\tMitotic Recombination (GO:0006312)\n",
      "\n",
      "Melanin Biosynthesis R-HSA-5662702\tMelanin Biosynthetic Process (GO:0042438)\n",
      "\n",
      "Membrane Trafficking R-HSA-199991\tVesicle Cytoskeletal Trafficking (GO:0099518)\n",
      "\n",
      "Metabolism Of RNA R-HSA-8953854\tRegulation Of RNA Metabolic Process (GO:0051252)\n",
      "\n",
      "Metabolism Of Fat-Soluble Vitamins R-HSA-6806667\tFat-Soluble Vitamin Metabolic Process (GO:0006775)\n",
      "\n",
      "Metabolism Of Nitric Oxide: NOS3 Activation And Regulation R-HSA-202131\tRegulation Of Nitric Oxide Metabolic Process (GO:0080164)\n",
      "\n",
      "Metabolism Of Non-Coding RNA R-HSA-194441\tncRNA Metabolic Process (GO:0034660)\n",
      "\n",
      "Metabolism Of Nucleotides R-HSA-15869\tNucleotide Metabolic Process (GO:0009117)\n",
      "\n",
      "Metabolism Of Polyamines R-HSA-351202\tPolyamine Metabolic Process (GO:0006595)\n",
      "\n",
      "Metabolism Of Steroid Hormones R-HSA-196071\tSteroid Metabolic Process (GO:0008202)\n",
      "\n",
      "Metabolism Of Steroids R-HSA-8957322\tSteroid Metabolic Process (GO:0008202)\n",
      "\n",
      "Metabolism Of Water-Soluble Vitamins And Cofactors R-HSA-196849\tWater-Soluble Vitamin Metabolic Process (GO:0006767)\n",
      "\n",
      "Metal Ion SLC Transporters R-HSA-425410\tMetal Ion Transport (GO:0030001)\n",
      "\n",
      "Methionine Salvage Pathway R-HSA-1237112\tL-methionine Salvage (GO:0071267)\n",
      "\n",
      "Methylation R-HSA-156581\tDNA Methylation (GO:0006306)\n",
      "\n",
      "MicroRNA (miRNA) Biogenesis R-HSA-203927\tPrimary miRNA Processing (GO:0031053)\n",
      "\n",
      "Mineralocorticoid Biosynthesis R-HSA-193993\tRegulation Of Aldosterone Biosynthetic Process (GO:0032347)\n",
      "\n",
      "Mismatch Repair R-HSA-5358508\tMismatch Repair (GO:0006298)\n",
      "\n",
      "Mismatch Repair (MMR) Directed By MSH2:MSH3 (MutSbeta) R-HSA-5358606\tMismatch Repair (GO:0006298)\n",
      "\n",
      "Mitochondrial Fatty Acid Beta-Oxidation R-HSA-77289\tFatty Acid Beta-Oxidation (GO:0006635)\n",
      "\n",
      "Mitochondrial Calcium Ion Transport R-HSA-8949215\tMitochondrial Calcium Ion Transmembrane Transport (GO:0006851)\n",
      "\n",
      "Mitochondrial Iron-Sulfur Cluster Biogenesis R-HSA-1362409\tIron-Sulfur Cluster Assembly (GO:0016226)\n",
      "\n",
      "Mitochondrial Protein Import R-HSA-1268020\tMitochondrial Protein Processing (GO:0034982)\n",
      "\n",
      "Mitochondrial tRNA Aminoacylation R-HSA-379726\ttRNA Aminoacylation For Mitochondrial Protein Translation (GO:0070127)\n",
      "\n",
      "Mitochondrial Translation R-HSA-5368287\tMitochondrial Translation (GO:0032543)\n",
      "\n",
      "Mitochondrial Translation Elongation R-HSA-5389840\tMitochondrial Translational Elongation (GO:0070125)\n",
      "\n",
      "Mitochondrial Translation Initiation R-HSA-5368286\tMitochondrial Translation (GO:0032543)\n",
      "\n",
      "Mitochondrial Translation Termination R-HSA-5419276\tMitochondrial Translation (GO:0032543)\n",
      "\n",
      "Mitophagy R-HSA-5205647\tMitophagy (GO:0000423)\n",
      "\n",
      "Mitotic G1 Phase And G1/S Transition R-HSA-453279\tG1/S Transition Of Mitotic Cell Cycle (GO:0000082)\n",
      "\n",
      "Mitotic G2-G2/M Phases R-HSA-453274\tCell Cycle G2/M Phase Transition (GO:0044839)\n",
      "\n",
      "Mitotic Spindle Checkpoint R-HSA-69618\tMitotic Spindle Checkpoint Signaling (GO:0071174)\n",
      "\n",
      "Mitotic Telophase/Cytokinesis R-HSA-68884\tMitotic Cytokinesis (GO:0000281)\n",
      "\n",
      "Muscle Contraction R-HSA-397014\tMuscle Contraction (GO:0006936)\n",
      "\n",
      "MyD88 Dependent Cascade Initiated On Endosome R-HSA-975155\tMyD88-dependent Toll-Like Receptor Signaling Pathway (GO:0002755)\n",
      "\n",
      "MyD88-independent TLR4 Cascade R-HSA-166166\tMyD88-independent Toll-Like Receptor Signaling Pathway (GO:0002756)\n",
      "\n",
      "N-Glycan Antennae Elongation R-HSA-975577\tN-glycan Processing (GO:0006491)\n",
      "\n",
      "N-glycan Antennae Elongation In medial/trans-Golgi R-HSA-975576\tN-glycan Processing (GO:0006491)\n",
      "\n",
      "N-glycan Trimming And Elongation In cis-Golgi R-HSA-964739\tN-glycan Processing (GO:0006491)\n",
      "\n",
      "N-glycan Trimming In ER And Calnexin/Calreticulin Cycle R-HSA-532668\tEndoplasmic Reticulum Mannose Trimming (GO:1904380)\n",
      "\n",
      "NIK To Noncanonical NF-kB Signaling R-HSA-5676590\tNIK/NF-kappaB Signaling (GO:0038061)\n",
      "\n",
      "NOD1/2 Signaling Pathway R-HSA-168638\tNucleotide-Binding Oligomerization Domain Containing 2 Signaling Pathway (GO:0070431)\n",
      "\n",
      "NOTCH1 Intracellular Domain Regulates Transcription R-HSA-2122947\tPositive Regulation Of Transcription Of Notch Receptor Target (GO:0007221)\n",
      "\n",
      "NOTCH2 Intracellular Domain Regulates Transcription R-HSA-2197563\tPositive Regulation Of Transcription Of Notch Receptor Target (GO:0007221)\n",
      "\n",
      "NR1H3 And NR1H2 Regulate Gene Expression Linked To Cholesterol Transport And Efflux R-HSA-9029569\tPositive Regulation Of Cholesterol Efflux (GO:0010875)\n",
      "\n",
      "Neddylation R-HSA-8951664\tProtein Neddylation (GO:0045116)\n",
      "\n",
      "Negative Epigenetic Regulation Of rRNA Expression R-HSA-5250941\tNegative Regulation Of Gene Expression, Epigenetic (GO:0045814)\n",
      "\n",
      "Negative Feedback Regulation Of MAPK Pathway R-HSA-5674499\tNegative Regulation Of MAP Kinase Activity (GO:0043407)\n",
      "\n",
      "Negative Regulation Of FGFR1 Signaling R-HSA-5654726\tNegative Regulation Of Fibroblast Growth Factor Receptor Signaling Pathway (GO:0040037)\n",
      "\n",
      "Negative Regulation Of FGFR2 Signaling R-HSA-5654727\tNegative Regulation Of Fibroblast Growth Factor Receptor Signaling Pathway (GO:0040037)\n",
      "\n",
      "Negative Regulation Of FGFR3 Signaling R-HSA-5654732\tNegative Regulation Of Fibroblast Growth Factor Receptor Signaling Pathway (GO:0040037)\n",
      "\n",
      "Negative Regulation Of FGFR4 Signaling R-HSA-5654733\tNegative Regulation Of Fibroblast Growth Factor Receptor Signaling Pathway (GO:0040037)\n",
      "\n",
      "Negative Regulation Of MAPK Pathway R-HSA-5675221\tNegative Regulation Of MAPK Cascade (GO:0043409)\n",
      "\n",
      "Negative Regulation Of MET Activity R-HSA-6807004\tNegative Regulation Of Catalytic Activity (GO:0043086)\n",
      "\n",
      "Negative Regulation Of NMDA Receptor-Mediated Neuronal Transmission R-HSA-9617324\tNegative Regulation Of Synaptic Transmission (GO:0050805)\n",
      "\n",
      "Negative Regulation Of NOTCH4 Signaling R-HSA-9604323\tNegative Regulation Of Notch Signaling Pathway (GO:0045746)\n",
      "\n",
      "Negative Regulation Of TCF-dependent Signaling By WNT Ligand Antagonists R-HSA-3772470\tNegative Regulation Of Wnt Signaling Pathway (GO:0030178)\n",
      "\n",
      "Negative Regulation Of Activity Of TFAP2 (AP-2) Family Transcription Factors R-HSA-8866904\tNegative Regulation Of DNA-binding Transcription Factor Activity (GO:0043433)\n",
      "\n",
      "Negative Regulation Of PI3K/AKT Network R-HSA-199418\tNegative Regulation Of Phosphatidylinositol 3-Kinase Signaling (GO:0014067)\n",
      "\n",
      "Negative Regulators Of DDX58/IFIH1 Signaling R-HSA-936440\tNegative Regulation Of RIG-I Signaling Pathway (GO:0039536)\n",
      "\n",
      "Nervous System Development R-HSA-9675108\tNervous System Development (GO:0007399)\n",
      "\n",
      "Neurotransmitter Clearance R-HSA-112311\tNeurotransmitter Transport (GO:0006836)\n",
      "\n",
      "Neurotransmitter Receptors And Postsynaptic Signal Transmission R-HSA-112314\tNeurotransmitter Receptor Transport To Postsynaptic Membrane (GO:0098969)\n",
      "\n",
      "Neurotransmitter Release Cycle R-HSA-112310\tNeurotransmitter Secretion (GO:0007269)\n",
      "\n",
      "Neutrophil Degranulation R-HSA-6798695\tNeutrophil Degranulation (GO:0043312)\n",
      "\n",
      "Nicotinamide Salvaging R-HSA-197264\tNAD Biosynthesis Via Nicotinamide Riboside Salvage Pathway (GO:0034356)\n",
      "\n",
      "NoRC Negatively Regulates rRNA Expression R-HSA-427413\tPositive Regulation Of rRNA Processing (GO:2000234)\n",
      "\n",
      "Nonhomologous End-Joining (NHEJ) R-HSA-5693571\tDouble-Strand Break Repair Via Nonhomologous End Joining (GO:0006303)\n",
      "\n",
      "Notch-HLH Transcription Pathway R-HSA-350054\tNotch Signaling Pathway (GO:0007219)\n",
      "\n",
      "Nuclear Envelope (NE) Reassembly R-HSA-2995410\tNuclear Membrane Reassembly (GO:0031468)\n",
      "\n",
      "Nuclear Envelope Breakdown R-HSA-2980766\tNuclear Membrane Disassembly (GO:0051081)\n",
      "\n",
      "Nuclear Pore Complex (NPC) Disassembly R-HSA-3301854\tNuclear Pore Complex Assembly (GO:0051292)\n",
      "\n",
      "Nuclear Signaling By ERBB4 R-HSA-1251985\tERBB Signaling Pathway (GO:0038127)\n",
      "\n",
      "Nucleotide Excision Repair R-HSA-5696398\tNucleotide-Excision Repair (GO:0006289)\n",
      "\n",
      "Nucleotide Biosynthesis R-HSA-8956320\tNucleotide Biosynthetic Process (GO:0009165)\n",
      "\n",
      "Nucleotide Catabolism R-HSA-8956319\tNucleotide Catabolic Process (GO:0009166)\n",
      "\n",
      "Nucleotide-binding Domain, Leucine Rich Repeat Containing NLR Signaling Pathways R-HSA-168643\tNucleotide-Binding Domain, Leucine Rich Repeat Containing Receptor Signaling Pathway (GO:0035872)\n",
      "\n",
      "O-linked Glycosylation R-HSA-5173105\tProtein O-linked Glycosylation (GO:0006493)\n",
      "\n",
      "O-linked Glycosylation Of Mucins R-HSA-913709\tProtein O-linked Glycosylation (GO:0006493)\n",
      "\n",
      "Olfactory Signaling Pathway R-HSA-381753\tOsmosensory Signaling Pathway (GO:0007231)\n",
      "\n",
      "Organelle Biogenesis And Maintenance R-HSA-1852241\tOrganelle Assembly (GO:0070925)\n",
      "\n",
      "Organic Anion Transport R-HSA-561048\tOrganic Anion Transport (GO:0015711)\n",
      "\n",
      "Organic Anion Transporters R-HSA-428643\tOrganic Anion Transport (GO:0015711)\n",
      "\n",
      "Organic Cation Transport R-HSA-549127\tOrganic Cation Transport (GO:0015695)\n",
      "\n",
      "Organic Cation/Anion/Zwitterion Transport R-HSA-549132\tOrganic Cation Transport (GO:0015695)\n",
      "\n",
      "Other Interleukin Signaling R-HSA-449836\tInterleukin-3-Mediated Signaling Pathway (GO:0038156)\n",
      "\n",
      "Oxidative Stress Induced Senescence R-HSA-2559580\tStress-Induced Premature Senescence (GO:0090400)\n",
      "\n",
      "Oxygen-dependent Proline Hydroxylation Of Hypoxia-inducible Factor Alpha R-HSA-1234176\tPeptidyl-Proline Hydroxylation (GO:0019511)\n",
      "\n",
      "PI Metabolism R-HSA-1483255\tIMP Metabolic Process (GO:0046040)\n",
      "\n",
      "PI3K Cascade R-HSA-109704\tMAPK Cascade (GO:0000165)\n",
      "\n",
      "PI3K Events In ERBB2 Signaling R-HSA-1963642\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "PI3K/AKT Signaling In Cancer R-HSA-2219528\tPhosphatidylinositol 3-Kinase Signaling (GO:0014065)\n",
      "\n",
      "PI3K/AKT Activation R-HSA-198203\tActivation Of Protein Kinase B Activity (GO:0032148)\n",
      "\n",
      "PI5P, PP2A And IER3 Regulate PI3K/AKT Signaling R-HSA-6811558\tRegulation Of Phosphatidylinositol 3-Kinase Signaling (GO:0014066)\n",
      "\n",
      "PINK1-PRKN Mediated Mitophagy R-HSA-5205685\tRegulation Of Mitophagy (GO:1901524)\n",
      "\n",
      "PKMTs Methylate Histone Lysines R-HSA-3214841\tHistone Lysine Methylation (GO:0034968)\n",
      "\n",
      "Packaging Of Telomere Ends R-HSA-171306\tTelomere Capping (GO:0016233)\n",
      "\n",
      "Pentose Phosphate Pathway R-HSA-71336\tPentose-Phosphate Shunt (GO:0006098)\n",
      "\n",
      "Peptide Chain Elongation R-HSA-156902\tTranslational Elongation (GO:0006414)\n",
      "\n",
      "Peptide Hormone Biosynthesis R-HSA-209952\tPeptide Biosynthetic Process (GO:0043043)\n",
      "\n",
      "Peptide Hormone Metabolism R-HSA-2980736\tPeptide Metabolic Process (GO:0006518)\n",
      "\n",
      "Peroxisomal Lipid Metabolism R-HSA-390918\tPeroxisomal Transport (GO:0043574)\n",
      "\n",
      "Peroxisomal Protein Import R-HSA-9033241\tProtein Import Into Peroxisome Matrix (GO:0016558)\n",
      "\n",
      "Pexophagy R-HSA-9664873\tPexophagy (GO:0000425)\n",
      "\n",
      "Phenylalanine Metabolism R-HSA-8964208\tL-phenylalanine Metabolic Process (GO:0006558)\n",
      "\n",
      "Phospholipid Metabolism R-HSA-1483257\tPhospholipid Metabolic Process (GO:0006644)\n",
      "\n",
      "Plasma Lipoprotein Assembly R-HSA-8963898\tHigh-Density Lipoprotein Particle Assembly (GO:0034380)\n",
      "\n",
      "Plasma Lipoprotein Assembly, Remodeling, And Clearance R-HSA-174824\tHigh-Density Lipoprotein Particle Assembly (GO:0034380)\n",
      "\n",
      "Plasma Lipoprotein Clearance R-HSA-8964043\tRegulation Of Low-Density Lipoprotein Particle Clearance (GO:0010988)\n",
      "\n",
      "Plasma Lipoprotein Remodeling R-HSA-8963899\tLow-Density Lipoprotein Particle Remodeling (GO:0034374)\n",
      "\n",
      "Platelet Aggregation (Plug Formation) R-HSA-76009\tPlatelet Aggregation (GO:0070527)\n",
      "\n",
      "Platelet Activation, Signaling And Aggregation R-HSA-76002\tPlatelet Aggregation (GO:0070527)\n",
      "\n",
      "Platelet Calcium Homeostasis R-HSA-418360\tCalcium Ion Homeostasis (GO:0055074)\n",
      "\n",
      "Platelet Degranulation R-HSA-114608\tNeutrophil Degranulation (GO:0043312)\n",
      "\n",
      "Positive Epigenetic Regulation Of rRNA Expression R-HSA-5250913\tPositive Regulation Of rRNA Processing (GO:2000234)\n",
      "\n",
      "Post-transcriptional Silencing By Small RNAs R-HSA-426496\tPost-Transcriptional Gene Silencing (GO:0016441)\n",
      "\n",
      "Post-translational Modification: Synthesis Of GPI-anchored Proteins R-HSA-163125\tGPI Anchor Biosynthetic Process (GO:0006506)\n",
      "\n",
      "Post-translational Protein Modification R-HSA-597592\tPost-Translational Protein Modification (GO:0043687)\n",
      "\n",
      "Post-translational Protein Phosphorylation R-HSA-8957275\tPost-Translational Protein Modification (GO:0043687)\n",
      "\n",
      "Postmitotic Nuclear Pore Complex (NPC) Reformation R-HSA-9615933\tNuclear Pore Complex Assembly (GO:0051292)\n",
      "\n",
      "Processing Of DNA Double-Strand Break Ends R-HSA-5693607\tDNA Double-Strand Break Processing (GO:0000729)\n",
      "\n",
      "Processing Of Intronless Pre-mRNAs R-HSA-77595\tmRNA Processing (GO:0006397)\n",
      "\n",
      "Programmed Cell Death R-HSA-5357801\tProgrammed Necrotic Cell Death (GO:0097300)\n",
      "\n",
      "Prolactin Receptor Signaling R-HSA-1170546\tProgesterone Receptor Signaling Pathway (GO:0050847)\n",
      "\n",
      "Protein Localization R-HSA-9609507\tProtein Localization (GO:0008104)\n",
      "\n",
      "Protein Methylation R-HSA-8876725\tProtein Methylation (GO:0006479)\n",
      "\n",
      "Protein Repair R-HSA-5676934\tProtein Repair (GO:0030091)\n",
      "\n",
      "Protein Ubiquitination R-HSA-8852135\tProtein Ubiquitination (GO:0016567)\n",
      "\n",
      "Protein-protein Interactions At Synapses R-HSA-6794362\tProtein Localization To Synapse (GO:0035418)\n",
      "\n",
      "Proton-coupled Monocarboxylate Transport R-HSA-433692\tMonocarboxylic Acid Transport (GO:0015718)\n",
      "\n",
      "Purine Catabolism R-HSA-74259\tPurine-Containing Compound Catabolic Process (GO:0072523)\n",
      "\n",
      "Purine Ribonucleoside Monophosphate Biosynthesis R-HSA-73817\tPurine Ribonucleoside Monophosphate Biosynthetic Process (GO:0009168)\n",
      "\n",
      "Pyrimidine Catabolism R-HSA-73621\tPyrimidine Nucleobase Catabolic Process (GO:0006208)\n",
      "\n",
      "Pyrimidine Salvage R-HSA-73614\tPyrimidine Nucleobase Catabolic Process (GO:0006208)\n",
      "\n",
      "Pyroptosis R-HSA-5620971\tPyroptosis (GO:0070269)\n",
      "\n",
      "Pyruvate Metabolism R-HSA-70268\tPyruvate Metabolic Process (GO:0006090)\n",
      "\n",
      "RA Biosynthesis Pathway R-HSA-5365859\tIMP Biosynthetic Process (GO:0006188)\n",
      "\n",
      "RAB Geranylgeranylation R-HSA-8873719\tProtein Geranylgeranylation (GO:0018344)\n",
      "\n",
      "RAF/MAP Kinase Cascade R-HSA-5673001\tMAPK Cascade (GO:0000165)\n",
      "\n",
      "RMTs Methylate Histone Arginines R-HSA-3214858\tHistone Arginine Methylation (GO:0034969)\n",
      "\n",
      "RNA Polymerase I Promoter Clearance R-HSA-73854\tTranscription Initiation At RNA Polymerase I Promoter (GO:0006361)\n",
      "\n",
      "RNA Polymerase I Promoter Escape R-HSA-73772\tTranscription Initiation At RNA Polymerase I Promoter (GO:0006361)\n",
      "\n",
      "RNA Polymerase I Promoter Opening R-HSA-73728\tTranscription Initiation At RNA Polymerase I Promoter (GO:0006361)\n",
      "\n",
      "RNA Polymerase I Transcription Initiation R-HSA-73762\tTranscription Initiation At RNA Polymerase I Promoter (GO:0006361)\n",
      "\n",
      "RNA Polymerase I Transcription R-HSA-73864\tTranscription By RNA Polymerase I (GO:0006360)\n",
      "\n",
      "RNA Polymerase I Transcription Termination R-HSA-73863\tTermination Of RNA Polymerase II Transcription (GO:0006369)\n",
      "\n",
      "RNA Polymerase II Pre-transcription Events R-HSA-674695\tRNA Polymerase II Preinitiation Complex Assembly (GO:0051123)\n",
      "\n",
      "RNA Polymerase II Transcription R-HSA-73857\tTranscription By RNA Polymerase II (GO:0006366)\n",
      "\n",
      "RNA Polymerase II Transcription Termination R-HSA-73856\tTermination Of RNA Polymerase II Transcription (GO:0006369)\n",
      "\n",
      "RNA Polymerase III Transcription Initiation From Type 1 Promoter R-HSA-76061\tTranscription Initiation At RNA Polymerase I Promoter (GO:0006361)\n",
      "\n",
      "RNA Polymerase III Transcription Initiation From Type 2 Promoter R-HSA-76066\tTranscription By RNA Polymerase III (GO:0006383)\n",
      "\n",
      "RNA Polymerase III Transcription Initiation From Type 3 Promoter R-HSA-76071\tTranscription By RNA Polymerase III (GO:0006383)\n",
      "\n",
      "RNA Polymerase III Transcription Initiation R-HSA-76046\tTranscription By RNA Polymerase III (GO:0006383)\n",
      "\n",
      "RNA Polymerase III Transcription Termination R-HSA-73980\tTermination Of RNA Polymerase II Transcription (GO:0006369)\n",
      "\n",
      "RNA Polymerase II Transcribes snRNA Genes R-HSA-6807505\tsnRNA Transcription By RNA Polymerase II (GO:0042795)\n",
      "\n",
      "RUNX2 Regulates Osteoblast Differentiation R-HSA-8940973\tRegulation Of Osteoblast Differentiation (GO:0045667)\n",
      "\n",
      "Rap1 Signaling R-HSA-392517\tRas Protein Signal Transduction (GO:0007265)\n",
      "\n",
      "Receptor Mediated Mitophagy R-HSA-8934903\tPositive Regulation Of Mitophagy (GO:1901526)\n",
      "\n",
      "Receptor-type Tyrosine-Protein Phosphatases R-HSA-388844\tTransmembrane Receptor Protein Tyrosine Phosphatase Signaling Pathway (GO:0007185)\n",
      "\n",
      "Regulated Necrosis R-HSA-5218859\tNecroptotic Process (GO:0070266)\n",
      "\n",
      "Regulation Of Apoptosis R-HSA-169911\tRegulation Of Apoptotic Process (GO:0042981)\n",
      "\n",
      "Regulation Of Complement Cascade R-HSA-977606\tRegulation Of Complement Activation (GO:0030449)\n",
      "\n",
      "Regulation Of HSF1-mediated Heat Shock Response R-HSA-3371453\tRegulation Of Cellular Response To Heat (GO:1900034)\n",
      "\n",
      "Regulation Of IFNA/IFNB Signaling R-HSA-912694\tRegulation Of Type I Interferon-Mediated Signaling Pathway (GO:0060338)\n",
      "\n",
      "Regulation Of IFNG Signaling R-HSA-877312\tRegulation Of Type I Interferon-Mediated Signaling Pathway (GO:0060338)\n",
      "\n",
      "Regulation Of NF-kappa B Signaling R-HSA-9758274\tRegulation Of I-kappaB kinase/NF-kappaB Signaling (GO:0043122)\n",
      "\n",
      "Regulation Of PLK1 Activity At G2/M Transition R-HSA-2565942\tRegulation Of Cell Cycle G2/M Phase Transition (GO:1902749)\n",
      "\n",
      "Regulation Of RAS By GAPs R-HSA-5658442\tRegulation Of Ras Protein Signal Transduction (GO:0046578)\n",
      "\n",
      "Regulation Of RUNX2 Expression And Activity R-HSA-8939902\tRegulation Of Osteoblast Differentiation (GO:0045667)\n",
      "\n",
      "Regulation Of TLR By Endogenous Ligand R-HSA-5686938\tRegulation Of Toll-Like Receptor Signaling Pathway (GO:0034121)\n",
      "\n",
      "Regulation Of TNFR1 Signaling R-HSA-5357905\tRegulation Of Tumor Necrosis Factor-Mediated Signaling Pathway (GO:0010803)\n",
      "\n",
      "Regulation Of TP53 Activity R-HSA-5633007\tRegulation Of Signal Transduction By P53 Class Mediator (GO:1901796)\n",
      "\n",
      "Regulation Of Actin Dynamics For Phagocytic Cup Formation R-HSA-2029482\tRegulation Of Actin Cytoskeleton Organization (GO:0032956)\n",
      "\n",
      "Regulation Of Beta-Cell Development R-HSA-186712\tRegulation Of Type B Pancreatic Cell Development (GO:2000074)\n",
      "\n",
      "Regulation Of Cholesterol Biosynthesis By SREBP (SREBF) R-HSA-1655829\tRegulation Of Cholesterol Biosynthetic Process (GO:0045540)\n",
      "\n",
      "Regulation Of Gene Expression By Hypoxia-inducible Factor R-HSA-1234158\tRegulation Of Cellular Response To Hypoxia (GO:1900037)\n",
      "\n",
      "Regulation Of Gene Expression In Beta Cells R-HSA-210745\tRegulation Of Gene Expression (GO:0010468)\n",
      "\n",
      "Regulation Of Gene Expression In Early Pancreatic Precursor Cells R-HSA-210747\tRegulation Of Type B Pancreatic Cell Development (GO:2000074)\n",
      "\n",
      "Regulation Of Glycolysis By Fructose 2,6-Bisphosphate Metabolism R-HSA-9634600\tRegulation Of Glycolytic Process (GO:0006110)\n",
      "\n",
      "Regulation Of Innate Immune Responses To Cytosolic DNA R-HSA-3134975\tRegulation Of Innate Immune Response (GO:0045088)\n",
      "\n",
      "Regulation Of Insulin Secretion R-HSA-422356\tRegulation Of Insulin Secretion (GO:0050796)\n",
      "\n",
      "Regulation Of Localization Of FOXO Transcription Factors R-HSA-9614399\tRegulation Of Cellular Localization (GO:0060341)\n",
      "\n",
      "Regulation Of mRNA Stability By Proteins That Bind AU-rich Elements R-HSA-450531\tRegulation Of mRNA Stability (GO:0043488)\n",
      "\n",
      "Regulation Of Necroptotic Cell Death R-HSA-5675482\tRegulation Of Necrotic Cell Death (GO:0010939)\n",
      "\n",
      "Regulation Of Pyruvate Dehydrogenase (PDH) Complex R-HSA-204174\tRegulation Of Pyruvate Dehydrogenase Activity (GO:1904182)\n",
      "\n",
      "Release Of Apoptotic Factors From Mitochondria R-HSA-111457\tRelease Of Cytochrome C From Mitochondria (GO:0001836)\n",
      "\n",
      "Repression Of WNT Target Genes R-HSA-4641265\tNegative Regulation Of Wnt Signaling Pathway (GO:0030178)\n",
      "\n",
      "Resolution Of D-loop Structures Thru Holliday Junction Intermediates R-HSA-5693568\tResolution Of Recombination Intermediates (GO:0071139)\n",
      "\n",
      "Resolution Of D-loop Structures Thru Synthesis-Dependent Strand Annealing (SDSA) R-HSA-5693554\tDouble-Strand Break Repair Via Synthesis-Dependent Strand Annealing (GO:0045003)\n",
      "\n",
      "Resolution Of Sister Chromatid Cohesion R-HSA-2500257\tMitotic Sister Chromatid Cohesion (GO:0007064)\n",
      "\n",
      "Respiratory Electron Transport R-HSA-611105\tRespiratory Electron Transport Chain (GO:0022904)\n",
      "\n",
      "Response Of EIF2AK4 (GCN2) To Amino Acid Deficiency R-HSA-9633012\tResponse To Amino Acid Starvation (GO:1990928)\n",
      "\n",
      "Response To Metal Ions R-HSA-5660526\tResponse To Metal Ion (GO:0010038)\n",
      "\n",
      "Retinoid Metabolism And Transport R-HSA-975634\tRetinoid Metabolic Process (GO:0001523)\n",
      "\n",
      "Retrograde Neurotrophin Signaling R-HSA-177504\tNeurotrophin Signaling Pathway (GO:0038179)\n",
      "\n",
      "Retrograde Transport At Trans-Golgi-Network R-HSA-6811440\tRetrograde Transport, Endosome To Golgi (GO:0042147)\n",
      "\n",
      "SEMA3A-Plexin Repulsion Signaling By Inhibiting Integrin Adhesion R-HSA-399955\tSemaphorin-Plexin Signaling Pathway (GO:0071526)\n",
      "\n",
      "SHC1 Events In ERBB2 Signaling R-HSA-1250196\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "SMAD2/SMAD3:SMAD4 Heterotrimer Regulates Transcription R-HSA-2173796\tRegulation Of SMAD Protein Signal Transduction (GO:0060390)\n",
      "\n",
      "SRP-dependent Cotranslational Protein Targeting To Membrane R-HSA-1799339\tSRP-dependent Cotranslational Protein Targeting To Membrane (GO:0006614)\n",
      "\n",
      "SUMO E3 Ligases SUMOylate Target Proteins R-HSA-3108232\tRegulation Of Protein Sumoylation (GO:0033233)\n",
      "\n",
      "SUMOylation R-HSA-2990846\tProtein Sumoylation (GO:0016925)\n",
      "\n",
      "SUMOylation Of SUMOylation Proteins R-HSA-4085377\tRegulation Of Protein Sumoylation (GO:0033233)\n",
      "\n",
      "SUMOylation Of Chromatin Organization Proteins R-HSA-4551638\tProtein Sumoylation (GO:0016925)\n",
      "\n",
      "SUMOylation Of Immune Response Proteins R-HSA-4755510\tProtein Sumoylation (GO:0016925)\n",
      "\n",
      "SUMOylation Of Transcription Cofactors R-HSA-3899300\tProtein Sumoylation (GO:0016925)\n",
      "\n",
      "SUMOylation Of Transcription Factors R-HSA-3232118\tProtein Sumoylation (GO:0016925)\n",
      "\n",
      "SUMOylation Of Ubiquitinylation Proteins R-HSA-3232142\tRegulation Of Protein Sumoylation (GO:0033233)\n",
      "\n",
      "Selective Autophagy R-HSA-9663891\tSelective Autophagy (GO:0061912)\n",
      "\n",
      "Selenocysteine Synthesis R-HSA-2408557\tSelenocysteine Incorporation (GO:0001514)\n",
      "\n",
      "Semaphorin Interactions R-HSA-373755\tSemaphorin-Plexin Signaling Pathway (GO:0071526)\n",
      "\n",
      "Sensing Of DNA Double Strand Breaks R-HSA-5693548\tSignal Transduction In Response To DNA Damage (GO:0042770)\n",
      "\n",
      "Sensory Perception R-HSA-9709957\tSensory Perception (GO:0007600)\n",
      "\n",
      "Sensory Perception Of Salty Taste R-HSA-9730628\tSensory Perception Of Taste (GO:0050909)\n",
      "\n",
      "Sensory Perception Of Sweet, Bitter, And Umami (Glutamate) Taste R-HSA-9717207\tSensory Perception Of Taste (GO:0050909)\n",
      "\n",
      "Sensory Perception Of Taste R-HSA-9717189\tSensory Perception Of Taste (GO:0050909)\n",
      "\n",
      "Sensory Processing Of Sound R-HSA-9659379\tSensory Perception Of Sound (GO:0007605)\n",
      "\n",
      "Separation Of Sister Chromatids R-HSA-2467813\tSister Chromatid Segregation (GO:0000819)\n",
      "\n",
      "Serine Biosynthesis R-HSA-977347\tSerine Family Amino Acid Biosynthetic Process (GO:0009070)\n",
      "\n",
      "Signaling By Activin R-HSA-1502540\tActivin Receptor Signaling Pathway (GO:0032924)\n",
      "\n",
      "Signaling By BMP R-HSA-201451\tRegulation Of BMP Signaling Pathway (GO:0030510)\n",
      "\n",
      "Signaling By EGFR R-HSA-177929\tEpidermal Growth Factor Receptor Signaling Pathway (GO:0007173)\n",
      "\n",
      "Signaling By ERBB2 KD Mutants R-HSA-9664565\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "Signaling By ERBB2 R-HSA-1227986\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "Signaling By ERBB2 In Cancer R-HSA-1227990\tERBB2 Signaling Pathway (GO:0038128)\n",
      "\n",
      "Signaling By ERBB4 R-HSA-1236394\tERBB Signaling Pathway (GO:0038127)\n",
      "\n",
      "Signaling By FGFR R-HSA-190236\tFibroblast Growth Factor Receptor Signaling Pathway (GO:0008543)\n",
      "\n",
      "Signaling By FGFR1 R-HSA-5654736\tFibroblast Growth Factor Receptor Signaling Pathway (GO:0008543)\n",
      "\n",
      "Signaling By Hippo R-HSA-2028269\tHippo Signaling (GO:0035329)\n",
      "\n",
      "Signaling By Insulin Receptor R-HSA-74752\tInsulin Receptor Signaling Pathway (GO:0008286)\n",
      "\n",
      "Signaling By Leptin R-HSA-2586552\tLeptin-Mediated Signaling Pathway (GO:0033210)\n",
      "\n",
      "Signaling By NOTCH R-HSA-157118\tNotch Signaling Pathway (GO:0007219)\n",
      "\n",
      "Signaling By NOTCH1 R-HSA-1980143\tRegulation Of Notch Signaling Pathway (GO:0008593)\n",
      "\n",
      "Signaling By NOTCH2 R-HSA-1980145\tRegulation Of Notch Signaling Pathway (GO:0008593)\n",
      "\n",
      "Signaling By NTRK1 (TRKA) R-HSA-187037\tNeurotrophin TRK Receptor Signaling Pathway (GO:0048011)\n",
      "\n",
      "Signaling By NTRK2 (TRKB) R-HSA-9006115\tNeurotrophin TRK Receptor Signaling Pathway (GO:0048011)\n",
      "\n",
      "Signaling By NTRK3 (TRKC) R-HSA-9034015\tNeurotrophin TRK Receptor Signaling Pathway (GO:0048011)\n",
      "\n",
      "Signaling By NTRKs R-HSA-166520\tNeurotrophin TRK Receptor Signaling Pathway (GO:0048011)\n",
      "\n",
      "Signaling By PDGF R-HSA-186797\tPlatelet-Derived Growth Factor Receptor Signaling Pathway (GO:0048008)\n",
      "\n",
      "Signaling By Receptor Tyrosine Kinases R-HSA-9006934\tTransmembrane Receptor Protein Tyrosine Kinase Signaling Pathway (GO:0007169)\n",
      "\n",
      "Signaling By Retinoic Acid R-HSA-5362517\tRetinoic Acid Receptor Signaling Pathway (GO:0048384)\n",
      "\n",
      "Signaling By Rho GTPases R-HSA-194315\tRegulation Of Rho Protein Signal Transduction (GO:0035023)\n",
      "\n",
      "Signaling By TGF-beta Receptor Complex R-HSA-170834\tTransforming Growth Factor Beta Receptor Signaling Pathway (GO:0007179)\n",
      "\n",
      "Signaling By TGF-beta Receptor Complex In Cancer R-HSA-3304351\tTransforming Growth Factor Beta Receptor Signaling Pathway (GO:0007179)\n",
      "\n",
      "Signaling By Type 1 Insulin-like Growth Factor 1 Receptor (IGF1R) R-HSA-2404192\tInsulin-Like Growth Factor Receptor Signaling Pathway (GO:0048009)\n",
      "\n",
      "Signaling By VEGF R-HSA-194138\tVascular Endothelial Growth Factor Signaling Pathway (GO:0038084)\n",
      "\n",
      "Signaling By WNT R-HSA-195721\tCell-Cell Signaling By Wnt (GO:0198738)\n",
      "\n",
      "Signaling By WNT In Cancer R-HSA-4791275\tCell-Cell Signaling By Wnt (GO:0198738)\n",
      "\n",
      "Signaling By B Cell Receptor (BCR) R-HSA-983705\tB Cell Receptor Signaling Pathway (GO:0050853)\n",
      "\n",
      "Signaling To RAS R-HSA-167044\tRegulation Of Ras Protein Signal Transduction (GO:0046578)\n",
      "\n",
      "Small Interfering RNA (siRNA) Biogenesis R-HSA-426486\tsiRNA Processing (GO:0030422)\n",
      "\n",
      "Smooth Muscle Contraction R-HSA-445355\tSmooth Muscle Contraction (GO:0006939)\n",
      "\n",
      "Sodium-coupled Phosphate Cotransporters R-HSA-427652\tSodium-Dependent Phosphate Transport (GO:0044341)\n",
      "\n",
      "Sphingolipid De Novo Biosynthesis R-HSA-1660661\tSphingosine Biosynthetic Process (GO:0046512)\n",
      "\n",
      "Sphingolipid Metabolism R-HSA-428157\tSphingolipid Metabolic Process (GO:0006665)\n",
      "\n",
      "Striated Muscle Contraction R-HSA-390522\tStriated Muscle Contraction (GO:0006941)\n",
      "\n",
      "Sulfur Amino Acid Metabolism R-HSA-1614635\tSulfur Amino Acid Metabolic Process (GO:0000096)\n",
      "\n",
      "Surfactant Metabolism R-HSA-5683826\tSurfactant Homeostasis (GO:0043129)\n",
      "\n",
      "Synaptic Adhesion-Like Molecules R-HSA-8849932\tSynaptic Membrane Adhesion (GO:0099560)\n",
      "\n",
      "Synthesis Of GDP-mannose R-HSA-446205\tGDP-mannose Metabolic Process (GO:0019673)\n",
      "\n",
      "Synthesis Of Ketone Bodies R-HSA-77111\tKetone Body Biosynthetic Process (GO:0046951)\n",
      "\n",
      "Synthesis Of UDP-N-acetyl-glucosamine R-HSA-446210\tUDP-N-acetylglucosamine Biosynthetic Process (GO:0006048)\n",
      "\n",
      "Synthesis Of Bile Acids And Bile Salts R-HSA-192105\tBile Acid Biosynthetic Process (GO:0006699)\n",
      "\n",
      "Synthesis Of Bile Acids And Bile Salts Via 7Alpha-Hydroxycholesterol R-HSA-193368\tBile Acid Biosynthetic Process (GO:0006699)\n",
      "\n",
      "Synthesis Of diphthamide-EEF2 R-HSA-5358493\tPeptidyl-Diphthamide Biosynthetic Process From Peptidyl-Histidine (GO:0017183)\n",
      "\n",
      "Synthesis Of Glycosylphosphatidylinositol (GPI) R-HSA-162710\tGPI Anchor Biosynthetic Process (GO:0006506)\n",
      "\n",
      "Synthesis Of Very Long-Chain Fatty acyl-CoAs R-HSA-75876\tLong-Chain fatty-acyl-CoA Biosynthetic Process (GO:0035338)\n",
      "\n",
      "Synthesis Of Wybutosine At G37 Of tRNA(Phe) R-HSA-6782861\tWybutosine Biosynthetic Process (GO:0031591)\n",
      "\n",
      "TAK1-dependent IKK And NF-kappa-B Activation R-HSA-445989\tActivation Of NF-kappaB-inducing Kinase Activity (GO:0007250)\n",
      "\n",
      "TCR Signaling R-HSA-202403\tT Cell Receptor Signaling Pathway (GO:0050852)\n",
      "\n",
      "TGF-beta Receptor Signaling Activates SMADs R-HSA-2173789\tTransforming Growth Factor Beta Receptor Signaling Pathway (GO:0007179)\n",
      "\n",
      "TGF-beta Receptor Signaling In EMT (Epithelial To Mesenchymal Transition) R-HSA-2173791\tTransforming Growth Factor Beta Receptor Signaling Pathway (GO:0007179)\n",
      "\n",
      "TNF Signaling R-HSA-75893\tTumor Necrosis Factor-Mediated Signaling Pathway (GO:0033209)\n",
      "\n",
      "TNFR1-induced NFkappaB Signaling Pathway R-HSA-5357956\tTumor Necrosis Factor-Mediated Signaling Pathway (GO:0033209)\n",
      "\n",
      "TP53 Regulates Transcription Of DNA Repair Genes R-HSA-6796648\tPositive Regulation Of DNA Repair (GO:0045739)\n",
      "\n",
      "Telomere Extension By Telomerase R-HSA-171319\tTelomere Maintenance Via Telomerase (GO:0007004)\n",
      "\n",
      "Telomere Maintenance R-HSA-157579\tTelomere Maintenance (GO:0000723)\n",
      "\n",
      "NLRP3 Inflammasome R-HSA-844456\tPositive Regulation Of NLRP3 Inflammasome Complex Assembly (GO:1900227)\n",
      "\n",
      "Tight Junction Interactions R-HSA-420029\tTight Junction Organization (GO:0120193)\n",
      "\n",
      "Toll Like Receptor 3 (TLR3) Cascade R-HSA-168164\tToll-Like Receptor 3 Signaling Pathway (GO:0034138)\n",
      "\n",
      "Toll Like Receptor 4 (TLR4) Cascade R-HSA-166016\tToll-Like Receptor 4 Signaling Pathway (GO:0034142)\n",
      "\n",
      "Toll Like Receptor 9 (TLR9) Cascade R-HSA-168138\tToll-Like Receptor 9 Signaling Pathway (GO:0034162)\n",
      "\n",
      "Toll-like Receptor Cascades R-HSA-168898\tToll-Like Receptor Signaling Pathway (GO:0002224)\n",
      "\n",
      "Transcription-Coupled Nucleotide Excision Repair (TC-NER) R-HSA-6781827\tTranscription-Coupled Nucleotide-Excision Repair (GO:0006283)\n",
      "\n",
      "Transcriptional Activity Of SMAD2/SMAD3:SMAD4 Heterotrimer R-HSA-2173793\tPositive Regulation Of SMAD Protein Signal Transduction (GO:0060391)\n",
      "\n",
      "Transcriptional Regulation By Small RNAs R-HSA-5578749\tRegulation Of Post-Transcriptional Gene Silencing By RNA (GO:1900368)\n",
      "\n",
      "Transcriptional Regulation Of Granulopoiesis R-HSA-9616222\tRegulation Of Granulocyte Differentiation (GO:0030852)\n",
      "\n",
      "Transcriptional Regulation Of Testis Differentiation R-HSA-9690406\tPositive Regulation Of Male Gonad Development (GO:2000020)\n",
      "\n",
      "Transcriptional Regulation Of White Adipocyte Differentiation R-HSA-381340\tRegulation Of Fat Cell Differentiation (GO:0045598)\n",
      "\n",
      "Transferrin Endocytosis And Recycling R-HSA-917977\tTransferrin Transport (GO:0033572)\n",
      "\n",
      "Translation R-HSA-72766\tTranslation (GO:0006412)\n",
      "\n",
      "Translation Initiation Complex Formation R-HSA-72649\tFormation Of Translation Preinitiation Complex (GO:0001731)\n",
      "\n",
      "Translesion Synthesis By POLH R-HSA-110320\tTranslesion Synthesis (GO:0019985)\n",
      "\n",
      "Translesion Synthesis By POLI R-HSA-5656121\tTranslesion Synthesis (GO:0019985)\n",
      "\n",
      "Translesion Synthesis By POLK R-HSA-5655862\tTranslesion Synthesis (GO:0019985)\n",
      "\n",
      "Translesion Synthesis By REV1 R-HSA-110312\tError-Prone Translesion Synthesis (GO:0042276)\n",
      "\n",
      "Translesion Synthesis By Y Family DNA Polymerases Bypasses Lesions On DNA Template R-HSA-110313\tTranslesion Synthesis (GO:0019985)\n",
      "\n",
      "Transmission Across Chemical Synapses R-HSA-112315\tChemical Synaptic Transmission (GO:0007268)\n",
      "\n",
      "Transport Of Mature mRNA Derived From An Intronless Transcript R-HSA-159231\tmRNA Transport (GO:0051028)\n",
      "\n",
      "Transport Of Mature mRNAs Derived From Intronless Transcripts R-HSA-159234\tmRNA Transport (GO:0051028)\n",
      "\n",
      "Transport Of Bile Salts And Organic Acids, Metal Ions And Amine Compounds R-HSA-425366\tBile Acid And Bile Salt Transport (GO:0015721)\n",
      "\n",
      "Transport Of Fatty Acids R-HSA-804914\tFatty Acid Transport (GO:0015908)\n",
      "\n",
      "Transport Of Inorganic Cations/Anions And Amino Acids/Oligopeptides R-HSA-425393\tInorganic Cation Transmembrane Transport (GO:0098662)\n",
      "\n",
      "Transport Of Nucleosides And Free Purine And Pyrimidine Bases Across Plasma Membrane R-HSA-83936\tPurine Nucleobase Transport (GO:0006863)\n",
      "\n",
      "Transport Of Nucleotide Sugars R-HSA-727802\tNucleotide Transport (GO:0006862)\n",
      "\n",
      "Transport Of Organic Anions R-HSA-879518\tOrganic Anion Transport (GO:0015711)\n",
      "\n",
      "Transport Of Vitamins, Nucleosides, And Related Molecules R-HSA-425397\tNucleoside Transport (GO:0015858)\n",
      "\n",
      "Triglyceride Biosynthesis R-HSA-75109\tTriglyceride Biosynthetic Process (GO:0019432)\n",
      "\n",
      "Triglyceride Catabolism R-HSA-163560\tTriglyceride Catabolic Process (GO:0019433)\n",
      "\n",
      "Triglyceride Metabolism R-HSA-8979227\tTriglyceride Metabolic Process (GO:0006641)\n",
      "\n",
      "Tryptophan Catabolism R-HSA-71240\tTryptophan Catabolic Process (GO:0006569)\n",
      "\n",
      "Tyrosine Catabolism R-HSA-8963684\tTyrosine Metabolic Process (GO:0006570)\n",
      "\n",
      "Ubiquinol Biosynthesis R-HSA-2142789\tUbiquinone Biosynthetic Process (GO:0006744)\n",
      "\n",
      "Unfolded Protein Response (UPR) R-HSA-381119\tIRE1-mediated Unfolded Protein Response (GO:0036498)\n",
      "\n",
      "Unwinding Of DNA R-HSA-176974\tDNA Unwinding Involved In DNA Replication (GO:0006268)\n",
      "\n",
      "Urea Cycle R-HSA-70635\tUrea Cycle (GO:0000050)\n",
      "\n",
      "VEGFA-VEGFR2 Pathway R-HSA-4420097\tVascular Endothelial Growth Factor Receptor-2 Signaling Pathway (GO:0036324)\n",
      "\n",
      "VEGFR2 Mediated Vascular Permeability R-HSA-5218920\tPositive Regulation Of Vascular Permeability (GO:0043117)\n",
      "\n",
      "Vesicle-mediated Transport R-HSA-5653656\tVesicle-Mediated Transport (GO:0016192)\n",
      "\n",
      "Visual Phototransduction R-HSA-2187338\tPhototransduction, Visible Light (GO:0007603)\n",
      "\n",
      "Vitamin B1 (Thiamin) Metabolism R-HSA-196819\tThiamine-Containing Compound Metabolic Process (GO:0042723)\n",
      "\n",
      "Vitamin C (Ascorbate) Metabolism R-HSA-196836\tL-ascorbic Acid Metabolic Process (GO:0019852)\n",
      "\n",
      "Vitamin D (Calciferol) Metabolism R-HSA-196791\tVitamin D Metabolic Process (GO:0042359)\n",
      "\n",
      "Xenobiotics R-HSA-211981\tXenobiotic Transport (GO:0042908)\n",
      "\n",
      "Zinc Transporters R-HSA-435354\tZinc Ion Transport (GO:0006829)\n",
      "\n",
      "Alpha-Linolenic (Omega3) And Linoleic (Omega6) Acid Metabolism R-HSA-2046104\tAlpha-Linolenic Acid Metabolic Process (GO:0036109)\n",
      "\n",
      "cGMP Effects R-HSA-418457\tResponse To cGMP (GO:0070305)\n",
      "\n",
      "mRNA 3-End Processing R-HSA-72187\tmRNA 3'-End Processing (GO:0031124)\n",
      "\n",
      "mRNA Capping R-HSA-72086\t7-Methylguanosine mRNA Capping (GO:0006370)\n",
      "\n",
      "mRNA Editing R-HSA-75072\tmRNA Modification (GO:0016556)\n",
      "\n",
      "mRNA Splicing - Major Pathway R-HSA-72163\tmRNA Splicing, Via Spliceosome (GO:0000398)\n",
      "\n",
      "mRNA Splicing - Minor Pathway R-HSA-72165\tmRNA Splicing, Via Spliceosome (GO:0000398)\n",
      "\n",
      "mRNA Splicing R-HSA-72172\tmRNA Splicing, Via Spliceosome (GO:0000398)\n",
      "\n",
      "mTORC1-mediated Signaling R-HSA-166208\tTORC1 Signaling (GO:0038202)\n",
      "\n",
      "Mitochondrial Fatty Acid Beta-Oxidation Of Saturated Fatty Acids R-HSA-77286\tFatty Acid Beta-Oxidation (GO:0006635)\n",
      "\n",
      "Mitochondrial Fatty Acid Beta-Oxidation Of Unsaturated Fatty Acids R-HSA-77288\tFatty Acid Beta-Oxidation (GO:0006635)\n",
      "\n",
      "p38MAPK Events R-HSA-171007\tp38MAPK Cascade (GO:0038066)\n",
      "\n",
      "p53-Dependent G1 DNA Damage Response R-HSA-69563\tDNA Damage Response, Signal Transduction By P53 Class Mediator Resulting In Cell Cycle Arrest (GO:0006977)\n",
      "\n",
      "p75NTR Regulates Axonogenesis R-HSA-193697\tPositive Regulation Of Axonogenesis (GO:0050772)\n",
      "\n",
      "rRNA Modification In Mitochondrion R-HSA-6793080\tMitochondrial tRNA Modification (GO:0070900)\n",
      "\n",
      "rRNA Modification In Nucleus And Cytosol R-HSA-6790901\trRNA Modification (GO:0000154)\n",
      "\n",
      "rRNA Processing R-HSA-72312\trRNA Processing (GO:0006364)\n",
      "\n",
      "rRNA Processing In Mitochondrion R-HSA-8868766\tMitochondrial RNA Processing (GO:0000963)\n",
      "\n",
      "rRNA Processing In Nucleus And Cytosol R-HSA-8868773\trRNA Processing (GO:0006364)\n",
      "\n",
      "tRNA Aminoacylation R-HSA-379724\ttRNA Aminoacylation (GO:0043039)\n",
      "\n",
      "tRNA Modification In Mitochondrion R-HSA-6787450\tMitochondrial tRNA Modification (GO:0070900)\n",
      "\n",
      "tRNA Modification In Nucleus And Cytosol R-HSA-6782315\ttRNA Modification (GO:0006400)\n",
      "\n",
      "tRNA Processing R-HSA-72306\ttRNA Processing (GO:0008033)\n",
      "\n",
      "tRNA Processing In Mitochondrion R-HSA-6785470\tMitochondrial tRNA Processing (GO:0090646)\n",
      "\n",
      "tRNA Processing In Nucleus R-HSA-6784531\ttRNA Processing (GO:0008033)\n",
      "\n",
      "trans-Golgi Network Vesicle Budding R-HSA-199992\tVesicle Budding From Membrane (GO:0006900)\n",
      "\n",
      "635\n"
     ]
    }
   ],
   "source": [
    "iou_output = []\n",
    "for row, col in zip(f_rows, f_cols):\n",
    "    print(wiki[\"descr\"][row] + \"\\t\" + go[\"descr\"][col])\n",
    "    print()\n",
    "    #print(wikipathway[\"genes\"][row])\n",
    "    #print(go[\"genes\"][col])\n",
    "print(len(f_cols))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "iou_output = []\n",
    "for row, col in zip(f_rows, f_cols):\n",
    "    #print(f\"Value {dot_products[row, col]:.2f} at index ({row}, {col})\")\n",
    "    #print(wikipathway[\"descr_cleaned\"][row], go[\"descr_cleaned\"][col])\n",
    "    #print(wikipathway[\"genes\"][row])\n",
    "    #print(go[\"genes\"][col])\n",
    "    curated_genes = go[\"genes\"][col]\n",
    "    parsed_llm_genes = wiki[\"genes\"][row]\n",
    "    \n",
    "    llm_genes = list(set(parsed_llm_genes)) # make sure unique genes are selected\n",
    "\n",
    "\n",
    "    intersection = set(llm_genes).intersection(set(curated_genes))\n",
    "    p_val = hypergeom.sf(len(intersection)-1,\n",
    "                            19846, \n",
    "                            len(curated_genes), \n",
    "                            len(llm_genes))\n",
    "    \n",
    "    # generatio == recall \n",
    "    generatio = float(len(intersection)) / len(set(curated_genes))                                                                                                 \n",
    "    bgratio = float(len(set(llm_genes))) / 19846                                                                                                    \n",
    "                                                                                                                                                        \n",
    "    richFactor = None                                                                                                                                      \n",
    "    foldEnrich = None                                                                                                                                      \n",
    "    if len(llm_genes) > 0:   \n",
    "        # richFactor == precision                                                                                                                              \n",
    "        richFactor = float(len(intersection)) / len(set(llm_genes))                                                                                        \n",
    "        foldEnrich = generatio / bgratio                                                                                                                   \n",
    "\n",
    "    x = {\n",
    "        \"wiki\": wiki[\"descr_cleaned\"][row],\n",
    "        \"go\": go[\"descr_cleaned\"][col],\n",
    "        'ncurated': len(curated_genes),\n",
    "        'nllm': len(llm_genes),\n",
    "        'ninter': len(intersection),\n",
    "        'generatio': generatio,\n",
    "        'bgratio': bgratio,\n",
    "        'richFactor': richFactor,\n",
    "        'foldEnrich': foldEnrich,\n",
    "        'p_val': p_val\n",
    "    }\n",
    "    iou_output.append(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(635, 11)\n",
      "0.7133858267716535\n"
     ]
    }
   ],
   "source": [
    "df = pd.DataFrame(iou_output)\n",
    "_, pvals_corrected, _, _ = multipletests(df['p_val'], method='bonferroni')\n",
    "df['p_val_adj'] = pvals_corrected\n",
    "print(df.shape)\n",
    "print(df[(df['p_val_adj'] < 0.01) ].shape[0] / df.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.31028956179170936"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# Filter the dataframe\n",
    "filtered_df = df[df['p_val_adj'] < 0.01]\n",
    "#filtered_df.generatio.mean()\n",
    "filtered_df.richFactor.mean()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
