{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2c98f95b-0e84-4d53-8f8b-22c5268ffde0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import openai\n",
    "from pathlib import Path\n",
    "import json\n",
    "import llm2geneset\n",
    "import pandas as pd\n",
    "import random\n",
    "from itertools import combinations\n",
    "from rouge_score import rouge_scorer\n",
    "import numpy as np\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "\n",
    "aclient = openai.AsyncClient()\n",
    "client = openai.Client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "58bc8391-44b8-415e-87a0-869c8534fba5",
   "metadata": {},
   "outputs": [],
   "source": [
    "scorer = rouge_scorer.RougeScorer(['rouge1','rouge2'], use_stemmer=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "57e47773-bbac-4819-9857-fa2753f91297",
   "metadata": {},
   "outputs": [],
   "source": [
    "def sample_pairs(integer_list, num_pairs):\n",
    "    possible_pairs = [(a, b) for a, b in combinations(integer_list, 2)]\n",
    "    sampled_pairs = random.sample(possible_pairs, min(num_pairs, len(possible_pairs)))\n",
    "    return sampled_pairs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1e4b30ce-ec94-4cfe-ba54-e6df772d8bee",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_names = [\n",
    "    \"gsai_vs_llm2geneset_outputs_KEGG_2021_Human.tsv\",\n",
    "    \"gsai_vs_llm2geneset_outputs_Reactome_2022.tsv\",\n",
    "    \"gsai_vs_llm2geneset_outputs_WikiPathway_2023_Human.tsv\"\n",
    "]\n",
    "\n",
    "dfs = [pd.read_csv(\"outputs/\" + file, sep='\\t') for file in file_names]\n",
    "combined = pd.concat(dfs, ignore_index=True)\n",
    "gsai_out = combined[combined[\"method\"] == \"GSAI\"]\n",
    "llm2geneset_out = combined[combined[\"method\"] == \"llm2geneset\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c26fcddb-e45f-459c-994c-cfe973d320b0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gpt-4o-mini-2024-07-18\n",
      "KEGG_2021_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|██████████████████████████████████████████████████████████████████████████████████████████▋                                   | 36/50 [00:08<00:01,  8.16it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "conf is none\n",
      "Write a critical analysis of the biological processes performed by this system of interacting proteins.\n",
      "\n",
      "Base your analysis on prior knowledge available in your training data.  After completing your analysis, propose a brief and\n",
      "detailed name for the most prominent biological process performed by the system.\n",
      "\n",
      "After completing your analysis, please also assign a confidence score to the process name you selected.  This score should\n",
      "follow the name in parentheses and range from 0.00 to 1.00. A score of 0.00 indicates the lowest confidence, while 1.00 reflects\n",
      "the highest confidence. This score helps gauge how accurately the chosen name represents the functions and activities within\n",
      "the system of interacting proteins. When determining your score, consider the proportion of genes in the protein system that\n",
      "participate in the identified biological process. For instance, if you select \"Ribosome biogenesis\" as the process name but\n",
      "only a few genes in the system contribute to this process, the score should be lower compared to a scenario where a majority\n",
      "of the genes are involved in \"Ribosome biogenesis\".\n",
      "\n",
      "Put your chosen name at the top of the analysis as 'Name: <name>'.\n",
      "Put your confidence in a new line after the name as ‘LLM self-assessed confidence: <confidence score>’\n",
      "\n",
      "Be concise, do not use unnecessary words.\n",
      "Be factual, do not editorialize.\n",
      "Be specific, avoid overly general statements such as 'the proteins are involved in various cellular processes'.\n",
      "Avoid listing facts about individual proteins. Instead, try to group proteins with similar functions and discuss their interplay,\n",
      "synergistic or antagonistic effects and functional integration within the system. Also avoid choosing generic process names such\n",
      "as 'Cellular Signaling and Regulation'.  If you cannot identify a prominent biological process for the proteins in the system, I\n",
      "want you to communicate this in you analysis and name the process: \"System of unrelated proteins\". Provide a score of 0.00 for\n",
      "a \"System of unrelated proteins\".\n",
      "\n",
      "To help you in your work, I am providing an example system of interacting proteins and the corresponding example analysis output.\n",
      "\n",
      "The example system of interacting proteins is:\n",
      "PDX1, SLC2A2, NKX6-1, GLP1, GCG.\n",
      "\n",
      "The example analysis output is:\n",
      "Name: Pancreatic development and glucose homeostasis\n",
      "LLM self-assessed confidence: 0.96\n",
      "\n",
      "1. PDX1 is a homeodomain transcription factor involved in the specification of the early pancreatic epithelium and\n",
      "its subsequent differentiation. It activates the transcription of several genes including insulin, somatostatin, glucokinase\n",
      "and glucose transporter type 2. It is essential for maintenance of the normal hormone-producing phenotype in the\n",
      "pancreatic beta-cell. In pancreatic acinar cells, forms a complex with PBX1b and MEIS2b and mediates the activation\n",
      "of the ELA1 enhancer.\n",
      "\n",
      "2. NKX6-1 is also a transcription factor involved in the development of pancreatic beta-cells during the secondary transition.\n",
      "Together with NKX2-2 and IRX3, controls the generation of motor neurons in the neural tube and belongs to the neural progenitor\n",
      "factors induced by Sonic Hedgehog (SHH) signals.\n",
      "\n",
      "3.GCG and GLP1, respectively glucagon and glucagon-like peptide 1, are involved in glucose metabolism and homeostasis.\n",
      "GCG raises blood glucose levels by promoting gluconeogenesis and is the counter regulatory hormone of Insulin.\n",
      "GLP1 is a potent stimulator of Glucose-Induced Insulin Secretion (GSIS). Plays roles in gastric motility and\n",
      "suppresses blood glucagon levels. Promotes growth of the intestinal epithelium and pancreatic islet mass both by islet\n",
      "neogenesis and islet cell proliferation.\n",
      "\n",
      "4. SLC2A2, also known as GLUT2, is a facilitative hexose transporter. In hepatocytes, it mediates bi-directional\n",
      "transport of glucose accross the plasma membranes, while in the pancreatic beta-cell, it is the main transporter responsible\n",
      "for glucose uptake and part of the cell's glucose-sensing mechanism. It is involved in glucose transport in the small intestine\n",
      "and kidney too.\n",
      "\n",
      "To summarize, the genes in this set are involved in the specification, differentiation, growth and functionality of the pancreas,\n",
      "with a particular emphasis on the pancreatic beta-cell. Particularly, the architecture of the pancreatic islet ensures proper\n",
      "glucose sensing and homeostasis via a number of different hormones and receptors that can elicit both synergistic and antagonistic\n",
      "effects in the pancreas itself and other peripheral tissues.\n",
      "\n",
      "\n",
      "Here are the interacting proteins:\n",
      "\n",
      "Proteins: SLC32A1, GLS2, GPHN, HKDC1, GABRA2, GNAI3, ADCY1, NSF, SLC38A1, CMAS, GABRA6, GABRE, TRAK2, GNE, GFPT1, CACNA1B, UAP1, CYB5RL, KCNJ6, GNG13, GNB5, ADCY4, UGDH, GMPPA, GABARAPL1, CHIA, ADCY3, GABRA1, NAGK, SLC38A2, CYB5R3, GABRA5, HEXB, PGM2, GFUS, GABARAPL2, GLUL, PRKACA, ABAT, CYB5R2, GNPNAT1, UAP1L1, PGM3, FCSK, UXS1, PRKCG, SLC6A12, GNB2, ADCY6, HK1, GNAO1, ADCY8, PRKACB, GAD2, GABRQ, GNG3, GNB4, AMDHD2, ADCY2, GMPPB, GNG2, GALE, CACNA1A, HK3, SLC6A11, CACNA1D, GNGT1, SLC12A5, GABRG2, GNB3, PMM2, GNAI2, PRKCB, GNGT2, GNG4, GPI, PMM1, CACNA1S, GABRG1, CYB5R4, GNG8, FPGT, PRKCA, PRKACG, RENBP, GABRR2, GALK1, GMDS, GABRP, CYB5R1, GABRB1, GLS, GABRA4, GAD1, CHIT1, GNG7, GABRR3, GNPDA1, GABRB2, NANP, SRC, GNG11, GNG10, ADCY9, SLC6A1, SLC6A13, MPI, GABRB3, GABRA3, NPL, CACNA1F, GNG12, GNG5, SLC38A5, CACNA1C, PLCL1, PGM1, HEXA, NANS, SLC38A3, GCK, GABRD, GABRR1, HK2, UGP2, GNPDA2, GNB1, GFPT2, GNAI1, GABBR2, GABBR1, GABARAP, HAP1, ADCY5, GALT, GABRG3, ADCY7\n",
      "\n",
      "**Name:** Neurotransmitter signaling and synaptic transmission  \n",
      "**LLM self-assessed confidence:** 0.92  \n",
      "\n",
      "The system consists of a diverse collection of proteins that are critical for neurotransmitter signaling and synaptic transmission. Many of these proteins belong to families that play key roles in synaptic functioning, including GABAA receptor subunits (GABRA1, GABRA2, GABRA3, GABRA4, GABRA5, GABRA6, GABRB1, GABRB2, GABRB3, GABRG1, GABRG2, GABRR1, GABRR2, GABRR3), which mediate inhibitory neurotransmission, and glutamate receptors. The presence of GAD1 and GAD2, the glutamic acid decarboxylases, highlights the conversion of glutamate to gamma-aminobutyric acid (GABA), the principal inhibitory neurotransmitter in the brain, underscoring the role of this system in balancing excitatory and inhibitory signals.\n",
      "\n",
      "Proteins such as GNAI3, GNAO1, and multiple GNG proteins (e.g., GNG2, GNG3, GNG4) are part of the G protein signaling cascade, which is fundamental in propagating neurotransmitter signals from receptors to intracellular effectors. Additional proteins such as ADCY family members (ADCY1, ADCY2, ADCY3, ADCY4, ADCY5, ADCY6, ADCY7, ADCY8, ADCY9) produce cyclic AMP, a second messenger that modifies the effects of neurotransmitter signaling, thereby integrating signals through various pathways.\n",
      "\n",
      "Furthermore, the involvement of ion channels, including CACNA1 and KCNJ subfamilies, suggests a significant role in establishing membrane potential changes and propagating action potentials essential for neurotransmission. The combination of various transporters, such as SLC32A1 (a vesicular GABA transporter) and SLC6A family members (including SLC6A1, SLC6A11, SLC6A12, SLC6A13), illustrates a concerted effort to maintain neurotransmitter availability and recycling within synaptic vesicles.\n",
      "\n",
      "The integration of these proteins forms a coherent network essential for achieving proper neurotransmitter release, receptor signaling, and subsequent cellular responses, facilitating effective synaptic communication. Their synergistic roles emphasize not just the individual contributions of each protein, but also the cooperative dynamics necessary for regulating neurotransmission, plasticity, and overall neuronal function.\n",
      "\n",
      "This thorough interaction underscores the significance of this protein system in the biological processes underlying synaptic communication, ultimately influencing neuronal excitability and network activity within the central nervous system.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:18<00:00,  2.76it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:18<00:00,  2.70it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactome_2022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:10<00:00,  4.59it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:28<00:00,  1.77it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WikiPathway_2023_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 34%|██████████████████████████████████████████▊                                                                                   | 17/50 [00:06<00:04,  8.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "conf is none\n",
      "Write a critical analysis of the biological processes performed by this system of interacting proteins.\n",
      "\n",
      "Base your analysis on prior knowledge available in your training data.  After completing your analysis, propose a brief and\n",
      "detailed name for the most prominent biological process performed by the system.\n",
      "\n",
      "After completing your analysis, please also assign a confidence score to the process name you selected.  This score should\n",
      "follow the name in parentheses and range from 0.00 to 1.00. A score of 0.00 indicates the lowest confidence, while 1.00 reflects\n",
      "the highest confidence. This score helps gauge how accurately the chosen name represents the functions and activities within\n",
      "the system of interacting proteins. When determining your score, consider the proportion of genes in the protein system that\n",
      "participate in the identified biological process. For instance, if you select \"Ribosome biogenesis\" as the process name but\n",
      "only a few genes in the system contribute to this process, the score should be lower compared to a scenario where a majority\n",
      "of the genes are involved in \"Ribosome biogenesis\".\n",
      "\n",
      "Put your chosen name at the top of the analysis as 'Name: <name>'.\n",
      "Put your confidence in a new line after the name as ‘LLM self-assessed confidence: <confidence score>’\n",
      "\n",
      "Be concise, do not use unnecessary words.\n",
      "Be factual, do not editorialize.\n",
      "Be specific, avoid overly general statements such as 'the proteins are involved in various cellular processes'.\n",
      "Avoid listing facts about individual proteins. Instead, try to group proteins with similar functions and discuss their interplay,\n",
      "synergistic or antagonistic effects and functional integration within the system. Also avoid choosing generic process names such\n",
      "as 'Cellular Signaling and Regulation'.  If you cannot identify a prominent biological process for the proteins in the system, I\n",
      "want you to communicate this in you analysis and name the process: \"System of unrelated proteins\". Provide a score of 0.00 for\n",
      "a \"System of unrelated proteins\".\n",
      "\n",
      "To help you in your work, I am providing an example system of interacting proteins and the corresponding example analysis output.\n",
      "\n",
      "The example system of interacting proteins is:\n",
      "PDX1, SLC2A2, NKX6-1, GLP1, GCG.\n",
      "\n",
      "The example analysis output is:\n",
      "Name: Pancreatic development and glucose homeostasis\n",
      "LLM self-assessed confidence: 0.96\n",
      "\n",
      "1. PDX1 is a homeodomain transcription factor involved in the specification of the early pancreatic epithelium and\n",
      "its subsequent differentiation. It activates the transcription of several genes including insulin, somatostatin, glucokinase\n",
      "and glucose transporter type 2. It is essential for maintenance of the normal hormone-producing phenotype in the\n",
      "pancreatic beta-cell. In pancreatic acinar cells, forms a complex with PBX1b and MEIS2b and mediates the activation\n",
      "of the ELA1 enhancer.\n",
      "\n",
      "2. NKX6-1 is also a transcription factor involved in the development of pancreatic beta-cells during the secondary transition.\n",
      "Together with NKX2-2 and IRX3, controls the generation of motor neurons in the neural tube and belongs to the neural progenitor\n",
      "factors induced by Sonic Hedgehog (SHH) signals.\n",
      "\n",
      "3.GCG and GLP1, respectively glucagon and glucagon-like peptide 1, are involved in glucose metabolism and homeostasis.\n",
      "GCG raises blood glucose levels by promoting gluconeogenesis and is the counter regulatory hormone of Insulin.\n",
      "GLP1 is a potent stimulator of Glucose-Induced Insulin Secretion (GSIS). Plays roles in gastric motility and\n",
      "suppresses blood glucagon levels. Promotes growth of the intestinal epithelium and pancreatic islet mass both by islet\n",
      "neogenesis and islet cell proliferation.\n",
      "\n",
      "4. SLC2A2, also known as GLUT2, is a facilitative hexose transporter. In hepatocytes, it mediates bi-directional\n",
      "transport of glucose accross the plasma membranes, while in the pancreatic beta-cell, it is the main transporter responsible\n",
      "for glucose uptake and part of the cell's glucose-sensing mechanism. It is involved in glucose transport in the small intestine\n",
      "and kidney too.\n",
      "\n",
      "To summarize, the genes in this set are involved in the specification, differentiation, growth and functionality of the pancreas,\n",
      "with a particular emphasis on the pancreatic beta-cell. Particularly, the architecture of the pancreatic islet ensures proper\n",
      "glucose sensing and homeostasis via a number of different hormones and receptors that can elicit both synergistic and antagonistic\n",
      "effects in the pancreas itself and other peripheral tissues.\n",
      "\n",
      "\n",
      "Here are the interacting proteins:\n",
      "\n",
      "Proteins: PRKCZ, PRKX, CCR3, VAV1, NFKB1, NDUFAF1, NDUFAF2, KRAS, GRK1, ACAD9, AKT1, GNG8, GNAI2, SHC2, ROCK2, AKT3, PIK3R5, PRKCD, RAP1A, PAK1, PLCB4, NDUFB3, DMAC1, GNGT1, ARRB2, NDUFB6, STAT2, ND5, NDUFS1, CCL28, WASL, TMEM126B, CCR6, ADCY2, NDUFA10, PIK3R2, CCL11, NDUFC2, PLCB1, NDUFA7, DOCK2, CRK, GNG7, RAP1B, NDUFS5, ND2, PRKACG, PIK3CG, NDUFB5, CCL4, TIAM2, GRK5, CCR1, LYN, NDUFV1, CCL24, GNB4, JAK2, NDUFAF4, RELA, NDUFB4, CXCL3, NDUFA5, NCF1, GNG4, NDUFA3, CCL7, VAV2, PTK2, NDUFB10, SOS1, CXCL9, GNG12, CXCL10, GNG2, GNG13, PIK3R1, GRK2, NFKBIB, ADCY1, FOXO3, PRKACB, PF4, SHC1, GRB2, PLCB2, VAV3, BRAF, GNB1, CCL20, CCL15, CHUK, CCL19, NDUFA6, NDUFA1, CX3CL1, CCL3, TMEM70, ADCY3, CX3CR1, JAK3, PIK3CA, NDUFA2, PARD3, CXCR4, NDUFAF3, CCR7, GNB5, NDUFB8, CCR8, GNG11, MAP2K1, CDC42, NDUFV3, ADCY5, ELMO1, GRK6, CXCL14, ARRB1, CXCL13, STAT5B, TMEM186, CCL1, NFKBIA, GNB2, RAF1, ND6, CCL17, PRKCB, CCR10, CCR2, CXCL11, NDUFS4, ND4L, NDUFS3, CCL5, ADCY4, CCL25, ADCY9, CSK, GNG3, NDUFB9, BCAR1, GRK4, PREX1, PXN, SHC3, HCK, GNAI1, STAT3, CXCL5, XCR1, CXCR6, ADCY8, NDUFS6, SHC4, ECSIT, FGR, CCL22, NRAS, ND1, SOS2, ADCY7, CCL26, NDUFA12, NDUFAF7, PIK3R3, CRKL, HRAS, FOXRED1, TIMMDC1, COA1, GNGT2, NUBPL, IKBKB, CXCL12, NDUFV2, CCR4, CCL27, MAPK1, CCL21, CCR9, GNAI3, TIAM1, NDUFB7, RASGRP2, ROCK1, PIK3CD, NDUFA13, XCL1, GSK3B, GNG10, GNB3, NDUFS2, WAS, AKT2, CXCL16, STAT1, ITK, PIK3CB, ND4, RAC2, IKBKG, PPBP, NDUFAB1, DMAC2, GNG5, PTK2B, PLCB3, RAC1, MAPK3, CXCR3, NDUFAF6, NDUFA8, CXCR2, CXCR5, ADCY6, NDUFB1, NDUFC1, NDUFB11, NDUFB2\n",
      "\n",
      "**Name:** Immune response and signaling modulation  \n",
      "**LLM self-assessed confidence:** 0.85  \n",
      "\n",
      "The protein system comprises a diverse array of components involved in immune signaling, chemotaxis, and energy metabolism. Key players include the G-protein coupled receptors (GPCRs), such as CCR3, CCR5, CCR6, and CCR9, which mediate immune cell migration and activation in response to specific chemokines (e.g., CCL3, CCL4, CCL11). These receptors interact dynamically with intracellular signaling molecules like AKT, PLCB, and various Rho family GTPases (RAC1, CDC42), which are critical for cytoskeletal rearrangements and cell movement.\n",
      "\n",
      "At the nexus of immune signaling, the NF-kB pathway is prominently represented by NFKB1 and its regulators, facilitating cellular responses to extracellular stimuli and regulating the expression of genes involved in inflammation and survival. Furthermore, the interaction of components such as GRK1 and GRK2 with the chemokine receptors suggests a feedback mechanism that modulates receptor sensitivity and desensitization, thus fine-tuning immune responses.\n",
      "\n",
      "In terms of energy metabolism, the set includes multiple proteins related to mitochondrial function (NDUFA subunits, ACAD9), indicating a link between cellular metabolism and immune function. The presence of proteins such as the PRK family kinases suggests additional layers of signaling modulation that connect metabolic states with immune signaling networks.\n",
      "\n",
      "Thus, the interplay between chemotactic signaling, the adaptive immune response, and cellular metabolism provides a comprehensive view of the biological processes this protein system contributes to. These interactions not only support immune cell mobilization but also suggest that energy metabolism might be integrated with these signaling pathways to meet the bioenergetic demands of activated immune cells.\n",
      "\n",
      "Overall, while there is broad participation in immune signaling, metabolic processes also play a vital role, leading to a confident conclusion centered on immune response modulation and signaling dynamics. The score reflects a strong involvement of the proteins in these processes, albeit with some components primarily focused on metabolic functions.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 92%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉          | 46/50 [00:10<00:00,  8.04it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "conf is none\n",
      "Write a critical analysis of the biological processes performed by this system of interacting proteins.\n",
      "\n",
      "Base your analysis on prior knowledge available in your training data.  After completing your analysis, propose a brief and\n",
      "detailed name for the most prominent biological process performed by the system.\n",
      "\n",
      "After completing your analysis, please also assign a confidence score to the process name you selected.  This score should\n",
      "follow the name in parentheses and range from 0.00 to 1.00. A score of 0.00 indicates the lowest confidence, while 1.00 reflects\n",
      "the highest confidence. This score helps gauge how accurately the chosen name represents the functions and activities within\n",
      "the system of interacting proteins. When determining your score, consider the proportion of genes in the protein system that\n",
      "participate in the identified biological process. For instance, if you select \"Ribosome biogenesis\" as the process name but\n",
      "only a few genes in the system contribute to this process, the score should be lower compared to a scenario where a majority\n",
      "of the genes are involved in \"Ribosome biogenesis\".\n",
      "\n",
      "Put your chosen name at the top of the analysis as 'Name: <name>'.\n",
      "Put your confidence in a new line after the name as ‘LLM self-assessed confidence: <confidence score>’\n",
      "\n",
      "Be concise, do not use unnecessary words.\n",
      "Be factual, do not editorialize.\n",
      "Be specific, avoid overly general statements such as 'the proteins are involved in various cellular processes'.\n",
      "Avoid listing facts about individual proteins. Instead, try to group proteins with similar functions and discuss their interplay,\n",
      "synergistic or antagonistic effects and functional integration within the system. Also avoid choosing generic process names such\n",
      "as 'Cellular Signaling and Regulation'.  If you cannot identify a prominent biological process for the proteins in the system, I\n",
      "want you to communicate this in you analysis and name the process: \"System of unrelated proteins\". Provide a score of 0.00 for\n",
      "a \"System of unrelated proteins\".\n",
      "\n",
      "To help you in your work, I am providing an example system of interacting proteins and the corresponding example analysis output.\n",
      "\n",
      "The example system of interacting proteins is:\n",
      "PDX1, SLC2A2, NKX6-1, GLP1, GCG.\n",
      "\n",
      "The example analysis output is:\n",
      "Name: Pancreatic development and glucose homeostasis\n",
      "LLM self-assessed confidence: 0.96\n",
      "\n",
      "1. PDX1 is a homeodomain transcription factor involved in the specification of the early pancreatic epithelium and\n",
      "its subsequent differentiation. It activates the transcription of several genes including insulin, somatostatin, glucokinase\n",
      "and glucose transporter type 2. It is essential for maintenance of the normal hormone-producing phenotype in the\n",
      "pancreatic beta-cell. In pancreatic acinar cells, forms a complex with PBX1b and MEIS2b and mediates the activation\n",
      "of the ELA1 enhancer.\n",
      "\n",
      "2. NKX6-1 is also a transcription factor involved in the development of pancreatic beta-cells during the secondary transition.\n",
      "Together with NKX2-2 and IRX3, controls the generation of motor neurons in the neural tube and belongs to the neural progenitor\n",
      "factors induced by Sonic Hedgehog (SHH) signals.\n",
      "\n",
      "3.GCG and GLP1, respectively glucagon and glucagon-like peptide 1, are involved in glucose metabolism and homeostasis.\n",
      "GCG raises blood glucose levels by promoting gluconeogenesis and is the counter regulatory hormone of Insulin.\n",
      "GLP1 is a potent stimulator of Glucose-Induced Insulin Secretion (GSIS). Plays roles in gastric motility and\n",
      "suppresses blood glucagon levels. Promotes growth of the intestinal epithelium and pancreatic islet mass both by islet\n",
      "neogenesis and islet cell proliferation.\n",
      "\n",
      "4. SLC2A2, also known as GLUT2, is a facilitative hexose transporter. In hepatocytes, it mediates bi-directional\n",
      "transport of glucose accross the plasma membranes, while in the pancreatic beta-cell, it is the main transporter responsible\n",
      "for glucose uptake and part of the cell's glucose-sensing mechanism. It is involved in glucose transport in the small intestine\n",
      "and kidney too.\n",
      "\n",
      "To summarize, the genes in this set are involved in the specification, differentiation, growth and functionality of the pancreas,\n",
      "with a particular emphasis on the pancreatic beta-cell. Particularly, the architecture of the pancreatic islet ensures proper\n",
      "glucose sensing and homeostasis via a number of different hormones and receptors that can elicit both synergistic and antagonistic\n",
      "effects in the pancreas itself and other peripheral tissues.\n",
      "\n",
      "\n",
      "Here are the interacting proteins:\n",
      "\n",
      "Proteins: C1QB, PSMD7, PSMB6, C9, PSMD13, H2AC4, PSME3, MASP2, C4A, H2AX, PSMA2, PSMD6, HLA-E, C6, UBA7, C4B, C3, CFB, UCHL1, C8B, PSMD12, C8G, H2AZ1, UBC, UBE2D3, PSMB3, C2, UBE2B, CFP, PSMC1, HLA-A, LOC110384692, PSMC4, C5, PSMB7, C8A, C1R, PSMA4, PSMB10, PSMB8, PSMD5, C7, PSMD11, MASP1, NEDD4, PSMA7, PSMD4, C1QC, UBB, PSMB5, PSMD9, PSMC6, C1S, PSME2, C1QA, PSMA5, CFD, PSMA1, PSMD1, PSMC5, HLA-G, PSMC2, UBA1, UCHL3, RPN2, HLA-C, PSMB4, RPN1, UBE2D2, CD55, PSMD10, PSMB1, HLA-B, PSMA6, UBE2D1, HLA-F, PSMB9, PSMD8, PSMD2, PSMA3, IFNG, PSMC3, PSMB2, PSMD3, PSME1\n",
      "\n",
      "**Name:** Complement activation and protein degradation  \n",
      "**LLM self-assessed confidence:** 0.95\n",
      "\n",
      "The proteins in this system primarily participate in two interrelated biological processes: complement activation and the ubiquitin-proteasome system (UPS) for protein degradation.\n",
      "\n",
      "1. **Complement Activation:** A significant portion of this protein set includes complement components (C1Q, C2, C3, C4A, C4B, C5, C6, C7, C8A, C8B, C9) and regulators (C1R, C1S) that play vital roles in the complement system - a crucial part of the innate immune response. These proteins are responsible for opsonization of pathogens, recruitment of inflammatory cells, and the lysis of target cells. Their concerted action enhances the ability of antibodies and phagocytic cells to clear microbes and damaged cells.\n",
      "\n",
      "2. **Ubiquitin-Proteasome System (UPS):** Another substantial group of proteins within this system consists of proteasome subunits (PSMA1-7, PSMB1-10, PSMC1-6) and ubiquitin-conjugating enzymes (UBE2B, UBE2D1-3). This machinery is essential for the selective degradation of misfolded proteins and regulation of various cellular processes through protein turnover. Ubiquitination tags proteins for recognition and destruction by the proteasome, facilitating the removal of damaged or unneeded proteins, thereby maintaining cellular homeostasis.\n",
      "\n",
      "3. **Synergistic Interplay:** The complement components and the UPS can interact functionally. For instance, proper complement activation can influence the subsequent clearance of apoptotic cells and debris, which is subsequently regulated through the UPS to ensure no necrotic or dysfunctional proteins are allowed to accumulate. Furthermore, some components from both processes might interact at regulatory levels, offering a complex network of feedback mechanisms necessary for effective immune response and protein regulation.\n",
      "\n",
      "Overall, the genes present here exhibit a strong and coherent focus on immune response and protein quality control through degradation, indicating their essential roles in both pathogen clearance and maintaining cellular integrity. \n",
      "\n",
      "This analysis reflects that the majority of the proteins are involved in these interconnected biological processes, leading to a high confidence in the proposed name.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:14<00:00,  3.35it/s]\n",
      "  0%|                                                                                                                                       | 0/50 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "List 5 biological pathways, biological processes, or cellular components that contain the following genes \"\"\"RAD9A,GABBR2,CCNB3,RB1,BRCA1,RRM2B,ABAT,ALDH9A1,CDK6,CDK1,GABRB2,CCNB2,RPA2,RAD51,RAD52,GADD45G,CDK4,E2F1,CASP9,GABRD,CDC25A,GAD1,CHEK1,ATR,GABBR1,CCNE1,CCND2,GABRG3,CHEK2,AP2B1,TLK2,GABRG2,CDKN1A,MRE11,H2AX,AP2S1,BID,GABRA2,BBC3,CDK2,GABRA6,TNFRSF10B,FANCD2,AP2A1,CDC25C,RAD50,GABRA4,PRKDC,CDKN1B,CDK5,HUS1B,GADD45B,GABRP,FAS,MDM2,CASP8,SMC1A,SFN,SESN1,GABRE,ATM,AP2A2,GADD45A,GABRA5,RFC1,DDB2,MYC,PML,CYCS,PIDD1,TP53,AKT1,CREB1,GABRQ,CASP3,SLC6A1,GABRA3,GABRG1,AP2M1,GABRB3,RAD17,GAD2,APAF1,TP53AIP1,ABL1,CCND1,TLK1,ATRIP,SLC6A11,RAD1,CCND3,BAX,CCNB1,CCNE2,GPHN,NBN,GABRA1,GABRB1,SLC32A1,PMAIP1\"\"\" with high confidence. Be as specific as possible. List non-overlapping pathways, processes, or components. Do not include the gene names in the outputs. Use the following JSON schema:\n",
      "```json\n",
      "{\n",
      "    \"type\": \"array\",\n",
      "    \"items\": {\n",
      "        \"type\": \"object\",\n",
      "        \"properties\": {\n",
      "            \"p\": {\n",
      "                \"type\": \"string\",\n",
      "            },\n",
      "        },\n",
      "        \"required\": [\"p\"]\n",
      "    }\n",
      "}\n",
      "```\n",
      "Example output will look like the following:\n",
      "```json\n",
      "[{\"p\":\"BP or Pathway 1\"},\n",
      " {\"p\":\"BP or Pathway 2\"},\n",
      " {\"p\":\"BP or Pathway 3\"},\n",
      " {\"p\":\"BP or Pathway 4\"}\n",
      "```\n",
      "The element `p` designates a pathway, biological process or cellular component. Place the output in a JSON code block. Do not add any comments in the JSON code block.\n",
      "\n",
      "No code blocks found\n",
      "```json\n",
      "[{\"p\":\"Cell Cycle Regulation\"},\n",
      " {\"p\":\"DNA Repair Pathways\"},\n",
      " {\"p\":\"Apoptosis\"},\n",
      " {\"p\":\"Signal Transduction Pathways\"},\n",
      " {\"p\":\"Gamma-Aminobutyric Acid (GABA) Signaling\"}\n",
      "]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:17<00:00,  2.79it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gpt-3.5-turbo-0125\n",
      "KEGG_2021_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:15<00:00,  3.31it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:16<00:00,  3.00it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactome_2022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:08<00:00,  5.94it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:58<00:00,  1.16s/it]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WikiPathway_2023_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:08<00:00,  6.07it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:24<00:00,  2.07it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gpt-4o-2024-08-06\n",
      "KEGG_2021_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:19<00:00,  2.53it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:20<00:00,  2.45it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reactome_2022\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 46%|█████████████████████████████████████████████████████████▉                                                                    | 23/50 [00:11<00:03,  7.45it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "conf is none\n",
      "Write a critical analysis of the biological processes performed by this system of interacting proteins.\n",
      "\n",
      "Base your analysis on prior knowledge available in your training data.  After completing your analysis, propose a brief and\n",
      "detailed name for the most prominent biological process performed by the system.\n",
      "\n",
      "After completing your analysis, please also assign a confidence score to the process name you selected.  This score should\n",
      "follow the name in parentheses and range from 0.00 to 1.00. A score of 0.00 indicates the lowest confidence, while 1.00 reflects\n",
      "the highest confidence. This score helps gauge how accurately the chosen name represents the functions and activities within\n",
      "the system of interacting proteins. When determining your score, consider the proportion of genes in the protein system that\n",
      "participate in the identified biological process. For instance, if you select \"Ribosome biogenesis\" as the process name but\n",
      "only a few genes in the system contribute to this process, the score should be lower compared to a scenario where a majority\n",
      "of the genes are involved in \"Ribosome biogenesis\".\n",
      "\n",
      "Put your chosen name at the top of the analysis as 'Name: <name>'.\n",
      "Put your confidence in a new line after the name as ‘LLM self-assessed confidence: <confidence score>’\n",
      "\n",
      "Be concise, do not use unnecessary words.\n",
      "Be factual, do not editorialize.\n",
      "Be specific, avoid overly general statements such as 'the proteins are involved in various cellular processes'.\n",
      "Avoid listing facts about individual proteins. Instead, try to group proteins with similar functions and discuss their interplay,\n",
      "synergistic or antagonistic effects and functional integration within the system. Also avoid choosing generic process names such\n",
      "as 'Cellular Signaling and Regulation'.  If you cannot identify a prominent biological process for the proteins in the system, I\n",
      "want you to communicate this in you analysis and name the process: \"System of unrelated proteins\". Provide a score of 0.00 for\n",
      "a \"System of unrelated proteins\".\n",
      "\n",
      "To help you in your work, I am providing an example system of interacting proteins and the corresponding example analysis output.\n",
      "\n",
      "The example system of interacting proteins is:\n",
      "PDX1, SLC2A2, NKX6-1, GLP1, GCG.\n",
      "\n",
      "The example analysis output is:\n",
      "Name: Pancreatic development and glucose homeostasis\n",
      "LLM self-assessed confidence: 0.96\n",
      "\n",
      "1. PDX1 is a homeodomain transcription factor involved in the specification of the early pancreatic epithelium and\n",
      "its subsequent differentiation. It activates the transcription of several genes including insulin, somatostatin, glucokinase\n",
      "and glucose transporter type 2. It is essential for maintenance of the normal hormone-producing phenotype in the\n",
      "pancreatic beta-cell. In pancreatic acinar cells, forms a complex with PBX1b and MEIS2b and mediates the activation\n",
      "of the ELA1 enhancer.\n",
      "\n",
      "2. NKX6-1 is also a transcription factor involved in the development of pancreatic beta-cells during the secondary transition.\n",
      "Together with NKX2-2 and IRX3, controls the generation of motor neurons in the neural tube and belongs to the neural progenitor\n",
      "factors induced by Sonic Hedgehog (SHH) signals.\n",
      "\n",
      "3.GCG and GLP1, respectively glucagon and glucagon-like peptide 1, are involved in glucose metabolism and homeostasis.\n",
      "GCG raises blood glucose levels by promoting gluconeogenesis and is the counter regulatory hormone of Insulin.\n",
      "GLP1 is a potent stimulator of Glucose-Induced Insulin Secretion (GSIS). Plays roles in gastric motility and\n",
      "suppresses blood glucagon levels. Promotes growth of the intestinal epithelium and pancreatic islet mass both by islet\n",
      "neogenesis and islet cell proliferation.\n",
      "\n",
      "4. SLC2A2, also known as GLUT2, is a facilitative hexose transporter. In hepatocytes, it mediates bi-directional\n",
      "transport of glucose accross the plasma membranes, while in the pancreatic beta-cell, it is the main transporter responsible\n",
      "for glucose uptake and part of the cell's glucose-sensing mechanism. It is involved in glucose transport in the small intestine\n",
      "and kidney too.\n",
      "\n",
      "To summarize, the genes in this set are involved in the specification, differentiation, growth and functionality of the pancreas,\n",
      "with a particular emphasis on the pancreatic beta-cell. Particularly, the architecture of the pancreatic islet ensures proper\n",
      "glucose sensing and homeostasis via a number of different hormones and receptors that can elicit both synergistic and antagonistic\n",
      "effects in the pancreas itself and other peripheral tissues.\n",
      "\n",
      "\n",
      "Here are the interacting proteins:\n",
      "\n",
      "Proteins: YWHAZ, YWHAB, SIPA1, TPK1, RAP1GAP, RNASE3, PRKACG, THTPA, RAPGEF3, SLC19A2, RAP1B, PRKACB, PRKG1, RAP1GAP2, RASGRP2, SLC19A3, RAP1A, PRKCA, SLC25A19, RAPGEF4, RASGRP1\n",
      "\n",
      "**Name:** Regulation of RAP1 GTPase cycling\n",
      "\n",
      "**LLM self-assessed confidence:** 0.85\n",
      "\n",
      "The system of proteins provided is primarily centered around the regulation of RAP1 GTPase cycling, a critical process involved in intracellular signaling pathways that influence cell adhesion, proliferation, and cellular immune responses. RAP1A, RAP1B, and associated regulatory proteins such as RAP1GAP, RAP1GAP2, RAPGEF3, and RAPGEF4 form a network that controls the switch between active (GTP-bound) and inactive (GDP-bound) states of RAP1. RAPGEFs facilitate the exchange of GDP for GTP, activating RAP1, while RAP1GAP proteins enhance GTP hydrolysis, inactivating RAP1. \n",
      "\n",
      "Proteins such as RASGRP1 and RASGRP2, which also have guanine nucleotide exchange factor (GEF) activity, contribute to RAP1 activation, thereby playing a role in signal transduction. The interplay between these activators and inactivators ensures precise spatial and temporal control of RAP1 activity, which is vital for maintaining cellular functions ranging from cytoskeletal dynamics to integrin-mediated adhesion.\n",
      "\n",
      "Furthermore, YWHAZ and YWHAB, members of the 14-3-3 protein family, likely participate in modulating this regulatory network by affecting the localization and activity of various associated proteins through phosphorylation-dependent interactions.\n",
      "\n",
      "PRKACB, PRKACG, PRKCA, and PRKG1, which are serine/threonine kinases, may also interact by phosphorylating target proteins to influence RAP1 pathways, thus integrating signals from upstream signaling cascades.\n",
      "\n",
      "The presence of proteins involved in thiamine metabolism (like TPK1 and SLC19A2) and RNA degradation (such as RNASE3) suggests additional roles these proteins may play in cellular metabolism and immune defense, but these are not closely aligned with the primary function of RAP1 regulation observed in this system.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 56%|██████████████████████████████████████████████████████████████████████▌                                                       | 28/50 [00:11<00:01, 11.70it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "retrying\n",
      "conf is none\n",
      "Write a critical analysis of the biological processes performed by this system of interacting proteins.\n",
      "\n",
      "Base your analysis on prior knowledge available in your training data.  After completing your analysis, propose a brief and\n",
      "detailed name for the most prominent biological process performed by the system.\n",
      "\n",
      "After completing your analysis, please also assign a confidence score to the process name you selected.  This score should\n",
      "follow the name in parentheses and range from 0.00 to 1.00. A score of 0.00 indicates the lowest confidence, while 1.00 reflects\n",
      "the highest confidence. This score helps gauge how accurately the chosen name represents the functions and activities within\n",
      "the system of interacting proteins. When determining your score, consider the proportion of genes in the protein system that\n",
      "participate in the identified biological process. For instance, if you select \"Ribosome biogenesis\" as the process name but\n",
      "only a few genes in the system contribute to this process, the score should be lower compared to a scenario where a majority\n",
      "of the genes are involved in \"Ribosome biogenesis\".\n",
      "\n",
      "Put your chosen name at the top of the analysis as 'Name: <name>'.\n",
      "Put your confidence in a new line after the name as ‘LLM self-assessed confidence: <confidence score>’\n",
      "\n",
      "Be concise, do not use unnecessary words.\n",
      "Be factual, do not editorialize.\n",
      "Be specific, avoid overly general statements such as 'the proteins are involved in various cellular processes'.\n",
      "Avoid listing facts about individual proteins. Instead, try to group proteins with similar functions and discuss their interplay,\n",
      "synergistic or antagonistic effects and functional integration within the system. Also avoid choosing generic process names such\n",
      "as 'Cellular Signaling and Regulation'.  If you cannot identify a prominent biological process for the proteins in the system, I\n",
      "want you to communicate this in you analysis and name the process: \"System of unrelated proteins\". Provide a score of 0.00 for\n",
      "a \"System of unrelated proteins\".\n",
      "\n",
      "To help you in your work, I am providing an example system of interacting proteins and the corresponding example analysis output.\n",
      "\n",
      "The example system of interacting proteins is:\n",
      "PDX1, SLC2A2, NKX6-1, GLP1, GCG.\n",
      "\n",
      "The example analysis output is:\n",
      "Name: Pancreatic development and glucose homeostasis\n",
      "LLM self-assessed confidence: 0.96\n",
      "\n",
      "1. PDX1 is a homeodomain transcription factor involved in the specification of the early pancreatic epithelium and\n",
      "its subsequent differentiation. It activates the transcription of several genes including insulin, somatostatin, glucokinase\n",
      "and glucose transporter type 2. It is essential for maintenance of the normal hormone-producing phenotype in the\n",
      "pancreatic beta-cell. In pancreatic acinar cells, forms a complex with PBX1b and MEIS2b and mediates the activation\n",
      "of the ELA1 enhancer.\n",
      "\n",
      "2. NKX6-1 is also a transcription factor involved in the development of pancreatic beta-cells during the secondary transition.\n",
      "Together with NKX2-2 and IRX3, controls the generation of motor neurons in the neural tube and belongs to the neural progenitor\n",
      "factors induced by Sonic Hedgehog (SHH) signals.\n",
      "\n",
      "3.GCG and GLP1, respectively glucagon and glucagon-like peptide 1, are involved in glucose metabolism and homeostasis.\n",
      "GCG raises blood glucose levels by promoting gluconeogenesis and is the counter regulatory hormone of Insulin.\n",
      "GLP1 is a potent stimulator of Glucose-Induced Insulin Secretion (GSIS). Plays roles in gastric motility and\n",
      "suppresses blood glucagon levels. Promotes growth of the intestinal epithelium and pancreatic islet mass both by islet\n",
      "neogenesis and islet cell proliferation.\n",
      "\n",
      "4. SLC2A2, also known as GLUT2, is a facilitative hexose transporter. In hepatocytes, it mediates bi-directional\n",
      "transport of glucose accross the plasma membranes, while in the pancreatic beta-cell, it is the main transporter responsible\n",
      "for glucose uptake and part of the cell's glucose-sensing mechanism. It is involved in glucose transport in the small intestine\n",
      "and kidney too.\n",
      "\n",
      "To summarize, the genes in this set are involved in the specification, differentiation, growth and functionality of the pancreas,\n",
      "with a particular emphasis on the pancreatic beta-cell. Particularly, the architecture of the pancreatic islet ensures proper\n",
      "glucose sensing and homeostasis via a number of different hormones and receptors that can elicit both synergistic and antagonistic\n",
      "effects in the pancreas itself and other peripheral tissues.\n",
      "\n",
      "\n",
      "Here are the interacting proteins:\n",
      "\n",
      "Proteins: SPG7, PHB2, PMPCB, LETM1, SLC8A3, AFG3L2, SMDT1, YME1L1, SLC19A3, SLC19A2, VDAC2, MICU2, MAIP1, VDAC1, MICU1, AKAP1, STOML2, PARL, MCU, THTPA, MICU3, SLC25A19, PHB1, MCUB, PMPCA, TPK1, SLC8B1\n",
      "\n",
      "**Name:** Mitochondrial Ion Homeostasis and Quality Control  \n",
      "**LLM self-assessed confidence:** 0.93\n",
      "\n",
      "This system of interacting proteins predominantly functions in mitochondrial ion homeostasis and quality control, crucial for maintaining mitochondrial integrity and function.\n",
      "\n",
      "1. **Ion Transport and Homeostasis:** Several proteins, such as MCU (mitochondrial calcium uniporter), MICU1, MICU2, and MICU3, are directly involved in the regulation of mitochondrial calcium uptake and homeostasis, critical for cellular metabolism and signaling. SLC8A3 and SLC8B1 (solute carriers) contribute to further ion transport, highlighting an integrated system for ion regulation across mitochondrial membranes.\n",
      "\n",
      "2. **Protein Processing and Mitochondrial Dynamics:** Proteases such as PMPCB and PMPCA are involved in the processing of mitochondrial precursor proteins, ensuring functional protein formation. Proteins like PARL participate in mitochondrial membrane remodeling, also influencing apoptosis regulation.\n",
      "\n",
      "3. **Complex Assembly and Quality Control:** AFG3L2 and YME1L1 are part of the mitochondrial AAA ATPases, important for mitochondrial inner membrane protein quality control and complex assembly/maintenance. They work to degrade misfolded proteins, preventing mitochondrial dysfunction.\n",
      "\n",
      "4. **Mitochondrial Membrane Potential and Conductance:** VDAC1 and VDAC2 (voltage-dependent anion channel proteins) form a crucial part of the mitochondrial outer membrane, regulating metabolite and ion exchange, thereby affecting the mitochondrial membrane potential.\n",
      "\n",
      "5. **Miscellaneous Transport and Metabolism:** SLC25A19, SLC19A2, and SLC19A3 facilitate the transport of cofactors and metabolites, supporting mitochondrial metabolic pathways. TPK1 and THTPA are involved in thiamine metabolism, a crucial cofactor for mitochondrial enzymatic reactions.\n",
      "\n",
      "Together, these proteins create a robust network focused on integrating mitochondrial ion transport, protein quality control, and dynamic structural maintenance, supporting efficient mitochondrial function and cellular energy metabolism.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:20<00:00,  2.40it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:24<00:00,  2.07it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WikiPathway_2023_Human\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:28<00:00,  1.77it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:16<00:00,  3.11it/s]\n"
     ]
    }
   ],
   "source": [
    "lib_names = [\"KEGG_2021_Human\", \n",
    "             \"Reactome_2022\", \n",
    "             \"WikiPathway_2023_Human\"]\n",
    "models = [\"gpt-4o-mini-2024-07-18\", \"gpt-3.5-turbo-0125\", \"gpt-4o-2024-08-06\"]\n",
    "random.seed(30)\n",
    "output = []\n",
    "for model in models:\n",
    "    print(model)\n",
    "    for lib_name in lib_names:\n",
    "        print(lib_name)\n",
    "        with open(\"libs_human/\" + model + \"/\" + lib_name + \".json\") as f:\n",
    "            gen_res = json.load(f)\n",
    "\n",
    "        # Limit to gene sets easily identified alone.\n",
    "        gsai_cur = gsai_out[(gsai_out[\"model\"] == model) & (gsai_out[\"library\"] == lib_name)]\n",
    "        gsai_cur = gsai_cur[gsai_cur[\"csim\"] >= 0.7]\n",
    "        llm2geneset_cur = llm2geneset_out[(llm2geneset_out[\"model\"] == model) & (llm2geneset_out[\"library\"] == lib_name)]\n",
    "        llm2geneset_cur = llm2geneset_cur[llm2geneset_cur[\"csim\"] >= 0.7]\n",
    "        llm2geneset_gt = set(llm2geneset_cur[\"gt_name\"].to_list())\n",
    "        gsai_gt = set(gsai_cur[\"gt_name\"].to_list())\n",
    "        selected = gsai_gt.intersection(llm2geneset_gt)\n",
    "\n",
    "        # Extract selected gene sets.\n",
    "        sel_idx = [i for i, descr in enumerate(gen_res[\"descr\"]) if descr in selected]\n",
    "        descr_sel = [ gen_res[\"descr\"][i] for i in sel_idx ]\n",
    "        descr_cleaned_sel = [ gen_res[\"descr_cleaned\"][i] for i in sel_idx ]\n",
    "        curated_sel = [ gen_res[\"curated_genesets\"][i] for i in sel_idx ]\n",
    "\n",
    "        # Construct pairs.\n",
    "        pairs = sample_pairs(range(len(descr_sel)), 50)\n",
    "        test_descr = []\n",
    "        test_descr_cleaned = []\n",
    "        test_genes = []\n",
    "        for p in pairs:\n",
    "            merged_genes = curated_sel[p[0]] + curated_sel[p[1]]\n",
    "            test_genes.append(merged_genes)\n",
    "            random.shuffle(merged_genes)\n",
    "            test_descr.append(descr_sel[p[0]] + \", \" + descr_sel[p[1]])\n",
    "            test_descr_cleaned.append(descr_cleaned_sel[p[0]] + \", \" + descr_cleaned_sel[p[1]])\n",
    "\n",
    "        # Embed ground truth descriptions.\n",
    "        gt_emb = llm2geneset.get_embeddings(client, test_descr_cleaned)\n",
    "        \n",
    "        # use GSAI to generate geneset name and embed\n",
    "        gsai_res = await llm2geneset.gsai_bench(aclient, test_genes, model=model, n_retry=3)\n",
    "        gsai_name = [i['name'] for i in gsai_res]\n",
    "        gsai_name_emb = llm2geneset.get_embeddings(client, gsai_name)\n",
    "\n",
    "        # Evaluate GSAI results.\n",
    "        for idx in range(len(test_descr)):\n",
    "            scores = scorer.score(test_descr_cleaned[idx], gsai_name[idx])\n",
    "            gsai_rouge1= scores['rouge1'].recall\n",
    "            gsai_rouge2= scores['rouge2'].recall\n",
    "            gsai_csim = np.dot(gt_emb[idx],gsai_name_emb[idx])\n",
    "        \n",
    "            x={\"model\":model,\n",
    "               \"library\":lib_name,\n",
    "               \"gt_name\":test_descr[idx],\n",
    "               \"gt_name_clean\":test_descr_cleaned[idx],\n",
    "               \"name\":gsai_name[idx],\n",
    "               \"ROUGE1\":gsai_rouge1,\n",
    "               \"ROUGE2\":gsai_rouge2,   \n",
    "               \"csim\":gsai_csim,\n",
    "               \"method\":\"GSAI\"}    \n",
    "            \n",
    "            output.append(x)\n",
    "\n",
    "        # use LLM2geneset to generate geneset name and embed\n",
    "        llm2geneset_res = await llm2geneset.gs_proposal_bench(aclient, test_genes, model=model, n_retry=3)\n",
    "        llm2geneset_name = []\n",
    "        def res2name(res):\n",
    "            df = res[\"ora_results\"]\n",
    "            _, pvals_corrected, _, _ = multipletests(df[\"p_val\"], method='bonferroni')\n",
    "            df[\"p_adj\"] = pvals_corrected\n",
    "            df = df[df[\"p_adj\"] < 0.01]\n",
    "            comb_name = \", \".join(df[\"set_descr\"].to_list())\n",
    "            if comb_name == \"\":\n",
    "                comb_name = \"None found.\"\n",
    "            return comb_name\n",
    "            \n",
    "        llm2geneset_name = list(map(res2name, llm2geneset_res))\n",
    "        llm2geneset_name_emb = llm2geneset.get_embeddings(client, llm2geneset_name)\n",
    "        \n",
    "        for idx in range(len(test_descr)):\n",
    "            # evaluate llm2geneset proposed gene set name\n",
    "            scores = scorer.score(test_descr_cleaned[idx], llm2geneset_name[idx])\n",
    "            llm2geneset_rouge1= scores['rouge1'].recall\n",
    "            llm2geneset_rouge2= scores['rouge2'].recall\n",
    "            llm2geneset_csim = np.dot(gt_emb[idx],llm2geneset_name_emb[idx])\n",
    "\n",
    "            x={\"model\":model,\n",
    "               \"library\":lib_name,\n",
    "               \"gt_name\":test_descr[idx],\n",
    "               \"gt_name_clean\":test_descr_cleaned[idx],\n",
    "               \"name\":llm2geneset_name[idx],\n",
    "               \"ROUGE1\":llm2geneset_rouge1,\n",
    "               \"ROUGE2\":llm2geneset_rouge2,   \n",
    "               \"csim\":llm2geneset_csim,\n",
    "               \"method\":\"llm2geneset\"}    \n",
    "        \n",
    "            output.append(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8eb56a9a-172d-440c-8354-663c85198943",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "53bbd968-558d-45fd-ad60-88eca3aa1b89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>model</th>\n",
       "      <th>library</th>\n",
       "      <th>gt_name</th>\n",
       "      <th>gt_name_clean</th>\n",
       "      <th>name</th>\n",
       "      <th>ROUGE1</th>\n",
       "      <th>ROUGE2</th>\n",
       "      <th>csim</th>\n",
       "      <th>method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gpt-4o-mini-2024-07-18</td>\n",
       "      <td>KEGG_2021_Human</td>\n",
       "      <td>Glycolysis / Gluconeogenesis, Wnt signaling pa...</td>\n",
       "      <td>Glycolysis / Gluconeogenesis, Wnt signaling pa...</td>\n",
       "      <td>Wnt signaling pathway regulation and development</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>0.50</td>\n",
       "      <td>0.559984</td>\n",
       "      <td>GSAI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gpt-4o-mini-2024-07-18</td>\n",
       "      <td>KEGG_2021_Human</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism, U...</td>\n",
       "      <td>Amino sugar and nucleotide sugar metabolism, U...</td>\n",
       "      <td>Ubiquitin-mediated proteolysis and protein mod...</td>\n",
       "      <td>0.444444</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.624214</td>\n",
       "      <td>GSAI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gpt-4o-mini-2024-07-18</td>\n",
       "      <td>KEGG_2021_Human</td>\n",
       "      <td>Fatty acid elongation, Pentose phosphate pathway</td>\n",
       "      <td>Fatty acid elongation, Pentose phosphate pathway</td>\n",
       "      <td>Lipid and carbohydrate metabolism integration</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.476270</td>\n",
       "      <td>GSAI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>gpt-4o-mini-2024-07-18</td>\n",
       "      <td>KEGG_2021_Human</td>\n",
       "      <td>Glycerophospholipid metabolism, Phosphatidylin...</td>\n",
       "      <td>Glycerophospholipid metabolism, Phosphatidylin...</td>\n",
       "      <td>Phosphoinositide metabolism and signaling</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.790195</td>\n",
       "      <td>GSAI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gpt-4o-mini-2024-07-18</td>\n",
       "      <td>KEGG_2021_Human</td>\n",
       "      <td>B cell receptor signaling pathway, Oxidative p...</td>\n",
       "      <td>B cell receptor signaling pathway, Oxidative p...</td>\n",
       "      <td>Mitochondrial Bioenergetics and Immune Signaling</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.564130</td>\n",
       "      <td>GSAI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    model          library  \\\n",
       "0  gpt-4o-mini-2024-07-18  KEGG_2021_Human   \n",
       "1  gpt-4o-mini-2024-07-18  KEGG_2021_Human   \n",
       "2  gpt-4o-mini-2024-07-18  KEGG_2021_Human   \n",
       "3  gpt-4o-mini-2024-07-18  KEGG_2021_Human   \n",
       "4  gpt-4o-mini-2024-07-18  KEGG_2021_Human   \n",
       "\n",
       "                                             gt_name  \\\n",
       "0  Glycolysis / Gluconeogenesis, Wnt signaling pa...   \n",
       "1  Amino sugar and nucleotide sugar metabolism, U...   \n",
       "2   Fatty acid elongation, Pentose phosphate pathway   \n",
       "3  Glycerophospholipid metabolism, Phosphatidylin...   \n",
       "4  B cell receptor signaling pathway, Oxidative p...   \n",
       "\n",
       "                                       gt_name_clean  \\\n",
       "0  Glycolysis / Gluconeogenesis, Wnt signaling pa...   \n",
       "1  Amino sugar and nucleotide sugar metabolism, U...   \n",
       "2   Fatty acid elongation, Pentose phosphate pathway   \n",
       "3  Glycerophospholipid metabolism, Phosphatidylin...   \n",
       "4  B cell receptor signaling pathway, Oxidative p...   \n",
       "\n",
       "                                                name    ROUGE1  ROUGE2  \\\n",
       "0   Wnt signaling pathway regulation and development  0.600000    0.50   \n",
       "1  Ubiquitin-mediated proteolysis and protein mod...  0.444444    0.25   \n",
       "2    Lipid and carbohydrate metabolism integration    0.000000    0.00   \n",
       "3          Phosphoinositide metabolism and signaling  0.400000    0.00   \n",
       "4  Mitochondrial Bioenergetics and Immune Signaling   0.142857    0.00   \n",
       "\n",
       "       csim method  \n",
       "0  0.559984   GSAI  \n",
       "1  0.624214   GSAI  \n",
       "2  0.476270   GSAI  \n",
       "3  0.790195   GSAI  \n",
       "4  0.564130   GSAI  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "dde6bdd4-0400-4177-940d-5cac1c5b9795",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('outputs/mix_genesets_outputs.tsv', sep=\"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
