{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8155bb21-9d82-4d88-b109-e48ef7b4974e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import openai\n",
    "import llm2geneset\n",
    "import pandas as pd\n",
    "import math\n",
    "import random\n",
    "import gc\n",
    "\n",
    "aclient = openai.AsyncClient()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d816471e-1170-4fe6-9e5a-83d36682494e",
   "metadata": {},
   "outputs": [],
   "source": [
    "al002 = pd.read_csv(\"data/Stats.TECAN.AL002_vs_isotype.txt\", sep=\"\\t\")\n",
    "al002 = al002[al002[\"PercentDetectedSamples\"] > 0.9]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "611817d5-43be-460d-8839-e4c28a25a753",
   "metadata": {},
   "outputs": [],
   "source": [
    "al002_sig = al002[(al002[\"padj\"] < 0.1) ]\n",
    "genes = al002_sig[\"gene\"].to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "355fab47-9dcc-46f6-bb15-fc577c5b73d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "random.seed(138337)\n",
    "random.shuffle(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "dfb6faaa-5ebd-4a84-aa9f-78da46c21eb0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LILRB2\n",
      "MX1\n",
      "GSN\n",
      "MFGE8\n",
      "CCL3\n",
      "GPD1\n",
      "TRPV2\n",
      "DHCR24\n",
      "TNFAIP3\n",
      "MATK\n",
      "CXCL8\n",
      "CHIT1\n",
      "TPRKB\n",
      "RASSF4\n",
      "SMC6\n",
      "FOS\n",
      "RHOBTB2\n",
      "RNF19B\n",
      "FBP1\n",
      "SLC11A1\n",
      "DCSTAMP\n",
      "HSD3B7\n",
      "OAS2\n",
      "TGM2\n",
      "MCEMP1\n",
      "FAM20C\n",
      "ACP5\n",
      "CD9\n",
      "AP5B1\n"
     ]
    }
   ],
   "source": [
    "print(\"\\n\".join(genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cddc2bc8-b07e-44bf-a6bb-40b740c5fef9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:25<00:00, 25.52s/it]\n"
     ]
    }
   ],
   "source": [
    "context_str = \"\"\n",
    "llm2geneset_no_context_res = await llm2geneset.gs_proposal_bench(aclient, [genes], model=\"gpt-4o-2024-05-13\", context=context_str,\n",
    "                                                                 n_pathways=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "79bd3a25-9edd-49b6-b521-db9445fe70dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [set_descr, generatio, bgratio, richFactor, foldEnrich, p_val, p_adj, intersection, set_genes, ngenes, nset, ninter, in_toks, out_toks]\n",
       "Index: []"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = llm2geneset_no_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/no_context_AL002.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b2c2d9aa-5103-4271-ae0f-68726da8cf62",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:28<00:00, 28.74s/it]\n"
     ]
    }
   ],
   "source": [
    "context_str = \"in vitro microglia treated with a TREM2 agonist antibody\"\n",
    "llm2geneset_context_res = await llm2geneset.gs_proposal_bench(aclient, [genes], model=\"gpt-4o-2024-05-13\", \n",
    "                                                              context=context_str,\n",
    "                                                              n_pathways=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9972a22d-fd42-47c2-9c82-29d99a8d9341",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Myeloid Cell Activation</td>\n",
       "      <td>0.103448</td>\n",
       "      <td>0.001512</td>\n",
       "      <td>0.1</td>\n",
       "      <td>68.434483</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.001092</td>\n",
       "      <td>CXCL8,TNFAIP3,CCL3</td>\n",
       "      <td>CSF2,ITGB2,TLR2,JAK2,CXCR1,CXCL8,MPL,TLR4,STAT...</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>3</td>\n",
       "      <td>158</td>\n",
       "      <td>394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  set_descr  generatio   bgratio  richFactor  foldEnrich  \\\n",
       "64  Myeloid Cell Activation   0.103448  0.001512         0.1   68.434483   \n",
       "\n",
       "       p_val     p_adj        intersection  \\\n",
       "64  0.000011  0.001092  CXCL8,TNFAIP3,CCL3   \n",
       "\n",
       "                                            set_genes  ngenes  nset  ninter  \\\n",
       "64  CSF2,ITGB2,TLR2,JAK2,CXCR1,CXCL8,MPL,TLR4,STAT...      29    30       3   \n",
       "\n",
       "    in_toks  out_toks  \n",
       "64      158       394  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = llm2geneset_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/with_context_AL002.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "04ea295e-8701-4785-9ebf-1b2738dfea77",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>Rheumatoid arthritis</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.004686</td>\n",
       "      <td>0.043011</td>\n",
       "      <td>29.434186</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.003157</td>\n",
       "      <td>CXCL8,ACP5,CCL3,FOS</td>\n",
       "      <td>CSF2,HLA-DMB,HLA-DOA,TCIRG1,FOS,TGFB1,ANGPT1,T...</td>\n",
       "      <td>29</td>\n",
       "      <td>93</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>Measles</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.007004</td>\n",
       "      <td>0.028777</td>\n",
       "      <td>19.693376</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.007686</td>\n",
       "      <td>MX1,TNFAIP3,OAS2,FOS</td>\n",
       "      <td>SLAMF1,OAS1,IKBKB,IL12B,FOS,CASP3,CD3E,BAK1,NF...</td>\n",
       "      <td>29</td>\n",
       "      <td>139</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>NOD-like receptor signaling pathway</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.009120</td>\n",
       "      <td>0.022099</td>\n",
       "      <td>15.123643</td>\n",
       "      <td>0.000134</td>\n",
       "      <td>0.014265</td>\n",
       "      <td>CXCL8,TNFAIP3,OAS2,TRPV2</td>\n",
       "      <td>MAPK12,OAS1,IKBKB,CARD9,CASP12,TRPM2,ERBIN,XIA...</td>\n",
       "      <td>29</td>\n",
       "      <td>181</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>IL-17 signaling pathway</td>\n",
       "      <td>0.103448</td>\n",
       "      <td>0.004736</td>\n",
       "      <td>0.031915</td>\n",
       "      <td>21.840792</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.021238</td>\n",
       "      <td>CXCL8,TNFAIP3,FOS</td>\n",
       "      <td>USP25,CSF2,CXCL10,MAPK12,IL17B,IKBKB,FOS,CEBPB...</td>\n",
       "      <td>29</td>\n",
       "      <td>94</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Coronavirus disease</td>\n",
       "      <td>0.137931</td>\n",
       "      <td>0.011690</td>\n",
       "      <td>0.017241</td>\n",
       "      <td>11.799049</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.021238</td>\n",
       "      <td>CXCL8,MX1,OAS2,FOS</td>\n",
       "      <td>CFB,CSF2,CXCL10,RPS11,MAPK12,RPL3L,OAS1,IKBKB,...</td>\n",
       "      <td>29</td>\n",
       "      <td>232</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Chagas disease</td>\n",
       "      <td>0.103448</td>\n",
       "      <td>0.005140</td>\n",
       "      <td>0.029412</td>\n",
       "      <td>20.127789</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0.021238</td>\n",
       "      <td>CXCL8,CCL3,FOS</td>\n",
       "      <td>MAPK12,IKBKB,IL12B,FOS,TGFB1,PPP2R2A,CD3E,NFKB...</td>\n",
       "      <td>29</td>\n",
       "      <td>102</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>290</th>\n",
       "      <td>Toll-like receptor signaling pathway</td>\n",
       "      <td>0.103448</td>\n",
       "      <td>0.005240</td>\n",
       "      <td>0.028846</td>\n",
       "      <td>19.740716</td>\n",
       "      <td>0.000465</td>\n",
       "      <td>0.021238</td>\n",
       "      <td>CXCL8,CCL3,FOS</td>\n",
       "      <td>CXCL10,MAPK12,IKBKB,IL12B,FOS,CCL4L1,TLR5,NFKB...</td>\n",
       "      <td>29</td>\n",
       "      <td>104</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>Osteoclast differentiation</td>\n",
       "      <td>0.103448</td>\n",
       "      <td>0.006399</td>\n",
       "      <td>0.023622</td>\n",
       "      <td>16.165626</td>\n",
       "      <td>0.000832</td>\n",
       "      <td>0.033260</td>\n",
       "      <td>ACP5,LILRB2,FOS</td>\n",
       "      <td>MAPK12,NOX1,IKBKB,FOS,TGFB1,OSCAR,NFKBIA,MAPK8...</td>\n",
       "      <td>29</td>\n",
       "      <td>127</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                set_descr  generatio   bgratio  richFactor  \\\n",
       "251                  Rheumatoid arthritis   0.137931  0.004686    0.043011   \n",
       "161                               Measles   0.137931  0.007004    0.028777   \n",
       "173   NOD-like receptor signaling pathway   0.137931  0.009120    0.022099   \n",
       "136               IL-17 signaling pathway   0.103448  0.004736    0.031915   \n",
       "64                    Coronavirus disease   0.137931  0.011690    0.017241   \n",
       "50                         Chagas disease   0.103448  0.005140    0.029412   \n",
       "290  Toll-like receptor signaling pathway   0.103448  0.005240    0.028846   \n",
       "191            Osteoclast differentiation   0.103448  0.006399    0.023622   \n",
       "\n",
       "     foldEnrich     p_val     p_adj              intersection  \\\n",
       "251   29.434186  0.000010  0.003157       CXCL8,ACP5,CCL3,FOS   \n",
       "161   19.693376  0.000048  0.007686      MX1,TNFAIP3,OAS2,FOS   \n",
       "173   15.123643  0.000134  0.014265  CXCL8,TNFAIP3,OAS2,TRPV2   \n",
       "136   21.840792  0.000345  0.021238         CXCL8,TNFAIP3,FOS   \n",
       "64    11.799049  0.000345  0.021238        CXCL8,MX1,OAS2,FOS   \n",
       "50    20.127789  0.000439  0.021238            CXCL8,CCL3,FOS   \n",
       "290   19.740716  0.000465  0.021238            CXCL8,CCL3,FOS   \n",
       "191   16.165626  0.000832  0.033260           ACP5,LILRB2,FOS   \n",
       "\n",
       "                                             set_genes  ngenes  nset  ninter  \n",
       "251  CSF2,HLA-DMB,HLA-DOA,TCIRG1,FOS,TGFB1,ANGPT1,T...      29    93       4  \n",
       "161  SLAMF1,OAS1,IKBKB,IL12B,FOS,CASP3,CD3E,BAK1,NF...      29   139       4  \n",
       "173  MAPK12,OAS1,IKBKB,CARD9,CASP12,TRPM2,ERBIN,XIA...      29   181       4  \n",
       "136  USP25,CSF2,CXCL10,MAPK12,IL17B,IKBKB,FOS,CEBPB...      29    94       3  \n",
       "64   CFB,CSF2,CXCL10,RPS11,MAPK12,RPL3L,OAS1,IKBKB,...      29   232       4  \n",
       "50   MAPK12,IKBKB,IL12B,FOS,TGFB1,PPP2R2A,CD3E,NFKB...      29   102       3  \n",
       "290  CXCL10,MAPK12,IKBKB,IL12B,FOS,CCL4L1,TLR5,NFKB...      29   104       3  \n",
       "191  MAPK12,NOX1,IKBKB,FOS,TGFB1,OSCAR,NFKBIA,MAPK8...      29   127       3  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gmt = llm2geneset.read_gmt(\"libs_human/gmt/KEGG_2021_Human.txt\")\n",
    "set_descr = gmt[\"descr\"]\n",
    "gene_sets = gmt[\"genes\"]\n",
    "df = llm2geneset.simple_ora(genes, set_descr, gene_sets)\n",
    "df.to_csv(\"outputs/KEGG_AL002.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2bd97efe-43bb-4bd1-b54d-673f36a553df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PTX3,VWF,JUN,PRDX5,MGLL,SNAP23,CCL22,APBB1IP,LBR,KRT19,SKAP2,ARG1,INPP1,S100A11,PSG1,SH2B3,CXCL11,PLA2G2A,GSAP,XG,PDGFB,IL1RN,EGLN1,TFRC,OMG,PM20D1,CCN4,FBP1,ST6GAL1,IL5,ALPP,SMAD5,SPINK5,NADK,CCL26,LEFTY2,ITGB2,PI3,SERPINA12,LTA4H,FCAR,CIAPIN1,CA13,GH1,MPI,CST7,PSPN,NCF2,SERPINE1,DPY30,PAEP,C19orf12,CRACR2A,ANXA11,LRRC25,ADGRE5,PARP1,MASP1,CCL3,MZB1,F7,MMP3,CLC,NUB1,SLAMF7,VMO1,PPME1,IDUA,CDHR1,DDC,KLB,GOLM2,SPINK4,TRIM21,CLIP2,TNF,ABHD14B,MSLN,ENPP5,KLK4,APEX1,ERBIN,RHOC,ANXA3,CCL23,CES3,PFKFB2,RET,CCL28,BTC,PPY,NOS3,MNDA,WFIKKN1,ITGA6,LHB,SRP14,CDHR2,LEP,CLEC4A,LY75,LAT,KIR3DL1\n"
     ]
    }
   ],
   "source": [
    "#\n",
    "ppmi_olink_p196 = pd.read_csv(\"data/alt_ppmi_csf_PDvsHC.lm.20241004.tsv\", sep=\"\\t\")\n",
    "ppmi_olink_p196 = ppmi_olink_p196[(ppmi_olink_p196[\"wt_pvalue\"]<1e-2)]# & ((ppmi_olink_p196[\"l2fc\"])>0)]\n",
    "#\n",
    "# ppmi_olink_p196 = pd.read_csv(\"data/covariates_ppmi_csf_PDvsHC.lm20241008.tsv\", sep=\"\\t\")\n",
    "# print(ppmi_olink_p196.columns)\n",
    "# ppmi_olink_p196 = ppmi_olink_p196[(ppmi_olink_p196[\"wt_pvalue\"]<0.1) & (ppmi_olink_p196[\"l2fc\"]<0)]\n",
    "#\n",
    "genes = ppmi_olink_p196[\"GENE\"].to_list()\n",
    "#\n",
    "random.seed(20241009)\n",
    "random.shuffle(genes)\n",
    "#\n",
    "print(\",\".join(genes))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "70a56a3c-673b-4915-bb6d-a6fe7eb52f47",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Cytokine-cytokine receptor interaction</td>\n",
       "      <td>0.106796</td>\n",
       "      <td>0.014864</td>\n",
       "      <td>0.037288</td>\n",
       "      <td>7.184663</td>\n",
       "      <td>4.057176e-07</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>IL1RN,CCL22,CCL28,CCL26,CXCL11,GH1,LEP,TNF,CCL...</td>\n",
       "      <td>IL20,CSF2,CXCL10,IL23R,CCR3,TNFRSF18,TNFSF4,IL...</td>\n",
       "      <td>103</td>\n",
       "      <td>295</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>Viral protein interaction with cytokine and cy...</td>\n",
       "      <td>0.067961</td>\n",
       "      <td>0.005039</td>\n",
       "      <td>0.070000</td>\n",
       "      <td>13.487573</td>\n",
       "      <td>9.208993e-07</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>CCL22,CCL28,CCL26,CXCL11,TNF,CCL23,CCL3</td>\n",
       "      <td>IL20,CXCL10,CCR3,IL37,IL6R,IL34,CCL4L1,CCL1,CC...</td>\n",
       "      <td>103</td>\n",
       "      <td>100</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Complement and coagulation cascades</td>\n",
       "      <td>0.048544</td>\n",
       "      <td>0.004283</td>\n",
       "      <td>0.058824</td>\n",
       "      <td>11.334095</td>\n",
       "      <td>8.257319e-05</td>\n",
       "      <td>0.008808</td>\n",
       "      <td>ITGB2,F7,SERPINE1,VWF,MASP1</td>\n",
       "      <td>CFHR5,CFB,CD59,FGB,PLG,F9,SERPINB2,C5,C4A,C7,S...</td>\n",
       "      <td>103</td>\n",
       "      <td>85</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             set_descr  generatio   bgratio  \\\n",
       "68              Cytokine-cytokine receptor interaction   0.106796  0.014864   \n",
       "308  Viral protein interaction with cytokine and cy...   0.067961  0.005039   \n",
       "63                 Complement and coagulation cascades   0.048544  0.004283   \n",
       "\n",
       "     richFactor  foldEnrich         p_val     p_adj  \\\n",
       "68     0.037288    7.184663  4.057176e-07  0.000130   \n",
       "308    0.070000   13.487573  9.208993e-07  0.000147   \n",
       "63     0.058824   11.334095  8.257319e-05  0.008808   \n",
       "\n",
       "                                          intersection  \\\n",
       "68   IL1RN,CCL22,CCL28,CCL26,CXCL11,GH1,LEP,TNF,CCL...   \n",
       "308            CCL22,CCL28,CCL26,CXCL11,TNF,CCL23,CCL3   \n",
       "63                         ITGB2,F7,SERPINE1,VWF,MASP1   \n",
       "\n",
       "                                             set_genes  ngenes  nset  ninter  \n",
       "68   IL20,CSF2,CXCL10,IL23R,CCR3,TNFRSF18,TNFSF4,IL...     103   295      11  \n",
       "308  IL20,CXCL10,CCR3,IL37,IL6R,IL34,CCL4L1,CCL1,CC...     103   100       7  \n",
       "63   CFHR5,CFB,CD59,FGB,PLG,F9,SERPINB2,C5,C4A,C7,S...     103    85       5  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gmt = llm2geneset.read_gmt(\"libs_human/gmt/KEGG_2021_Human.txt\")\n",
    "set_descr = gmt[\"descr\"]\n",
    "gene_sets = gmt[\"genes\"]\n",
    "df = llm2geneset.simple_ora(genes, set_descr, gene_sets)\n",
    "df.to_csv(\"outputs/KEGG_ppmi-olink-p196.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ebeed202-f19c-4fb7-8ad6-74866d9e1a6c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:24<00:00, 24.49s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Chemokine-mediated signaling pathway</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>0.004132</td>\n",
       "      <td>0.073171</td>\n",
       "      <td>14.098508</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000429</td>\n",
       "      <td>CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3</td>\n",
       "      <td>CXCL10,CCR3,PDPK1,MAPK8,CCL1,CCL19,GNAI1,CCL21...</td>\n",
       "      <td>103</td>\n",
       "      <td>82</td>\n",
       "      <td>6</td>\n",
       "      <td>160</td>\n",
       "      <td>740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Annexin-mediated pathways</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.000705</td>\n",
       "      <td>0.214286</td>\n",
       "      <td>41.288488</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.002335</td>\n",
       "      <td>S100A11,ANXA11,ANXA3</td>\n",
       "      <td>ANXA1,ANXA5,ANXA11,ANXA3,ANXA8,S100A10,S100A11...</td>\n",
       "      <td>103</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>158</td>\n",
       "      <td>190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Coagulation cascade</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.001260</td>\n",
       "      <td>0.120000</td>\n",
       "      <td>23.121553</td>\n",
       "      <td>0.000292</td>\n",
       "      <td>0.009434</td>\n",
       "      <td>SERPINE1,F7,VWF</td>\n",
       "      <td>F2,F12,VWF,SERPINC1,PLG,F9,SERPINF2,F11,F10,PR...</td>\n",
       "      <td>103</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>156</td>\n",
       "      <td>320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Ret signaling pathway</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.001411</td>\n",
       "      <td>0.107143</td>\n",
       "      <td>20.644244</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>0.009964</td>\n",
       "      <td>RET,JUN,PSPN</td>\n",
       "      <td>RET,MAPK3,GFRA3,DOK5,MAPK1,STAT3,PTPN11,NRTN,C...</td>\n",
       "      <td>103</td>\n",
       "      <td>28</td>\n",
       "      <td>3</td>\n",
       "      <td>154</td>\n",
       "      <td>364</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               set_descr  generatio   bgratio  richFactor  \\\n",
       "6   Chemokine-mediated signaling pathway   0.058252  0.004132    0.073171   \n",
       "53             Annexin-mediated pathways   0.029126  0.000705    0.214286   \n",
       "58                   Coagulation cascade   0.029126  0.001260    0.120000   \n",
       "82                 Ret signaling pathway   0.029126  0.001411    0.107143   \n",
       "\n",
       "    foldEnrich     p_val     p_adj                         intersection  \\\n",
       "6    14.098508  0.000004  0.000429  CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3   \n",
       "53   41.288488  0.000048  0.002335                 S100A11,ANXA11,ANXA3   \n",
       "58   23.121553  0.000292  0.009434                      SERPINE1,F7,VWF   \n",
       "82   20.644244  0.000411  0.009964                         RET,JUN,PSPN   \n",
       "\n",
       "                                            set_genes  ngenes  nset  ninter  \\\n",
       "6   CXCL10,CCR3,PDPK1,MAPK8,CCL1,CCL19,GNAI1,CCL21...     103    82       6   \n",
       "53  ANXA1,ANXA5,ANXA11,ANXA3,ANXA8,S100A10,S100A11...     103    14       3   \n",
       "58  F2,F12,VWF,SERPINC1,PLG,F9,SERPINF2,F11,F10,PR...     103    25       3   \n",
       "82  RET,MAPK3,GFRA3,DOK5,MAPK1,STAT3,PTPN11,NRTN,C...     103    28       3   \n",
       "\n",
       "    in_toks  out_toks  \n",
       "6       160       740  \n",
       "53      158       190  \n",
       "58      156       320  \n",
       "82      154       364  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "context_str = \"\"\n",
    "llm2geneset_no_context_res = await llm2geneset.gs_proposal_bench(aclient, \n",
    "                                                                 [genes], \n",
    "                                                                 model=\"gpt-4o-2024-05-13\", \n",
    "                                                                 context=context_str,\n",
    "                                                                 n_pathways=100)\n",
    "\n",
    "df = llm2geneset_no_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/no_context_ppmi-olink-p196.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "41e25e00-0b6a-499e-bf60-47fee0de4d80",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:45<00:00, 45.90s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Chemokine-mediated signaling pathway</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>0.005895</td>\n",
       "      <td>0.051282</td>\n",
       "      <td>9.881006</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.003205</td>\n",
       "      <td>CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3</td>\n",
       "      <td>CXCR7,CXCL10,CCR3,STAT6,GNG5,CCL1,CCL19,GNAI1,...</td>\n",
       "      <td>103</td>\n",
       "      <td>117</td>\n",
       "      <td>6</td>\n",
       "      <td>160</td>\n",
       "      <td>1062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Wound healing</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.001008</td>\n",
       "      <td>0.150000</td>\n",
       "      <td>28.901942</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>0.005994</td>\n",
       "      <td>JUN,TNF,PDGFB</td>\n",
       "      <td>FGF2,ANGPT1,PDGFB,SERPINB2,ITGB1,SPARC,VEGFA,F...</td>\n",
       "      <td>103</td>\n",
       "      <td>20</td>\n",
       "      <td>3</td>\n",
       "      <td>154</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Extracellular region</td>\n",
       "      <td>0.038835</td>\n",
       "      <td>0.002771</td>\n",
       "      <td>0.072727</td>\n",
       "      <td>14.013063</td>\n",
       "      <td>0.000194</td>\n",
       "      <td>0.005994</td>\n",
       "      <td>MMP3,SERPINE1,VWF,PDGFB</td>\n",
       "      <td>PECAM1,ICAM4,FGF2,THBS3,FBN1,VWF,CDH1,TGFB1,TI...</td>\n",
       "      <td>103</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>156</td>\n",
       "      <td>715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Leukocyte chemotaxis</td>\n",
       "      <td>0.038835</td>\n",
       "      <td>0.002973</td>\n",
       "      <td>0.067797</td>\n",
       "      <td>13.063025</td>\n",
       "      <td>0.000255</td>\n",
       "      <td>0.005994</td>\n",
       "      <td>ITGB2,TNF,CCL3,CXCL11</td>\n",
       "      <td>PECAM1,MAPK3,CXCL3,CXCL10,ITGB2,TLR2,Selectin ...</td>\n",
       "      <td>103</td>\n",
       "      <td>59</td>\n",
       "      <td>4</td>\n",
       "      <td>160</td>\n",
       "      <td>546</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               set_descr  generatio   bgratio  richFactor  \\\n",
       "60  Chemokine-mediated signaling pathway   0.058252  0.005895    0.051282   \n",
       "1                          Wound healing   0.029126  0.001008    0.150000   \n",
       "87                  Extracellular region   0.038835  0.002771    0.072727   \n",
       "23                  Leukocyte chemotaxis   0.038835  0.002973    0.067797   \n",
       "\n",
       "    foldEnrich     p_val     p_adj                         intersection  \\\n",
       "60    9.881006  0.000034  0.003205  CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3   \n",
       "1    28.901942  0.000147  0.005994                        JUN,TNF,PDGFB   \n",
       "87   14.013063  0.000194  0.005994              MMP3,SERPINE1,VWF,PDGFB   \n",
       "23   13.063025  0.000255  0.005994                ITGB2,TNF,CCL3,CXCL11   \n",
       "\n",
       "                                            set_genes  ngenes  nset  ninter  \\\n",
       "60  CXCR7,CXCL10,CCR3,STAT6,GNG5,CCL1,CCL19,GNAI1,...     103   117       6   \n",
       "1   FGF2,ANGPT1,PDGFB,SERPINB2,ITGB1,SPARC,VEGFA,F...     103    20       3   \n",
       "87  PECAM1,ICAM4,FGF2,THBS3,FBN1,VWF,CDH1,TGFB1,TI...     103    55       4   \n",
       "23  PECAM1,MAPK3,CXCL3,CXCL10,ITGB2,TLR2,Selectin ...     103    59       4   \n",
       "\n",
       "    in_toks  out_toks  \n",
       "60      160      1062  \n",
       "1       154       264  \n",
       "87      156       715  \n",
       "23      160       546  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "context_str = \"human cerebrospinal fluid\"\n",
    "llm2geneset_context_res = await llm2geneset.gs_proposal_bench(aclient, [genes], model=\"gpt-4o-2024-05-13\", \n",
    "                                                              context=context_str,\n",
    "                                                              n_pathways=100)\n",
    "\n",
    "df = llm2geneset_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/with_lesscontext_ppmi-olink-p196.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "fb343d10-f23b-4b1c-a65b-8177f6283894",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:31<00:00, 31.59s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>Chemokine-mediated signaling pathway</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>0.003426</td>\n",
       "      <td>0.088235</td>\n",
       "      <td>17.001142</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.000141</td>\n",
       "      <td>CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3</td>\n",
       "      <td>CXCR7,CXCL3,CCL11,CCR3,CXCR1,CXCR3,CXCL10,CXCL...</td>\n",
       "      <td>103</td>\n",
       "      <td>68</td>\n",
       "      <td>6</td>\n",
       "      <td>160</td>\n",
       "      <td>612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Vascular permeability</td>\n",
       "      <td>0.038835</td>\n",
       "      <td>0.002469</td>\n",
       "      <td>0.081633</td>\n",
       "      <td>15.728948</td>\n",
       "      <td>0.000124</td>\n",
       "      <td>0.005932</td>\n",
       "      <td>JUN,NOS3,TNF,PDGFB</td>\n",
       "      <td>PECAM1,CDH5,MTOR,KDR,PIK3CA,FOS,PLG,ANGPT1,PDG...</td>\n",
       "      <td>103</td>\n",
       "      <td>49</td>\n",
       "      <td>4</td>\n",
       "      <td>154</td>\n",
       "      <td>632</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               set_descr  generatio   bgratio  richFactor  \\\n",
       "92  Chemokine-mediated signaling pathway   0.058252  0.003426    0.088235   \n",
       "75                 Vascular permeability   0.038835  0.002469    0.081633   \n",
       "\n",
       "    foldEnrich     p_val     p_adj                         intersection  \\\n",
       "92   17.001142  0.000001  0.000141  CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3   \n",
       "75   15.728948  0.000124  0.005932                   JUN,NOS3,TNF,PDGFB   \n",
       "\n",
       "                                            set_genes  ngenes  nset  ninter  \\\n",
       "92  CXCR7,CXCL3,CCL11,CCR3,CXCR1,CXCR3,CXCL10,CXCL...     103    68       6   \n",
       "75  PECAM1,CDH5,MTOR,KDR,PIK3CA,FOS,PLG,ANGPT1,PDG...     103    49       4   \n",
       "\n",
       "    in_toks  out_toks  \n",
       "92      160       612  \n",
       "75      154       632  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "context_str = \"Parkinson's disease\"\n",
    "llm2geneset_context_res = await llm2geneset.gs_proposal_bench(aclient, [genes], model=\"gpt-4o-2024-05-13\", \n",
    "                                                              context=context_str,\n",
    "                                                              n_pathways=100)\n",
    "\n",
    "df = llm2geneset_context_res[0][\"ora_results\"]\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8bb8df6f-ee54-43eb-b6b7-388b98143c14",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:32<00:00, 32.35s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Chemokine-Mediated Signaling Pathway</td>\n",
       "      <td>0.058252</td>\n",
       "      <td>0.003376</td>\n",
       "      <td>0.089552</td>\n",
       "      <td>17.254891</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3</td>\n",
       "      <td>CXCR7,CXCL3,CCL11,CCR3,CXCR1,CXCR3,CXCL10,CXCL...</td>\n",
       "      <td>103</td>\n",
       "      <td>67</td>\n",
       "      <td>6</td>\n",
       "      <td>168</td>\n",
       "      <td>604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Wound Healing</td>\n",
       "      <td>0.038835</td>\n",
       "      <td>0.001612</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>24.084951</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.001089</td>\n",
       "      <td>SERPINE1,NOS3,TNF,PDGFB</td>\n",
       "      <td>FGF2,CDH1,CXCL12,ANGPT1,PDGFB,CCL2,ITGB1,NOS3,...</td>\n",
       "      <td>103</td>\n",
       "      <td>32</td>\n",
       "      <td>4</td>\n",
       "      <td>154</td>\n",
       "      <td>420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Endothelial Cell Activation</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>38.535922</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.001938</td>\n",
       "      <td>TNF,NOS3,PDGFB</td>\n",
       "      <td>TEK,RELA,CXCL8,NFKB1,ANGPT2,TNF,PTGS2,VCAM1,IL...</td>\n",
       "      <td>103</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>158</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Acute Inflammatory Response</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.001260</td>\n",
       "      <td>0.120000</td>\n",
       "      <td>23.121553</td>\n",
       "      <td>0.000292</td>\n",
       "      <td>0.007075</td>\n",
       "      <td>IL1RN,SERPINE1,TNF</td>\n",
       "      <td>MAPK3,CSF2,S100A8,CXCL8,TLR4,MAPK1,STAT3,CCR5,...</td>\n",
       "      <td>103</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>158</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Coagulation</td>\n",
       "      <td>0.029126</td>\n",
       "      <td>0.001411</td>\n",
       "      <td>0.107143</td>\n",
       "      <td>20.644244</td>\n",
       "      <td>0.000411</td>\n",
       "      <td>0.007971</td>\n",
       "      <td>SERPINE1,F7,VWF</td>\n",
       "      <td>F2,APOH,F12,VWF,FGB,SERPINC1,PLG,KNG1,F9,SERPI...</td>\n",
       "      <td>103</td>\n",
       "      <td>28</td>\n",
       "      <td>3</td>\n",
       "      <td>154</td>\n",
       "      <td>355</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               set_descr  generatio   bgratio  richFactor  \\\n",
       "15  Chemokine-Mediated Signaling Pathway   0.058252  0.003376    0.089552   \n",
       "38                         Wound Healing   0.038835  0.001612    0.125000   \n",
       "12           Endothelial Cell Activation   0.029126  0.000756    0.200000   \n",
       "18           Acute Inflammatory Response   0.029126  0.001260    0.120000   \n",
       "33                           Coagulation   0.029126  0.001411    0.107143   \n",
       "\n",
       "    foldEnrich     p_val     p_adj                         intersection  \\\n",
       "15   17.254891  0.000001  0.000130  CCL22,CCL28,CCL26,CXCL11,CCL23,CCL3   \n",
       "38   24.084951  0.000022  0.001089              SERPINE1,NOS3,TNF,PDGFB   \n",
       "12   38.535922  0.000060  0.001938                       TNF,NOS3,PDGFB   \n",
       "18   23.121553  0.000292  0.007075                   IL1RN,SERPINE1,TNF   \n",
       "33   20.644244  0.000411  0.007971                      SERPINE1,F7,VWF   \n",
       "\n",
       "                                            set_genes  ngenes  nset  ninter  \\\n",
       "15  CXCR7,CXCL3,CCL11,CCR3,CXCR1,CXCR3,CXCL10,CXCL...     103    67       6   \n",
       "38  FGF2,CDH1,CXCL12,ANGPT1,PDGFB,CCL2,ITGB1,NOS3,...     103    32       4   \n",
       "12  TEK,RELA,CXCL8,NFKB1,ANGPT2,TNF,PTGS2,VCAM1,IL...     103    15       3   \n",
       "18  MAPK3,CSF2,S100A8,CXCL8,TLR4,MAPK1,STAT3,CCR5,...     103    25       3   \n",
       "33  F2,APOH,F12,VWF,FGB,SERPINC1,PLG,KNG1,F9,SERPI...     103    28       3   \n",
       "\n",
       "    in_toks  out_toks  \n",
       "15      168       604  \n",
       "38      154       420  \n",
       "12      158       196  \n",
       "18      158       327  \n",
       "33      154       355  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "# context_str = \"cerebrospinal fluid of Parkinson's disease patients based on Olink proteomics\"\n",
    "context_str = \"cerebrospinal fluid in Parkinson's disease\"\n",
    "llm2geneset_context_res = await llm2geneset.gs_proposal_bench(aclient, [genes], model=\"gpt-4o-2024-05-13\", \n",
    "                                                              context=context_str,\n",
    "                                                              n_pathways=100)\n",
    "\n",
    "df = llm2geneset_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/with_context_ppmi-olink-p196.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec145934-6bed-4c70-a92d-d47edcd95ea8",
   "metadata": {},
   "source": [
    "### Pseudobulk of scRNAseq from Substantia Nigra - a PD atlas dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "87617790-29fd-4e27-a623-aba9e45c920d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#\n",
    "mgl_scpdsn_df = pd.read_csv(\"data/Stats.scRNA_pseudobulk_microglia.PD_vs_Healthy.txt\", sep=\"\\t\")\n",
    "# Filtering for significants DEGs post multi-testing\n",
    "mgl_scpdsn_sig = mgl_scpdsn_df[(mgl_scpdsn_df[\"padj\"] < 0.05)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "b2e4e61d-9d92-4e72-91b4-5f3818ea16a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NEFL,KIF3A,GNG3,ZFP36L1,GABRG2,TAGLN3,FTL,PCBD2,NFX1,CKB,FGFR1OP2,PRKCQ,PKIB,EXOC5,DHTKD1,SULT4A1,ARHGDIG,TYW1,CLEC17A,ELAVL2,SORT1,THAP2,ZNF714,AL139220.2,LRP10,TPT1,LEPROTL1,ITGB2-AS1,SULT1A1,ALDH1A1,ZC3H14,KLHL1,ZNF839,CADM3,SESN1,BNIP3L,CLEC1A,PPA2,CDKL1,LSMEM1,MARCO,FXYD5,CD247,LINC00847,NUPR1,TTN,FGF12,MTPN,COLEC12,HP1BP3,C9orf153,EMB,HDGF,RALGAPA2,DOK6,RPL17,SLC6A3,CHPT1,LINC02397,HIST2H2BE,DAPK2,BMI1,LINC00862,LMO3,UNC13C,SYCP2,ASMTL-AS1,KLHL21,AKAP12,SERPINB9,BIN2,LMF1,MITD1,MIA2,MID1IP1,TERF2IP,GRIA3,WASF3,VAV3,LYG1,ANAPC16,TSEN54,DOC2A,CDKN2D,DPH6,NFATC2IP,ARHGEF9,PSMD9,ATP6V0E2,SLC18A2,ZNF25,MAGOH,PPP3CB,CRYGS,TOR1AIP2,RETREG1,RAB6A,PAQR6,CSF3R,LUCAT1,CIRBP,PPID,MSH3,KCNQ3,CLDN12,ZBTB44,RIMS1,ACOT7,AC083949.1,INO80,DCTN4,BIN1,HIST1H2BG,L2HGDH,DTWD2,BEX1,ENOSF1,RELA,TMEM170A,PAPOLA,HTT,ZNF706,NHSL1,CARNS1,RSBN1,IER2,LINC01877,EMP2,DCUN1D4,ARG2,FSCN1,NAIP,AC233296.1,SV2B,TANC2,KIF21A,F13A1,PEMT,CCDC26,ANKRD13A,ARSG,AATK,MICU2,RSPO2,TAZ,KLHDC4,SYT1,KCNB1,SNX22,PTPN12,HIVEP3,ITGB1,SLC38A2,TIMP3,MFGE8,LINC01102,SNX27,OPALIN,CRYZ,PGAP2,HECW1,ARSI,GSG1L,PTS,GIPC1,AC090518.1,BTG2,SLC9B1,PRRG4,CX3CR1,SNRNP48,MTOR,MAP1A,ZBTB7C,ABL1,TUBB2A,MITF,CCDC170,SCG3,TP53TG5,IL21R,CHST10,RABEP2,SEM1,MYO15B,ZC3H12D,ZNF24,TRIM28,EVA1B,ZBTB37,AL035681.1,CALCOCO2,PIK3CA,VEZF1,AC007262.2,RIC3,PBX1,POMT1,PIN1,TFAM,PHF10,VKORC1L1,RBM14,C19orf12,CCDC126,VSNL1,WDHD1,PLS1,FOS,CD93,CH25H,TRPM7,WWP2,JADE2,ALOX15B,HIST1H4H,NIPSNAP1,CYP1B1-AS1,TBC1D16,CDC42EP3,CCDC93,PID1,HNRNPH3,CYTIP,COX16,PRKCA,AFMID,RABEPK,MARCH2,ASH2L,SFXN4,NHLRC3,SEMA3E,THRB,FANCM,ID2,NEK10,MDM4,CHRNA4,C14orf119,FAM126A,SCG5,CCDC40,KCNJ6,NUP93,RMND5B,PPP1R12A,LEMD3,FAM168A,FNBP4,EMSY,HTATIP2,CXCR4,SNX16,ICA1L,NDRG4,PRDM5,BSN,CTBS,ARHGAP4,C2orf69,CCNG2,SBF2-AS1,MAP1LC3A,MSH4,SNCB,COG7,CADPS,BRCA1,SPDYA,MNAT1,FGF13,IL27RA,DDAH1,ZMIZ2,WWTR1,PTPRG,TUBA4A,AC116366.3,OLFM1,STX8,HGSNAT,MAIP1,BEND6,NAA25,LIMA1,KCND3,RBM15,SLC25A12,CEND1,ITGAM,CFAP161,CCDC30,SAP30,IRF2BP2,REEP1,SV2C,RAD21,RGS16,RNF219-AS1,DMD,DHX57,CLSTN2,VNN2,REPS1,GSTA4,CADM1,CCDC117,PCYOX1L,ABCA10,AZGP1,AEBP2,PCBP4,FBXO36,MXRA7,ZP3,RERG,PINK1,LPGAT1,FUS,SECISBP2,ZBTB1,KANK4,ZNF804A,PLXNA1,DHDDS,TPMT,AC019205.1,FCHSD2,NFIL3,USP4,AHNAK,BET1,CYREN,NSG1,SRP54,IL1B,KLRD1,LINC00539,AP4E1,PPP6R2,NAP1L5,ABCA5,GBF1,ZBTB33,RHOB,TNPO1,TARBP1,GLIPR2,TMBIM6,ASCC1,TH,SH2D4A,PIGL,INTS6-AS1,CACNA1E,ZKSCAN1,CYFIP2,CCDC84,CD69,DDHD2,PCM1,CCDC136,PPFIA2,TUB,RBM25,CHGB,ZDHHC2,AC068025.2,ERBIN,TM7SF2,NUDT1,ARHGEF12,GCNA,CNKSR2,FNBP1,NAP1L1,SLC4A8,ZSCAN30,GPATCH2\n"
     ]
    }
   ],
   "source": [
    "#\n",
    "mgl_scpdsn_genes = mgl_scpdsn_sig[\"Gene\"].to_list()\n",
    "random.seed(999999)\n",
    "random.shuffle(mgl_scpdsn_genes)\n",
    "\n",
    "print(\",\".join(mgl_scpdsn_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "a98f6ac0-f4f4-4eb8-a08b-ab2774219781",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [set_descr, generatio, bgratio, richFactor, foldEnrich, p_val, p_adj, intersection, set_genes, ngenes, nset, ninter]\n",
       "Index: []"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gmt = llm2geneset.read_gmt(\"libs_human/gmt/KEGG_2021_Human.txt\")\n",
    "set_descr = gmt[\"descr\"]\n",
    "gene_sets = gmt[\"genes\"]\n",
    "df = llm2geneset.simple_ora(mgl_scpdsn_genes, set_descr, gene_sets)\n",
    "df.to_csv(\"outputs/KEGG_scRNA_mgl_PD.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "2c77fe30-7c05-42e0-9b17-a684354ac60e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:26<00:00, 26.88s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [set_descr, generatio, bgratio, richFactor, foldEnrich, p_val, p_adj, intersection, set_genes, ngenes, nset, ninter, in_toks, out_toks]\n",
       "Index: []"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "#\n",
    "context_str = \"\"\n",
    "llm2geneset_no_context_res = await llm2geneset.gs_proposal_bench(aclient, \n",
    "                                                                 [mgl_scpdsn_genes], \n",
    "                                                                 model=\"gpt-4o-2024-05-13\", \n",
    "                                                                 context=context_str,\n",
    "                                                                 n_pathways=100)\n",
    "\n",
    "df = llm2geneset_no_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/no_context_scRNA_mgl_PD.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "3165b273-2c5a-4f4d-9110-37648ea0a697",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:25<00:00, 25.96s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [set_descr, generatio, bgratio, richFactor, foldEnrich, p_val, p_adj, intersection, set_genes, ngenes, nset, ninter, in_toks, out_toks]\n",
       "Index: []"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "#\n",
    "context_str = \"Rheumatoid arthritis\"\n",
    "llm2geneset_context_res = await llm2geneset.gs_proposal_bench(aclient, [mgl_scpdsn_genes], model=\"gpt-4o-2024-05-13\", \n",
    "                                                              context=context_str,\n",
    "                                                              n_pathways=100)\n",
    "\n",
    "df = llm2geneset_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/wrong_context_scRNA_mgl_PD.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "471cac03-d3f4-4f35-b48a-d45816b6c2cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:27<00:00, 27.02s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>set_descr</th>\n",
       "      <th>generatio</th>\n",
       "      <th>bgratio</th>\n",
       "      <th>richFactor</th>\n",
       "      <th>foldEnrich</th>\n",
       "      <th>p_val</th>\n",
       "      <th>p_adj</th>\n",
       "      <th>intersection</th>\n",
       "      <th>set_genes</th>\n",
       "      <th>ngenes</th>\n",
       "      <th>nset</th>\n",
       "      <th>ninter</th>\n",
       "      <th>in_toks</th>\n",
       "      <th>out_toks</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Dopaminergic synapse</td>\n",
       "      <td>0.047619</td>\n",
       "      <td>0.00136</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>35.001764</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>0.008013</td>\n",
       "      <td>SLC6A3,TH,SLC18A2</td>\n",
       "      <td>NTRK2,DRD3,DRD5,GRK6,PRKN,PPP1R1B,PINK1,DJ1,VM...</td>\n",
       "      <td>63</td>\n",
       "      <td>27</td>\n",
       "      <td>3</td>\n",
       "      <td>160</td>\n",
       "      <td>246</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               set_descr  generatio  bgratio  richFactor  foldEnrich  \\\n",
       "41  Dopaminergic synapse   0.047619  0.00136    0.111111   35.001764   \n",
       "\n",
       "       p_val     p_adj       intersection  \\\n",
       "41  0.000085  0.008013  SLC6A3,TH,SLC18A2   \n",
       "\n",
       "                                            set_genes  ngenes  nset  ninter  \\\n",
       "41  NTRK2,DRD3,DRD5,GRK6,PRKN,PPP1R1B,PINK1,DJ1,VM...      63    27       3   \n",
       "\n",
       "    in_toks  out_toks  \n",
       "41      160       246  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#\n",
    "gc.collect()\n",
    "#\n",
    "context_str = \"differentially expressed genes from pseudobulk of microglia in Parkinson's disease\"\n",
    "llm2geneset_context_res = await llm2geneset.gs_proposal_bench(aclient, [mgl_scpdsn_genes], model=\"gpt-4o-2024-05-13\", \n",
    "                                                              context=context_str,\n",
    "                                                              n_pathways=100)\n",
    "\n",
    "df = llm2geneset_context_res[0][\"ora_results\"]\n",
    "df.to_csv(\"outputs/with_context_scRNA_mgl_PD.tsv\",sep=\"\\t\",index=None)\n",
    "df[df[\"p_adj\"] < 0.01]\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
